{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "99b38d2a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import bs4 as bs\n",
    "import urllib.request\n",
    "import re\n",
    "import nltk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2b8b630b",
   "metadata": {},
   "outputs": [],
   "source": [
    "scraped_data = urllib.request.urlopen('https://en.wikipedia.org/wiki/Severe_acute_respiratory_syndrome_coronavirus_2')\n",
    "article = scraped_data.read()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "9278caf6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# web scraping using beautifulSoap\n",
    "parsed_article = bs.BeautifulSoup(article,'lxml')\n",
    "\n",
    "paragraphs = parsed_article.find_all('p')\n",
    "\n",
    "article_text = \"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "bdbf0777",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[<p class=\"mw-empty-elt\">\n",
       " </p>,\n",
       " <p><b>Severe acute respiratory syndrome coronavirus 2</b> (<b>SARS‑CoV‑2</b>)<sup class=\"reference\" id=\"cite_ref-CoronavirusStudyGroup_2-0\"><a href=\"#cite_note-CoronavirusStudyGroup-2\">[2]</a></sup> is a strain of <a href=\"/wiki/Coronavirus\" title=\"Coronavirus\">coronavirus</a> that causes <a href=\"/wiki/COVID-19\" title=\"COVID-19\">COVID-19</a> (coronavirus disease 2019), the <a class=\"mw-redirect\" href=\"/wiki/Respiratory_illness\" title=\"Respiratory illness\">respiratory illness</a> responsible for the ongoing <a href=\"/wiki/COVID-19_pandemic\" title=\"COVID-19 pandemic\">COVID-19 pandemic</a>.<sup class=\"reference\" id=\"cite_ref-NYT-20210226_3-0\"><a href=\"#cite_note-NYT-20210226-3\">[3]</a></sup> The virus previously had a <a href=\"/wiki/Novel_coronavirus\" title=\"Novel coronavirus\">provisional name</a>, <b>2019 novel coronavirus</b> (<b>2019-nCoV</b>),<sup class=\"reference\" id=\"cite_ref-WHO21Jan2020_4-0\"><a href=\"#cite_note-WHO21Jan2020-4\">[4]</a></sup><sup class=\"reference\" id=\"cite_ref-CDC-nCoV_5-0\"><a href=\"#cite_note-CDC-nCoV-5\">[5]</a></sup><sup class=\"reference\" id=\"cite_ref-CDCAbout_6-0\"><a href=\"#cite_note-CDCAbout-6\">[6]</a></sup><sup class=\"reference\" id=\"cite_ref-NYT6_7-0\"><a href=\"#cite_note-NYT6-7\">[7]</a></sup> and has also been called the <b>human coronavirus 2019</b> (<b>HCoV-19</b> or <b>hCoV-19</b>).<sup class=\"reference\" id=\"cite_ref-Zoonotic_8-0\"><a href=\"#cite_note-Zoonotic-8\">[8]</a></sup><sup class=\"reference\" id=\"cite_ref-Proximal_9-0\"><a href=\"#cite_note-Proximal-9\">[9]</a></sup><sup class=\"reference\" id=\"cite_ref-NEJM-Stability_10-0\"><a href=\"#cite_note-NEJM-Stability-10\">[10]</a></sup><sup class=\"reference\" id=\"cite_ref-China_natl_GeneBank_11-0\"><a href=\"#cite_note-China_natl_GeneBank-11\">[11]</a></sup>  First identified in the city of <a href=\"/wiki/Wuhan\" title=\"Wuhan\">Wuhan</a>, <a href=\"/wiki/Hubei\" title=\"Hubei\">Hubei</a>, China, the <a href=\"/wiki/World_Health_Organization\" title=\"World Health Organization\">World Health Organization</a> declared the outbreak a <a href=\"/wiki/Public_health_emergency_of_international_concern\" title=\"Public health emergency of international concern\">public health emergency of international concern</a> on January 30, 2020, and a <a href=\"/wiki/Pandemic\" title=\"Pandemic\">pandemic</a> on March 11, 2020.<sup class=\"reference\" id=\"cite_ref-WHO-PHEIC_12-0\"><a href=\"#cite_note-WHO-PHEIC-12\">[12]</a></sup><sup class=\"reference\" id=\"cite_ref-WHOPandemic_13-0\"><a href=\"#cite_note-WHOPandemic-13\">[13]</a></sup> SARS‑CoV‑2 is a <a class=\"mw-redirect\" href=\"/wiki/Positive-sense_single-stranded_RNA_virus\" title=\"Positive-sense single-stranded RNA virus\">positive-sense single-stranded RNA virus</a><sup class=\"reference\" id=\"cite_ref-Machhi2020Sep_14-0\"><a href=\"#cite_note-Machhi2020Sep-14\">[14]</a></sup> that is <a href=\"/wiki/Contagious_disease\" title=\"Contagious disease\">contagious</a> in humans.<sup class=\"reference\" id=\"cite_ref-Chan24Jan2020_15-0\"><a href=\"#cite_note-Chan24Jan2020-15\">[15]</a></sup>\n",
       " </p>,\n",
       " <p>SARS‑CoV‑2 is a virus of the species <i><a class=\"mw-redirect\" href=\"/wiki/Severe_acute_respiratory_syndrome%E2%80%93related_coronavirus\" title=\"Severe acute respiratory syndrome–related coronavirus\">severe acute respiratory syndrome–related coronavirus</a></i> (SARSr-CoV), related to the <a class=\"mw-redirect\" href=\"/wiki/Severe_acute_respiratory_syndrome_coronavirus_1\" title=\"Severe acute respiratory syndrome coronavirus 1\">SARS-CoV-1</a> virus that caused the <a href=\"/wiki/2002%E2%80%932004_SARS_outbreak\" title=\"2002–2004 SARS outbreak\">2002–2004 SARS outbreak</a>.<sup class=\"reference\" id=\"cite_ref-CoronavirusStudyGroup_2-1\"><a href=\"#cite_note-CoronavirusStudyGroup-2\">[2]</a></sup><sup class=\"reference\" id=\"cite_ref-nihSARSr-CoV_16-0\"><a href=\"#cite_note-nihSARSr-CoV-16\">[16]</a></sup> Despite its close relation to SARS-CoV-1, its closest known relatives, with which it forms a sister group, are the derived SARS viruses BANAL-52 and RaTG13.<sup class=\"reference\" id=\"cite_ref-17\"><a href=\"#cite_note-17\">[17]</a></sup> Available evidence indicates that it is most likely of <a href=\"/wiki/Zoonosis\" title=\"Zoonosis\">zoonotic</a> origin and has close <a href=\"/wiki/Sequence_homology\" title=\"Sequence homology\">genetic similarity</a> to bat coronaviruses, suggesting it emerged from a <a href=\"/wiki/Bat_virome\" title=\"Bat virome\">bat-borne virus</a>.<sup class=\"reference\" id=\"cite_ref-biol_18-0\"><a href=\"#cite_note-biol-18\">[18]</a></sup> <a href=\"/wiki/Investigations_into_the_origin_of_COVID-19\" title=\"Investigations into the origin of COVID-19\">Research is ongoing</a> as to whether SARS‑CoV‑2 came directly from bats or indirectly through any intermediate hosts.<sup class=\"reference\" id=\"cite_ref-WHO-SR22_19-0\"><a href=\"#cite_note-WHO-SR22-19\">[19]</a></sup> The virus shows little <a href=\"/wiki/Genetic_diversity\" title=\"Genetic diversity\">genetic diversity</a>, indicating that the <a href=\"/wiki/Spillover_infection\" title=\"Spillover infection\">spillover event</a> introducing SARS‑CoV‑2 to humans is likely to have occurred in late 2019.<sup class=\"reference\" id=\"cite_ref-early_20-0\"><a href=\"#cite_note-early-20\">[20]</a></sup>\n",
       " </p>,\n",
       " <p><a class=\"mw-redirect\" href=\"/wiki/Epidemiological\" title=\"Epidemiological\">Epidemiological</a> studies estimate that, in the December 2019–September 2020 period, each infection resulted in an average of 2.4–3.4 new infections when no members of the community are <a href=\"/wiki/Immunity_(medical)\" title=\"Immunity (medical)\">immune</a> and no <a class=\"mw-redirect\" href=\"/wiki/Infection_control\" title=\"Infection control\">preventive measures</a> are taken.<sup class=\"reference\" id=\"cite_ref-BillahMiahKhan_21-0\"><a href=\"#cite_note-BillahMiahKhan-21\">[21]</a></sup> However, some subsequent variants have become more infectious.<sup class=\"reference\" id=\"cite_ref-22\"><a href=\"#cite_note-22\">[22]</a></sup> The virus is airborne and primarily spreads between people through close contact and via <a class=\"mw-redirect\" href=\"/wiki/Aerosol_transmission\" title=\"Aerosol transmission\">aerosols</a> and <a class=\"mw-redirect\" href=\"/wiki/Respiratory_droplets\" title=\"Respiratory droplets\">respiratory droplets</a> that are exhaled when talking, breathing, or otherwise exhaling, as well as those produced from coughs and sneezes.<sup class=\"reference\" id=\"cite_ref-23\"><a href=\"#cite_note-23\">[23]</a></sup><sup class=\"reference\" id=\"cite_ref-24\"><a href=\"#cite_note-24\">[24]</a></sup> It <a href=\"/wiki/Viral_entry\" title=\"Viral entry\">enters</a> human cells by binding to <a href=\"/wiki/Angiotensin-converting_enzyme_2\" title=\"Angiotensin-converting enzyme 2\">angiotensin-converting enzyme 2</a> (ACE2), a membrane protein that regulates the renin–angiotensin system.<sup class=\"reference\" id=\"cite_ref-HoffmanCell_25-0\"><a href=\"#cite_note-HoffmanCell-25\">[25]</a></sup><sup class=\"reference\" id=\"cite_ref-26\"><a href=\"#cite_note-26\">[26]</a></sup>\n",
       " </p>,\n",
       " <p>During the initial outbreak in <a href=\"/wiki/Wuhan\" title=\"Wuhan\">Wuhan</a>, China, various names were used for the virus; some names used by different sources included \"the coronavirus\" or \"Wuhan coronavirus\".<sup class=\"reference\" id=\"cite_ref-HuangNPR_27-0\"><a href=\"#cite_note-HuangNPR-27\">[27]</a></sup><sup class=\"reference\" id=\"cite_ref-Fox2020_28-0\"><a href=\"#cite_note-Fox2020-28\">[28]</a></sup> In January 2020, the <a href=\"/wiki/World_Health_Organization\" title=\"World Health Organization\">World Health Organization</a> (WHO) recommended \"2019 novel coronavirus\" (2019-nCoV)<sup class=\"reference\" id=\"cite_ref-CDC-nCoV_5-1\"><a href=\"#cite_note-CDC-nCoV-5\">[5]</a></sup><sup class=\"reference\" id=\"cite_ref-WHO-SR10_29-0\"><a href=\"#cite_note-WHO-SR10-29\">[29]</a></sup> as the provisional name for the virus. This was in accordance with WHO's 2015 guidance<sup class=\"reference\" id=\"cite_ref-WHOnamingguidelines_30-0\"><a href=\"#cite_note-WHOnamingguidelines-30\">[30]</a></sup> against using geographical locations, animal species, or groups of people in disease and virus names.<sup class=\"reference\" id=\"cite_ref-TodayNameMixup_31-0\"><a href=\"#cite_note-TodayNameMixup-31\">[31]</a></sup><sup class=\"reference\" id=\"cite_ref-EconomistSinophobia_32-0\"><a href=\"#cite_note-EconomistSinophobia-32\">[32]</a></sup>\n",
       " </p>,\n",
       " <p>On 11 February 2020, the <a href=\"/wiki/International_Committee_on_Taxonomy_of_Viruses\" title=\"International Committee on Taxonomy of Viruses\">International Committee on Taxonomy of Viruses</a> adopted the official name \"severe acute respiratory syndrome coronavirus 2\" (SARS‑CoV‑2).<sup class=\"reference\" id=\"cite_ref-COVIDname_date_33-0\"><a href=\"#cite_note-COVIDname_date-33\">[33]</a></sup> To avoid confusion with the disease <a href=\"/wiki/SARS\" title=\"SARS\">SARS</a>, the WHO sometimes refers to SARS‑CoV‑2 as \"the COVID-19 virus\" in public health communications<sup class=\"reference\" id=\"cite_ref-QZ-Why_34-0\"><a href=\"#cite_note-QZ-Why-34\">[34]</a></sup><sup class=\"reference\" id=\"cite_ref-COVIDname_public_35-0\"><a href=\"#cite_note-COVIDname_public-35\">[35]</a></sup> and the name HCoV-19 was included in some research articles.<sup class=\"reference\" id=\"cite_ref-Zoonotic_8-1\"><a href=\"#cite_note-Zoonotic-8\">[8]</a></sup><sup class=\"reference\" id=\"cite_ref-Proximal_9-1\"><a href=\"#cite_note-Proximal-9\">[9]</a></sup><sup class=\"reference\" id=\"cite_ref-NEJM-Stability_10-1\"><a href=\"#cite_note-NEJM-Stability-10\">[10]</a></sup> Referring to COVID-19 as the \"Wuhan virus\" has been described as dangerous by WHO officials, and as <a href=\"/wiki/Xenophobia\" title=\"Xenophobia\">xenophobic</a> by University of California at Berkeley Asian American studies lecturer Harvey Dong.<sup class=\"reference\" id=\"cite_ref-36\"><a href=\"#cite_note-36\">[36]</a></sup><sup class=\"reference\" id=\"cite_ref-37\"><a href=\"#cite_note-37\">[37]</a></sup><sup class=\"reference\" id=\"cite_ref-38\"><a href=\"#cite_note-38\">[38]</a></sup>\n",
       " </p>,\n",
       " <p>Human-to-human <a class=\"mw-redirect\" href=\"/wiki/Transmission_(medicine)\" title=\"Transmission (medicine)\">transmission</a> of SARS‑CoV‑2 was confirmed on 20 January 2020 during the <a href=\"/wiki/COVID-19_pandemic\" title=\"COVID-19 pandemic\">COVID-19 pandemic</a>.<sup class=\"reference\" id=\"cite_ref-Chan24Jan2020_15-1\"><a href=\"#cite_note-Chan24Jan2020-15\">[15]</a></sup><sup class=\"reference\" id=\"cite_ref-LiMar2020_39-0\"><a href=\"#cite_note-LiMar2020-39\">[39]</a></sup><sup class=\"reference\" id=\"cite_ref-Kessler17Apr2020_40-0\"><a href=\"#cite_note-Kessler17Apr2020-40\">[40]</a></sup><sup class=\"reference\" id=\"cite_ref-Kuo21Jan2020_41-0\"><a href=\"#cite_note-Kuo21Jan2020-41\">[41]</a></sup> Transmission was initially assumed to occur primarily via <a href=\"/wiki/Respiratory_droplet\" title=\"Respiratory droplet\">respiratory droplets</a> from coughs and sneezes within a range of about 1.8 metres (6 ft).<sup class=\"reference\" id=\"cite_ref-CDCTrans_42-0\"><a href=\"#cite_note-CDCTrans-42\">[42]</a></sup><sup class=\"reference\" id=\"cite_ref-NBCSpread_43-0\"><a href=\"#cite_note-NBCSpread-43\">[43]</a></sup> Laser light scattering experiments suggest that <a href=\"/wiki/Speech\" title=\"Speech\">speaking</a> is an additional mode of transmission<sup class=\"reference\" id=\"cite_ref-Anfrinrud_2020_44-0\"><a href=\"#cite_note-Anfrinrud_2020-44\">[44]</a></sup><sup class=\"reference\" id=\"cite_ref-Stadnytskyi_et_al_2020_45-0\"><a href=\"#cite_note-Stadnytskyi_et_al_2020-45\">[45]</a></sup> and a far-reaching<sup class=\"reference\" id=\"cite_ref-Klompas_46-0\"><a href=\"#cite_note-Klompas-46\">[46]</a></sup> one, indoors, with little air flow.<sup class=\"reference\" id=\"cite_ref-LivescienceTalkingWorse_47-0\"><a href=\"#cite_note-LivescienceTalkingWorse-47\">[47]</a></sup><sup class=\"reference\" id=\"cite_ref-deOliveira_48-0\"><a href=\"#cite_note-deOliveira-48\">[48]</a></sup> Other studies have suggested that the virus may be <a class=\"mw-redirect\" href=\"/wiki/Airborne_pathogen\" title=\"Airborne pathogen\">airborne</a> as well, with <a class=\"mw-redirect\" href=\"/wiki/Aerosols\" title=\"Aerosols\">aerosols</a> potentially being able to transmit the virus.<sup class=\"reference\" id=\"cite_ref-NYT-20200704am_49-0\"><a href=\"#cite_note-NYT-20200704am-49\">[49]</a></sup><sup class=\"reference\" id=\"cite_ref-ZZeynep_50-0\"><a href=\"#cite_note-ZZeynep-50\">[50]</a></sup><sup class=\"reference\" id=\"cite_ref-lew-d-mou_51-0\"><a href=\"#cite_note-lew-d-mou-51\">[51]</a></sup> During human-to-human transmission, between 200 and 800 infectious SARS‑CoV‑2 <a class=\"mw-redirect\" href=\"/wiki/Virion\" title=\"Virion\">virions</a> are thought to initiate a new infection.<sup class=\"reference\" id=\"cite_ref-SCI-20201123_52-0\"><a href=\"#cite_note-SCI-20201123-52\">[52]</a></sup><sup class=\"reference\" id=\"cite_ref-NatureTempShedding_53-0\"><a href=\"#cite_note-NatureTempShedding-53\">[53]</a></sup><sup class=\"reference\" id=\"cite_ref-54\"><a href=\"#cite_note-54\">[54]</a></sup> If confirmed, aerosol transmission has biosafety implications because a major concern associated with the risk of working with emerging viruses in the laboratory is the generation of aerosols from various laboratory activities which are not immediately recognizable and may affect other scientific personnel.<sup class=\"reference\" id=\"cite_ref-55\"><a href=\"#cite_note-55\">[55]</a></sup> Indirect contact via <a href=\"/wiki/Fomite\" title=\"Fomite\">contaminated surfaces</a> is another possible cause of infection.<sup class=\"reference\" id=\"cite_ref-WHO-Workplace_56-0\"><a href=\"#cite_note-WHO-Workplace-56\">[56]</a></sup> Preliminary research indicates that the virus may remain viable on plastic (<a href=\"/wiki/Polypropylene\" title=\"Polypropylene\">polypropylene</a>) and <a href=\"/wiki/Stainless_steel\" title=\"Stainless steel\">stainless steel</a> (<a href=\"/wiki/SAE_304_stainless_steel\" title=\"SAE 304 stainless steel\">AISI 304</a>) for up to three days, but it does not survive on cardboard for more than one day or on copper for more than four hours.<sup class=\"reference\" id=\"cite_ref-NEJM-Stability_10-2\"><a href=\"#cite_note-NEJM-Stability-10\">[10]</a></sup> The virus is inactivated by soap, which destabilizes its <a href=\"/wiki/Lipid_bilayer\" title=\"Lipid bilayer\">lipid bilayer</a>.<sup class=\"reference\" id=\"cite_ref-AtlanticSuccess_57-0\"><a href=\"#cite_note-AtlanticSuccess-57\">[57]</a></sup><sup class=\"reference\" id=\"cite_ref-NatGeoSoap_58-0\"><a href=\"#cite_note-NatGeoSoap-58\">[58]</a></sup> Viral <a href=\"/wiki/RNA\" title=\"RNA\">RNA</a> has also been found in <a href=\"/wiki/Stool_test\" title=\"Stool test\">stool samples</a> and semen from infected individuals.<sup class=\"reference\" id=\"cite_ref-NEJM-FirstUS_59-0\"><a href=\"#cite_note-NEJM-FirstUS-59\">[59]</a></sup><sup class=\"reference\" id=\"cite_ref-Semen_60-0\"><a href=\"#cite_note-Semen-60\">[60]</a></sup>\n",
       " </p>,\n",
       " <p>The degree to which the virus is infectious during the <a href=\"/wiki/Incubation_period\" title=\"Incubation period\">incubation period</a> is uncertain, but research has indicated that the <a href=\"/wiki/Pharynx\" title=\"Pharynx\">pharynx</a> reaches peak <a href=\"/wiki/Viral_load\" title=\"Viral load\">viral load</a> approximately four days after infection<sup class=\"reference\" id=\"cite_ref-NaturePeakLoad_61-0\"><a href=\"#cite_note-NaturePeakLoad-61\">[61]</a></sup><sup class=\"reference\" id=\"cite_ref-ScienceFlawed_62-0\"><a href=\"#cite_note-ScienceFlawed-62\">[62]</a></sup> or in the first week of symptoms and declines thereafter.<sup class=\"reference\" id=\"cite_ref-LancetLoad_63-0\"><a href=\"#cite_note-LancetLoad-63\">[63]</a></sup> The duration of SARS-CoV-2 <a href=\"/wiki/Viral_shedding\" title=\"Viral shedding\">RNA shedding</a> is generally between 3 and 46 days after symptom onset.<sup class=\"reference\" id=\"cite_ref-64\"><a href=\"#cite_note-64\">[64]</a></sup>\n",
       " </p>,\n",
       " <p>A study by a team of researchers from the <a href=\"/wiki/University_of_North_Carolina\" title=\"University of North Carolina\">University of North Carolina</a> found that the <a href=\"/wiki/Nasal_cavity\" title=\"Nasal cavity\">nasal cavity</a> is seemingly the dominant initial site of infection, with subsequent <a href=\"/wiki/Pulmonary_aspiration\" title=\"Pulmonary aspiration\">aspiration</a>-mediated virus-seeding into the lungs in SARS‑CoV‑2 pathogenesis.<sup class=\"reference\" id=\"cite_ref-Hou2020_65-0\"><a href=\"#cite_note-Hou2020-65\">[65]</a></sup> They found that there was an infection gradient from high in proximal towards low in distal pulmonary epithelial cultures, with a focal infection in ciliated cells and type 2 pneumocytes in the airway and alveolar regions respectively.<sup class=\"reference\" id=\"cite_ref-Hou2020_65-1\"><a href=\"#cite_note-Hou2020-65\">[65]</a></sup>\n",
       " </p>,\n",
       " <p>Studies have identified a range of animals—such as cats, ferrets, hamsters, non-human primates, minks, tree shrews, raccoon dogs, fruit bats, and rabbits—that are susceptible and permissive to SARS-CoV-2 infection.<sup class=\"reference\" id=\"cite_ref-66\"><a href=\"#cite_note-66\">[66]</a></sup><sup class=\"reference\" id=\"cite_ref-oie.int_67-0\"><a href=\"#cite_note-oie.int-67\">[67]</a></sup><sup class=\"reference\" id=\"cite_ref-BronxTiger_68-0\"><a href=\"#cite_note-BronxTiger-68\">[68]</a></sup> Some institutions have advised that those infected with SARS‑CoV‑2 restrict their contact with animals.<sup class=\"reference\" id=\"cite_ref-USDAtiger_69-0\"><a href=\"#cite_note-USDAtiger-69\">[69]</a></sup><sup class=\"reference\" id=\"cite_ref-CDCanimals_70-0\"><a href=\"#cite_note-CDCanimals-70\">[70]</a></sup>\n",
       " </p>,\n",
       " <p>On 1<span class=\"nowrap\"> </span>February 2020, the <a href=\"/wiki/World_Health_Organization\" title=\"World Health Organization\">World Health Organization</a> (WHO) indicated that \"transmission from <a href=\"/wiki/Asymptomatic\" title=\"Asymptomatic\">asymptomatic</a> cases is likely not a major driver of transmission\".<sup class=\"reference\" id=\"cite_ref-WHO-SR12_71-0\"><a href=\"#cite_note-WHO-SR12-71\">[71]</a></sup> One meta-analysis found that 17% of infections are asymptomatic, and asymptomatic individuals were 42% less likely to transmit the virus.<sup class=\"reference\" id=\"cite_ref-Nogrady2020Nov_72-0\"><a href=\"#cite_note-Nogrady2020Nov-72\">[72]</a></sup>\n",
       " </p>,\n",
       " <p>However, an epidemiological model of the beginning of the <a href=\"/wiki/COVID-19_pandemic_in_mainland_China\" title=\"COVID-19 pandemic in mainland China\">outbreak in China</a> suggested that \"pre-symptomatic <a href=\"/wiki/Viral_shedding\" title=\"Viral shedding\">shedding</a> may be typical among documented infections\" and that <a href=\"/wiki/Subclinical_infection\" title=\"Subclinical infection\">subclinical infections</a> may have been the source of a majority of infections.<sup class=\"reference\" id=\"cite_ref-ScienceRapid_73-0\"><a href=\"#cite_note-ScienceRapid-73\">[73]</a></sup> That may explain how out of 217 on board a <a class=\"mw-redirect\" href=\"/wiki/Cruise_liner\" title=\"Cruise liner\">cruise liner</a> that docked at <a href=\"/wiki/Montevideo\" title=\"Montevideo\">Montevideo</a>, only 24 of 128 who tested positive for viral RNA showed symptoms.<sup class=\"reference\" id=\"cite_ref-Thorax_74-0\"><a href=\"#cite_note-Thorax-74\">[74]</a></sup> Similarly, a study of ninety-four patients hospitalized in January and February 2020 estimated patients began shedding virus two to three days before symptoms appear and that \"a substantial proportion of transmission probably occurred before first symptoms in the <a href=\"/wiki/Index_case\" title=\"Index case\">index case</a>\".<sup class=\"reference\" id=\"cite_ref-NatureTempShedding_53-1\"><a href=\"#cite_note-NatureTempShedding-53\">[53]</a></sup> The authors later published a correction that showed that shedding began earlier than first estimated, four to five days before symptoms appear.<sup class=\"reference\" id=\"cite_ref-75\"><a href=\"#cite_note-75\">[75]</a></sup>\n",
       " </p>,\n",
       " <p>There is uncertainty about reinfection and long-term immunity.<sup class=\"reference\" id=\"cite_ref-led-rei_76-0\"><a href=\"#cite_note-led-rei-76\">[76]</a></sup> It is not known how common reinfection is, but reports have indicated that it is occurring with variable severity.<sup class=\"reference\" id=\"cite_ref-led-rei_76-1\"><a href=\"#cite_note-led-rei-76\">[76]</a></sup>\n",
       " </p>,\n",
       " <p>The first reported case of reinfection was a 33-year-old man from Hong Kong who first tested positive on 26 March 2020, was discharged on 15 April 2020 after two negative tests, and tested positive again on 15 August 2020 (142 days later), which was confirmed by whole-genome sequencing showing that the viral genomes between the episodes belong to different <a href=\"/wiki/Clade\" title=\"Clade\">clades</a>.<sup class=\"reference\" id=\"cite_ref-to-phyl_77-0\"><a href=\"#cite_note-to-phyl-77\">[77]</a></sup> The findings had the implications that <a href=\"/wiki/Herd_immunity\" title=\"Herd immunity\">herd immunity</a> may not eliminate the virus if reinfection is not an uncommon occurrence and that <a href=\"/wiki/COVID-19_vaccine\" title=\"COVID-19 vaccine\">vaccines</a> may not be able to provide lifelong protection against the virus.<sup class=\"reference\" id=\"cite_ref-to-phyl_77-1\"><a href=\"#cite_note-to-phyl-77\">[77]</a></sup>\n",
       " </p>,\n",
       " <p>Another case study described a 25-year-old man from Nevada who tested positive for SARS‑CoV‑2 on 18 April 2020 and on 5 June 2020 (separated by two negative tests). Since genomic analyses showed significant genetic differences between the SARS‑CoV‑2 variant sampled on those two dates, the case study authors determined this was a reinfection.<sup class=\"reference\" id=\"cite_ref-till-gen_78-0\"><a href=\"#cite_note-till-gen-78\">[78]</a></sup> The man's second infection was symptomatically more severe than the first infection, but the mechanisms that could account for this are not known.<sup class=\"reference\" id=\"cite_ref-till-gen_78-1\"><a href=\"#cite_note-till-gen-78\">[78]</a></sup>\n",
       " </p>,\n",
       " <p><span class=\"anchor\" id=\"origin\"></span>\n",
       " </p>,\n",
       " <p>No <a href=\"/wiki/Natural_reservoir\" title=\"Natural reservoir\">natural reservoir</a> for SARS-CoV-2 has been identified.<sup class=\"reference\" id=\"cite_ref-Holmes2021_79-0\"><a href=\"#cite_note-Holmes2021-79\">[79]</a></sup> Prior to the emergence of SARS-CoV-2 as a pathogen infecting humans, there had been two previous <a href=\"/wiki/Zoonosis\" title=\"Zoonosis\">zoonosis</a>-based coronavirus epidemics, those caused by <a href=\"/wiki/SARS-CoV-1\" title=\"SARS-CoV-1\">SARS-CoV-1</a> and <a class=\"mw-redirect\" href=\"/wiki/MERS-CoV\" title=\"MERS-CoV\">MERS-CoV</a>.<sup class=\"reference\" id=\"cite_ref-biol_18-1\"><a href=\"#cite_note-biol-18\">[18]</a></sup>\n",
       " </p>,\n",
       " <p>The first known infections from SARS‑CoV‑2 were discovered in Wuhan, China.<sup class=\"reference\" id=\"cite_ref-NatureZhou_80-0\"><a href=\"#cite_note-NatureZhou-80\">[80]</a></sup> The original source of viral transmission to humans remains unclear, as does whether the virus became <a href=\"/wiki/Pathogen\" title=\"Pathogen\">pathogenic</a> before or after the <a class=\"mw-redirect\" href=\"/wiki/Spillover_event\" title=\"Spillover event\">spillover event</a>.<sup class=\"reference\" id=\"cite_ref-Proximal_9-2\"><a href=\"#cite_note-Proximal-9\">[9]</a></sup><sup class=\"reference\" id=\"cite_ref-early_20-1\"><a href=\"#cite_note-early-20\">[20]</a></sup><sup class=\"reference\" id=\"cite_ref-PopSciJan_81-0\"><a href=\"#cite_note-PopSciJan-81\">[81]</a></sup> Because many of the early infectees were workers at the <a class=\"mw-redirect\" href=\"/wiki/Huanan_Seafood_Market\" title=\"Huanan Seafood Market\">Huanan Seafood Market</a>,<sup class=\"reference\" id=\"cite_ref-LancetClinical_82-0\"><a href=\"#cite_note-LancetClinical-82\">[82]</a></sup><sup class=\"reference\" id=\"cite_ref-LancetCharacteristics_83-0\"><a href=\"#cite_note-LancetCharacteristics-83\">[83]</a></sup> it has been suggested that the virus might have originated from the market.<sup class=\"reference\" id=\"cite_ref-Proximal_9-3\"><a href=\"#cite_note-Proximal-9\">[9]</a></sup><sup class=\"reference\" id=\"cite_ref-nature_feb2020_84-0\"><a href=\"#cite_note-nature_feb2020-84\">[84]</a></sup> However, other research indicates that visitors may have introduced the virus to the market, which then facilitated rapid expansion of the infections.<sup class=\"reference\" id=\"cite_ref-early_20-2\"><a href=\"#cite_note-early-20\">[20]</a></sup><sup class=\"reference\" id=\"cite_ref-XivDecoding_85-0\"><a href=\"#cite_note-XivDecoding-85\">[85]</a></sup> A March 2021 WHO-convened report stated that human spillover via an intermediate animal host was the most likely explanation, with direct spillover from bats next most likely. Introduction through the food supply chain and the Huanan Seafood Market was considered another possible, but less likely, explanation.<sup class=\"reference\" id=\"cite_ref-WHOconvened_86-0\"><a href=\"#cite_note-WHOconvened-86\">[86]</a></sup> An analysis in November 2021, however, said that the earliest-known case had been misidentified and that the preponderance of early cases linked to the Huanan Market argued for it being the source.<sup class=\"reference\" id=\"cite_ref-87\"><a href=\"#cite_note-87\">[87]</a></sup>\n",
       " </p>,\n",
       " <p>For a virus recently acquired through a cross-species transmission, rapid evolution is expected.<sup class=\"reference\" id=\"cite_ref-88\"><a href=\"#cite_note-88\">[88]</a></sup> The mutation rate estimated from early cases of SARS-CoV-2 was of <span class=\"nowrap\"><span data-sort-value=\"6996654000000000000♠\"></span>6.54<span style=\"margin-left:0.25em;margin-right:0.15em;\">×</span>10<sup>−4</sup></span> per site per year.<sup class=\"reference\" id=\"cite_ref-WHOconvened_86-1\"><a href=\"#cite_note-WHOconvened-86\">[86]</a></sup> Coronaviruses in general have high genetic <a href=\"/wiki/Phenotypic_plasticity\" title=\"Phenotypic plasticity\">plasticity</a>,<sup class=\"reference\" id=\"cite_ref-89\"><a href=\"#cite_note-89\">[89]</a></sup> but SARS-CoV-2's viral evolution is slowed by the <a href=\"/wiki/Proofreading_(biology)\" title=\"Proofreading (biology)\">RNA proofreading</a> capability of its replication machinery.<sup class=\"reference\" id=\"cite_ref-RobsonF2020Aug_90-0\"><a href=\"#cite_note-RobsonF2020Aug-90\">[90]</a></sup> For comparison, the viral mutation rate in vivo of SARS-CoV-2 has been found to be lower than that of influenza.<sup class=\"reference\" id=\"cite_ref-91\"><a href=\"#cite_note-91\">[91]</a></sup>\n",
       " </p>,\n",
       " <p>Research into the natural reservoir of the virus that caused the <a href=\"/wiki/2002%E2%80%932004_SARS_outbreak\" title=\"2002–2004 SARS outbreak\">2002–2004 SARS outbreak</a> has resulted in the discovery of many <a href=\"/wiki/Bat_SARS-like_coronavirus_WIV1\" title=\"Bat SARS-like coronavirus WIV1\">SARS-like bat coronaviruses</a>, most originating in <a href=\"/wiki/Horseshoe_bat\" title=\"Horseshoe bat\">horseshoe bats</a>. The closest match by far, published on <i><a href=\"/wiki/Nature_(journal)\" title=\"Nature (journal)\">Nature (journal)</a></i> in February 2022, were viruses <a class=\"new\" href=\"/w/index.php?title=BANAL-52&amp;action=edit&amp;redlink=1\" title=\"BANAL-52 (page does not exist)\">BANAL-52</a> (96.8% resemblance to SARS‑CoV‑2), BANAL-103 and BANAL-236, collected in three different species of bats in <a href=\"/wiki/Feuang_district\" title=\"Feuang district\">Feuang</a>, Laos.<sup class=\"reference\" id=\"cite_ref-92\"><a href=\"#cite_note-92\">[92]</a></sup><sup class=\"reference\" id=\"cite_ref-93\"><a href=\"#cite_note-93\">[93]</a></sup><sup class=\"reference\" id=\"cite_ref-94\"><a href=\"#cite_note-94\">[94]</a></sup> An earlier source published in February 2020 identified the virus <a href=\"/wiki/RaTG13\" title=\"RaTG13\">RaTG13</a>, collected in bats in <a class=\"mw-redirect\" href=\"/wiki/Mojiang_County\" title=\"Mojiang County\">Mojiang</a>, Yunnan, China to be the closest to SARS‑CoV‑2, with 96.1% resemblance.<sup class=\"reference\" id=\"cite_ref-NatureZhou_80-1\"><a href=\"#cite_note-NatureZhou-80\">[80]</a></sup><sup class=\"reference\" id=\"cite_ref-NCBI-Bat3_95-0\"><a href=\"#cite_note-NCBI-Bat3-95\">[95]</a></sup> None of the above are its direct ancestor.<sup class=\"reference\" id=\"cite_ref-96\"><a href=\"#cite_note-96\">[96]</a></sup>\n",
       " </p>,\n",
       " <p>Bats are considered the most likely natural reservoir of SARS‑CoV‑2.<sup class=\"reference\" id=\"cite_ref-WHOconvened_86-2\"><a href=\"#cite_note-WHOconvened-86\">[86]</a></sup><sup class=\"reference\" id=\"cite_ref-LancetBinding_97-0\"><a href=\"#cite_note-LancetBinding-97\">[97]</a></sup> Differences between the bat coronavirus and SARS‑CoV‑2 suggest that humans may have been infected via an intermediate host;<sup class=\"reference\" id=\"cite_ref-nature_feb2020_84-1\"><a href=\"#cite_note-nature_feb2020-84\">[84]</a></sup> although the source of introduction into humans remains unknown.<sup class=\"reference\" id=\"cite_ref-OKeeffeJ2021Mar_98-0\"><a href=\"#cite_note-OKeeffeJ2021Mar-98\">[98]</a></sup><sup class=\"reference\" id=\"cite_ref-Holmes2021_79-1\"><a href=\"#cite_note-Holmes2021-79\">[79]</a></sup>\n",
       " </p>,\n",
       " <p>Although the role of <a class=\"mw-redirect\" href=\"/wiki/Pangolins\" title=\"Pangolins\">pangolins</a> as an intermediate host was initially posited (a study published in July 2020 suggested that pangolins are an intermediate host of SARS‑CoV‑2-like coronaviruses<sup class=\"reference\" id=\"cite_ref-XiaoK2020July_99-0\"><a href=\"#cite_note-XiaoK2020July-99\">[99]</a></sup><sup class=\"reference\" id=\"cite_ref-ZhaoJ2020_100-0\"><a href=\"#cite_note-ZhaoJ2020-100\">[100]</a></sup>), subsequent studies have not substantiated their contribution to the spillover.<sup class=\"reference\" id=\"cite_ref-WHOconvened_86-3\"><a href=\"#cite_note-WHOconvened-86\">[86]</a></sup> Evidence against this hypothesis includes the fact that pangolin virus samples are too distant to SARS-CoV-2: isolates obtained from pangolins seized in <a href=\"/wiki/Guangdong\" title=\"Guangdong\">Guangdong</a> were only 92% identical in sequence to the SARS‑CoV‑2 genome (matches above 90 percent may sound high, but in genomic terms it is a wide evolutionary gap<sup class=\"reference\" id=\"cite_ref-101\"><a href=\"#cite_note-101\">[101]</a></sup>). In addition, despite similarities in a few critical amino acids,<sup class=\"reference\" id=\"cite_ref-HuB2020Oct_102-0\"><a href=\"#cite_note-HuB2020Oct-102\">[102]</a></sup> pangolin virus samples exhibit poor binding to the human ACE2 receptor.<sup class=\"reference\" id=\"cite_ref-GiovanettiM2020Nov_103-0\"><a href=\"#cite_note-GiovanettiM2020Nov-103\">[103]</a></sup>\n",
       " </p>,\n",
       " <p>SARS‑CoV‑2 belongs to the broad family of viruses known as <a href=\"/wiki/Coronavirus\" title=\"Coronavirus\">coronaviruses</a>.<sup class=\"reference\" id=\"cite_ref-Fox2020_28-1\"><a href=\"#cite_note-Fox2020-28\">[28]</a></sup> It is a <a class=\"mw-redirect\" href=\"/wiki/Positive-sense_single-stranded_RNA_virus\" title=\"Positive-sense single-stranded RNA virus\">positive-sense single-stranded RNA</a> (+ssRNA) virus, with a single linear RNA segment. Coronaviruses infect humans, other mammals, including livestock and companion animals, and avian species.<sup class=\"reference\" id=\"cite_ref-V'kovskik:21_104-0\"><a href=\"#cite_note-V'kovskik:21-104\">[104]</a></sup> Human coronaviruses are capable of causing illnesses ranging from the <a href=\"/wiki/Common_cold\" title=\"Common cold\">common cold</a> to more severe diseases such as <a class=\"mw-redirect\" href=\"/wiki/Middle_East_respiratory_syndrome\" title=\"Middle East respiratory syndrome\">Middle East respiratory syndrome</a> (MERS, fatality rate ~34%). SARS-CoV-2 is the seventh known coronavirus to infect people, after <a href=\"/wiki/Human_coronavirus_229E\" title=\"Human coronavirus 229E\">229E</a>, <a href=\"/wiki/Human_coronavirus_NL63\" title=\"Human coronavirus NL63\">NL63</a>, <a href=\"/wiki/Human_coronavirus_OC43\" title=\"Human coronavirus OC43\">OC43</a>, <a href=\"/wiki/Human_coronavirus_HKU1\" title=\"Human coronavirus HKU1\">HKU1</a>, <a class=\"mw-redirect\" href=\"/wiki/Middle_East_respiratory_syndrome-related_coronavirus\" title=\"Middle East respiratory syndrome-related coronavirus\">MERS-CoV</a>, and the original <a class=\"mw-redirect\" href=\"/wiki/Severe_acute_respiratory_syndrome_coronavirus\" title=\"Severe acute respiratory syndrome coronavirus\">SARS-CoV</a>.<sup class=\"reference\" id=\"cite_ref-NEJM-Novel_105-0\"><a href=\"#cite_note-NEJM-Novel-105\">[105]</a></sup>\n",
       " </p>,\n",
       " <p>Like the SARS-related coronavirus implicated in the 2003 SARS outbreak, SARS‑CoV‑2 is a member of the subgenus <i><a class=\"mw-redirect\" href=\"/wiki/Sarbecovirus\" title=\"Sarbecovirus\">Sarbecovirus</a></i> (<a class=\"mw-redirect\" href=\"/wiki/Beta-CoV\" title=\"Beta-CoV\">beta-CoV</a> lineage B).<sup class=\"reference\" id=\"cite_ref-NextstrainPhylogeny_106-0\"><a href=\"#cite_note-NextstrainPhylogeny-106\">[106]</a></sup><sup class=\"reference\" id=\"cite_ref-Wong2019_107-0\"><a href=\"#cite_note-Wong2019-107\">[107]</a></sup> Coronaviruses undergo frequent recombination.<sup class=\"reference\" id=\"cite_ref-Singhy:21_108-0\"><a href=\"#cite_note-Singhy:21-108\">[108]</a></sup> The mechanism of recombination in unsegmented RNA viruses such as SARS-CoV-2 is generally by copy-choice replication, in which gene material switches from one RNA template molecule to another during replication.<sup class=\"reference\" id=\"cite_ref-109\"><a href=\"#cite_note-109\">[109]</a></sup> The SARS-CoV-2 RNA sequence is approximately 30,000 <a href=\"/wiki/Nucleobase\" title=\"Nucleobase\">bases</a> in length,<sup class=\"reference\" id=\"cite_ref-gisaid_110-0\"><a href=\"#cite_note-gisaid-110\">[110]</a></sup> relatively long for a coronavirus—which in turn carry the largest genomes among all RNA families.<sup class=\"reference\" id=\"cite_ref-111\"><a href=\"#cite_note-111\">[111]</a></sup> Its genome consists nearly entirely of protein-coding sequences, a trait shared with other coronaviruses.<sup class=\"reference\" id=\"cite_ref-Singhy:21_108-1\"><a href=\"#cite_note-Singhy:21-108\">[108]</a></sup>\n",
       " </p>,\n",
       " <p>A distinguishing feature of SARS‑CoV‑2 is its incorporation of a <a href=\"/wiki/Amino_acid#Side_chains\" title=\"Amino acid\">polybasic</a> site cleaved by <a href=\"/wiki/Furin\" title=\"Furin\">furin</a>,<sup class=\"reference\" id=\"cite_ref-HuB2020Oct_102-1\"><a href=\"#cite_note-HuB2020Oct-102\">[102]</a></sup><sup class=\"reference\" id=\"cite_ref-112\"><a href=\"#cite_note-112\">[112]</a></sup> which appears to be an important element enhancing its virulence.<sup class=\"reference\" id=\"cite_ref-KaiWangTo2020_113-0\"><a href=\"#cite_note-KaiWangTo2020-113\">[113]</a></sup> It was suggested that the acquisition of the furin-cleavage site in the SARS-CoV-2 S protein was essential for zoonotic transfer to humans.<sup class=\"reference\" id=\"cite_ref-Mechanisms_of_SARS-CoV-2_entry_into_114-0\"><a href=\"#cite_note-Mechanisms_of_SARS-CoV-2_entry_into-114\">[114]</a></sup> The furin <a href=\"/wiki/Protease\" title=\"Protease\">protease</a> recognizes the canonical <a href=\"/wiki/Peptide\" title=\"Peptide\">peptide</a> sequence <a href=\"/wiki/Arginine\" title=\"Arginine\">R</a>X[<a href=\"/wiki/Arginine\" title=\"Arginine\">R</a>/<a href=\"/wiki/Lysine\" title=\"Lysine\">K</a>] <a href=\"/wiki/Arginine\" title=\"Arginine\">R</a>↓X where the cleavage site is indicated by a down arrow and X is any <a href=\"/wiki/Amino_acid\" title=\"Amino acid\">amino acid</a>.<sup class=\"reference\" id=\"cite_ref-115\"><a href=\"#cite_note-115\">[115]</a></sup><sup class=\"reference\" id=\"cite_ref-116\"><a href=\"#cite_note-116\">[116]</a></sup> In SARS-CoV-2 the recognition site is formed by the incorporated 12 <a href=\"/wiki/Genetic_code\" title=\"Genetic code\">codon nucleotide sequence</a> CCT CGG CGG GCA which corresponds to the amino acid sequence <a href=\"/wiki/Proline\" title=\"Proline\">P</a> <a href=\"/wiki/Arginine\" title=\"Arginine\">RR</a> <a href=\"/wiki/Alanine\" title=\"Alanine\">A</a>.<sup class=\"reference\" id=\"cite_ref-:2_117-0\"><a href=\"#cite_note-:2-117\">[117]</a></sup> This sequence is upstream of an arginine and serine which forms the S1/S2 cleavage site (<a href=\"/wiki/Proline\" title=\"Proline\">P</a> <a href=\"/wiki/Arginine\" title=\"Arginine\">RR</a> <a href=\"/wiki/Alanine\" title=\"Alanine\">A</a> <a href=\"/wiki/Arginine\" title=\"Arginine\">R</a>↓<a href=\"/wiki/Serine\" title=\"Serine\">S</a>) of the spike protein.<sup class=\"reference\" id=\"cite_ref-PangolinResidue_118-0\"><a href=\"#cite_note-PangolinResidue-118\">[118]</a></sup> Although such sites are a common naturally-occurring feature of other viruses within the Subfamily Orthocoronavirinae,<sup class=\"reference\" id=\"cite_ref-:2_117-1\"><a href=\"#cite_note-:2-117\">[117]</a></sup> it appears in few other viruses from the <a href=\"/wiki/Betacoronavirus\" title=\"Betacoronavirus\">Beta-CoV</a> genus,<sup class=\"reference\" id=\"cite_ref-119\"><a href=\"#cite_note-119\">[119]</a></sup> and it is unique among members of its subgenus for such a site.<sup class=\"reference\" id=\"cite_ref-HuB2020Oct_102-2\"><a href=\"#cite_note-HuB2020Oct-102\">[102]</a></sup> The furin cleavage site PRRAR↓ is highly similar to that of the <a href=\"/wiki/Feline_coronavirus\" title=\"Feline coronavirus\">feline coronavirus</a>, an <a href=\"/wiki/Alphacoronavirus_1\" title=\"Alphacoronavirus 1\">alphacoronavirus 1</a> strain.<sup class=\"reference\" id=\"cite_ref-feline_coronavirus_120-0\"><a href=\"#cite_note-feline_coronavirus-120\">[120]</a></sup>\n",
       " </p>,\n",
       " <p>Viral genetic sequence data can provide critical information about whether viruses separated by time and space are likely to be epidemiologically linked.<sup class=\"reference\" id=\"cite_ref-121\"><a href=\"#cite_note-121\">[121]</a></sup> With a sufficient number of sequenced <a href=\"/wiki/Genome\" title=\"Genome\">genomes</a>, it is possible to reconstruct a <a href=\"/wiki/Phylogenetic_tree\" title=\"Phylogenetic tree\">phylogenetic tree</a> of the mutation history of a family of viruses. By 12 January 2020, five genomes of SARS‑CoV‑2 had been isolated from Wuhan and reported by the <a href=\"/wiki/Chinese_Center_for_Disease_Control_and_Prevention\" title=\"Chinese Center for Disease Control and Prevention\">Chinese Center for Disease Control and Prevention</a> (CCDC) and other institutions;<sup class=\"reference\" id=\"cite_ref-gisaid_110-1\"><a href=\"#cite_note-gisaid-110\">[110]</a></sup><sup class=\"reference\" id=\"cite_ref-VirologicalInitial_122-0\"><a href=\"#cite_note-VirologicalInitial-122\">[122]</a></sup> the number of genomes increased to 42 by 30 January 2020.<sup class=\"reference\" id=\"cite_ref-NextstrainJanuary_123-0\"><a href=\"#cite_note-NextstrainJanuary-123\">[123]</a></sup> A phylogenetic analysis of those samples showed they were \"highly related with at most seven mutations relative to a <a class=\"mw-redirect\" href=\"/wiki/Common_ancestor\" title=\"Common ancestor\">common ancestor</a>\", implying that the first human infection occurred in November or December 2019.<sup class=\"reference\" id=\"cite_ref-NextstrainJanuary_123-1\"><a href=\"#cite_note-NextstrainJanuary-123\">[123]</a></sup> Examination of the topology of the phylogenetic tree at the start of the pandemic also found high similarities between human isolates.<sup class=\"reference\" id=\"cite_ref-Sun2020Trends_124-0\"><a href=\"#cite_note-Sun2020Trends-124\">[124]</a></sup>  As of 21 August 2021,<sup class=\"plainlinks noexcerpt noprint asof-tag update\" style=\"display:none;\"><a class=\"external text\" href=\"https://en.wikipedia.org/w/index.php?title=SARS-CoV-2&amp;action=edit\">[update]</a></sup> 3,422 SARS‑CoV‑2 genomes, belonging to 19 strains, sampled on all continents except Antarctica were publicly available.<sup class=\"reference\" id=\"cite_ref-NexstrainApril_125-0\"><a href=\"#cite_note-NexstrainApril-125\">[125]</a></sup>\n",
       " </p>,\n",
       " <p>On 11 February 2020, the <a href=\"/wiki/International_Committee_on_Taxonomy_of_Viruses\" title=\"International Committee on Taxonomy of Viruses\">International Committee on Taxonomy of Viruses</a> announced that according to existing rules that compute hierarchical relationships among coronaviruses based on five <a href=\"/wiki/Conserved_sequence\" title=\"Conserved sequence\">conserved sequences</a> of nucleic acids, the differences between what was then called 2019-nCoV and the virus from the 2003 SARS outbreak were insufficient to make them separate <a class=\"mw-redirect\" href=\"/wiki/Viral_species\" title=\"Viral species\">viral species</a>. Therefore, they identified 2019-nCoV as a virus of <i><a class=\"mw-redirect\" href=\"/wiki/Severe_acute_respiratory_syndrome%E2%80%93related_coronavirus\" title=\"Severe acute respiratory syndrome–related coronavirus\">Severe acute respiratory syndrome–related coronavirus</a></i>.<sup class=\"reference\" id=\"cite_ref-126\"><a href=\"#cite_note-126\">[126]</a></sup>\n",
       " </p>,\n",
       " <p>In July 2020, scientists reported that a more infectious SARS‑CoV‑2 variant with <a href=\"/wiki/Coronavirus_spike_protein\" title=\"Coronavirus spike protein\">spike protein</a> variant G614 has replaced D614 as the dominant form in the pandemic.<sup class=\"reference\" id=\"cite_ref-Medicalxpress.com_127-0\"><a href=\"#cite_note-Medicalxpress.com-127\">[127]</a></sup><sup class=\"reference\" id=\"cite_ref-Korber_et_al_2020_128-0\"><a href=\"#cite_note-Korber_et_al_2020-128\">[128]</a></sup>\n",
       " </p>,\n",
       " <p>Coronavirus genomes and subgenomes encode six <a href=\"/wiki/Open_reading_frame\" title=\"Open reading frame\">open reading frames</a> (ORFs).<sup class=\"reference\" id=\"cite_ref-Dhama2020_129-0\"><a href=\"#cite_note-Dhama2020-129\">[129]</a></sup> In October 2020, researchers discovered a possible <a href=\"/wiki/Overlapping_gene\" title=\"Overlapping gene\">overlapping gene</a> named <i>ORF3d</i>, in the SARS‑CoV‑2 <a href=\"/wiki/Genome\" title=\"Genome\">genome</a>. It is unknown if the protein produced by <i>ORF3d</i> has any function, but it provokes a strong immune response. <i>ORF3d</i> has been identified before, in a variant of coronavirus that infects <a href=\"/wiki/Pangolin\" title=\"Pangolin\">pangolins</a>.<sup class=\"reference\" id=\"cite_ref-SA-20201111_130-0\"><a href=\"#cite_note-SA-20201111-130\">[130]</a></sup><sup class=\"reference\" id=\"cite_ref-EL-20201001_131-0\"><a href=\"#cite_note-EL-20201001-131\">[131]</a></sup>\n",
       " </p>,\n",
       " <p>A phylogenetic tree based on whole-genome sequences of SARS-CoV-2 and related coronaviruses is:<sup class=\"reference\" id=\"cite_ref-RpYN06_132-0\"><a href=\"#cite_note-RpYN06-132\">[132]</a></sup><sup class=\"reference\" id=\"cite_ref-RacCS203_133-0\"><a href=\"#cite_note-RacCS203-133\">[133]</a></sup>\n",
       " </p>,\n",
       " <p>(<a href=\"/wiki/Bat\" title=\"Bat\">Bat</a>) <a href=\"/wiki/Rc-o319\" title=\"Rc-o319\">Rc-o319</a>, 81% to SARS-CoV-2, <i><a class=\"mw-redirect\" href=\"/wiki/Rhinolophus_cornutus\" title=\"Rhinolophus cornutus\">Rhinolophus cornutus</a></i>, <a href=\"/wiki/Iwate_Prefecture\" title=\"Iwate Prefecture\">Iwate</a>, <a href=\"/wiki/Japan\" title=\"Japan\">Japan</a><sup class=\"reference\" id=\"cite_ref-Rc-o319_134-0\"><a href=\"#cite_note-Rc-o319-134\">[134]</a></sup>\n",
       " </p>,\n",
       " <p>Bat <a href=\"/wiki/ZC45_and_ZXC21\" title=\"ZC45 and ZXC21\">SL-ZXC21</a>, 88% to SARS-CoV-2, <i><a class=\"mw-redirect\" href=\"/wiki/Rhinolophus_pusillus\" title=\"Rhinolophus pusillus\">Rhinolophus pusillus</a></i>, <a href=\"/wiki/Zhoushan\" title=\"Zhoushan\">Zhoushan</a>, <a href=\"/wiki/Zhejiang\" title=\"Zhejiang\">Zhejiang</a><sup class=\"reference\" id=\"cite_ref-ZXC21_135-0\"><a href=\"#cite_note-ZXC21-135\">[135]</a></sup>\n",
       " </p>,\n",
       " <p>Bat <a href=\"/wiki/ZC45_and_ZXC21\" title=\"ZC45 and ZXC21\">SL-ZC45</a>, 88% to SARS-CoV-2, <i>Rhinolophus pusillus</i>, Zhoushan, Zhejiang<sup class=\"reference\" id=\"cite_ref-ZXC21_135-1\"><a href=\"#cite_note-ZXC21-135\">[135]</a></sup>\n",
       " </p>,\n",
       " <p><a href=\"/wiki/Pangolin\" title=\"Pangolin\">Pangolin</a> SARSr-CoV-GX, 85.3% to SARS-CoV-2, <i><a class=\"mw-redirect\" href=\"/wiki/Manis_javanica\" title=\"Manis javanica\">Manis javanica</a></i>, smuggled from Southeast Asia<sup class=\"reference\" id=\"cite_ref-Pangolin_CoV_GX_136-0\"><a href=\"#cite_note-Pangolin_CoV_GX-136\">[136]</a></sup>\n",
       " </p>,\n",
       " <p>Pangolin SARSr-CoV-GD, 90.1% to SARS-CoV-2, <i>Manis javanica</i>, smuggled from Southeast Asia<sup class=\"reference\" id=\"cite_ref-Pangolin_CoV_GD_137-0\"><a href=\"#cite_note-Pangolin_CoV_GD-137\">[137]</a></sup>\n",
       " </p>,\n",
       " <p>Bat RshSTT182, 92.6% to SARS-CoV-2, <i><a class=\"mw-redirect\" href=\"/wiki/Rhinolophus_shameli\" title=\"Rhinolophus shameli\">Rhinolophus shameli</a></i>, <a class=\"mw-redirect\" href=\"/wiki/Stung_Treng_Province\" title=\"Stung Treng Province\">Steung Treng</a>, <a href=\"/wiki/Cambodia\" title=\"Cambodia\">Cambodia</a><sup class=\"reference\" id=\"cite_ref-RshSTT_138-0\"><a href=\"#cite_note-RshSTT-138\">[138]</a></sup>\n",
       " </p>,\n",
       " <p>Bat RshSTT200, 92.6% to SARS-CoV-2, <i>Rhinolophus shameli</i>, Steung Treng, Cambodia<sup class=\"reference\" id=\"cite_ref-RshSTT_138-1\"><a href=\"#cite_note-RshSTT-138\">[138]</a></sup>\n",
       " </p>,\n",
       " <p>(Bat) <a href=\"/wiki/RacCS203\" title=\"RacCS203\">RacCS203</a>, 91.5% to SARS-CoV-2, <i><a class=\"mw-redirect\" href=\"/wiki/Rhinolophus_acuminatus\" title=\"Rhinolophus acuminatus\">Rhinolophus acuminatus</a></i>, <a class=\"mw-redirect\" href=\"/wiki/Chachoengsao_Province\" title=\"Chachoengsao Province\">Chachoengsao</a>, <a href=\"/wiki/Thailand\" title=\"Thailand\">Thailand</a><sup class=\"reference\" id=\"cite_ref-RacCS203_133-1\"><a href=\"#cite_note-RacCS203-133\">[133]</a></sup>\n",
       " </p>,\n",
       " <p>(Bat) <a href=\"/wiki/RmYN02\" title=\"RmYN02\">RmYN02</a>, 93.3% to SARS-CoV-2, <i><a class=\"mw-redirect\" href=\"/wiki/Rhinolophus_malayanus\" title=\"Rhinolophus malayanus\">Rhinolophus malayanus</a></i>, <a href=\"/wiki/Mengla_County\" title=\"Mengla County\">Mengla</a>, <a href=\"/wiki/Yunnan\" title=\"Yunnan\">Yunnan</a><sup class=\"reference\" id=\"cite_ref-RmYN02_139-0\"><a href=\"#cite_note-RmYN02-139\">[139]</a></sup>\n",
       " </p>,\n",
       " <p>(Bat) <a href=\"/wiki/RpYN06\" title=\"RpYN06\">RpYN06</a>, 94.4% to SARS-CoV-2, <i>Rhinolophus pusillus</i>, <a class=\"mw-redirect\" href=\"/wiki/Xishuangbanna\" title=\"Xishuangbanna\">Xishuangbanna</a>, Yunnan<sup class=\"reference\" id=\"cite_ref-RpYN06_132-1\"><a href=\"#cite_note-RpYN06-132\">[132]</a></sup>\n",
       " </p>,\n",
       " <p>(Bat) <a href=\"/wiki/RaTG13\" title=\"RaTG13\">RaTG13</a>, 96.1% to SARS-CoV-2, <i><a class=\"mw-redirect\" href=\"/wiki/Rhinolophus_affinis\" title=\"Rhinolophus affinis\">Rhinolophus affinis</a></i>, <a href=\"/wiki/Mojiang_Hani_Autonomous_County\" title=\"Mojiang Hani Autonomous County\">Mojiang</a>, Yunnan<sup class=\"reference\" id=\"cite_ref-RaTG13_140-0\"><a href=\"#cite_note-RaTG13-140\">[140]</a></sup>\n",
       " </p>,\n",
       " <p>(Bat) <a class=\"new\" href=\"/w/index.php?title=BANAL-52&amp;action=edit&amp;redlink=1\" title=\"BANAL-52 (page does not exist)\">BANAL-52</a>, 96.8% to SARS-CoV-2, <i><a class=\"mw-redirect\" href=\"/wiki/Rhinolophus_malayanus\" title=\"Rhinolophus malayanus\">Rhinolophus malayanus</a></i>, <a href=\"/wiki/Vientiane_province\" title=\"Vientiane province\">Vientiane</a>, <a href=\"/wiki/Laos\" title=\"Laos\">Laos</a><sup class=\"reference\" id=\"cite_ref-BANAL-52_141-0\"><a href=\"#cite_note-BANAL-52-141\">[141]</a></sup>\n",
       " </p>,\n",
       " <p><a class=\"mw-redirect\" href=\"/wiki/Severe_acute_respiratory_syndrome_coronavirus_2\" title=\"Severe acute respiratory syndrome coronavirus 2\">SARS-CoV-2</a>\n",
       " </p>,\n",
       " <p><a class=\"mw-redirect\" href=\"/wiki/Severe_acute_respiratory_syndrome_coronavirus_1\" title=\"Severe acute respiratory syndrome coronavirus 1\">SARS-CoV-1</a>, 79% to SARS-CoV-2\n",
       " </p>,\n",
       " <p><br/>\n",
       " </p>,\n",
       " <p>There are many thousands of variants of SARS-CoV-2, which can be grouped into the much larger <a href=\"/wiki/Clade\" title=\"Clade\">clades</a>.<sup class=\"reference\" id=\"cite_ref-Koyama2020June_142-0\"><a href=\"#cite_note-Koyama2020June-142\">[142]</a></sup> Several different <a class=\"mw-redirect\" href=\"/wiki/Variants_of_severe_acute_respiratory_syndrome_coronavirus_2#Nomenclature\" title=\"Variants of severe acute respiratory syndrome coronavirus 2\">clade nomenclatures</a> have been proposed. Nextstrain divides the variants into five clades (19A, 19B, 20A, 20B, and 20C), while <a href=\"/wiki/GISAID\" title=\"GISAID\">GISAID</a> divides them into seven (L, O, V, S, G, GH, and GR).<sup class=\"reference\" id=\"cite_ref-Alm2020Aug_143-0\"><a href=\"#cite_note-Alm2020Aug-143\">[143]</a></sup>\n",
       " </p>,\n",
       " <p>Several notable variants of SARS-CoV-2 emerged in late 2020. The <a href=\"/wiki/World_Health_Organization\" title=\"World Health Organization\">World Health Organization</a> has currently declared five <a href=\"/wiki/Variant_of_concern\" title=\"Variant of concern\">variants of concern</a>, which are as follows:<sup class=\"reference\" id=\"cite_ref-144\"><a href=\"#cite_note-144\">[144]</a></sup>\n",
       " </p>,\n",
       " <p>Other notable variants include 6 other WHO-designated <a href=\"/wiki/Variants_of_SARS-CoV-2\" title=\"Variants of SARS-CoV-2\">variants under investigation</a> and <a href=\"/wiki/Cluster_5\" title=\"Cluster 5\">Cluster 5</a>, which emerged among <a href=\"/wiki/Mink\" title=\"Mink\">mink</a> in Denmark and resulted in a mink euthanasia campaign rendering it virtually extinct.<sup class=\"reference\" id=\"cite_ref-WHO2020_Cluster5_145-0\"><a href=\"#cite_note-WHO2020_Cluster5-145\">[145]</a></sup>\n",
       " </p>,\n",
       " <p>Each SARS-CoV-2 <a class=\"mw-redirect\" href=\"/wiki/Virion\" title=\"Virion\">virion</a> is 60–140 <a href=\"/wiki/Nanometre\" title=\"Nanometre\">nanometres</a> (2.4<span style=\"margin:0 .15em 0 .25em\">×</span>10<sup>−6</sup>–5.5<span style=\"margin:0 .15em 0 .25em\">×</span>10<sup>−6</sup> in) in diameter;<sup class=\"reference\" id=\"cite_ref-NEJM-Novel_105-1\"><a href=\"#cite_note-NEJM-Novel-105\">[105]</a></sup><sup class=\"reference\" id=\"cite_ref-LancetCharacteristics_83-1\"><a href=\"#cite_note-LancetCharacteristics-83\">[83]</a></sup> its mass within the global human populace has been estimated as being between 0.1 and 10 kilograms.<sup class=\"reference\" id=\"cite_ref-146\"><a href=\"#cite_note-146\">[146]</a></sup> Like other coronaviruses, SARS-CoV-2 has four structural proteins, known as the S (<a href=\"/wiki/Coronavirus_spike_protein\" title=\"Coronavirus spike protein\">spike</a>), E (<a href=\"/wiki/Coronavirus_envelope_protein\" title=\"Coronavirus envelope protein\">envelope</a>), M (<a href=\"/wiki/Coronavirus_membrane_protein\" title=\"Coronavirus membrane protein\">membrane</a>), and N (<a href=\"/wiki/Coronavirus_nucleocapsid_protein\" title=\"Coronavirus nucleocapsid protein\">nucleocapsid</a>) proteins; the N protein holds the RNA genome, and the S, E, and M proteins together create the <a href=\"/wiki/Viral_envelope\" title=\"Viral envelope\">viral envelope</a>.<sup class=\"reference\" id=\"cite_ref-WuStructure_147-0\"><a href=\"#cite_note-WuStructure-147\">[147]</a></sup> Coronavirus S proteins are <a href=\"/wiki/Glycoprotein\" title=\"Glycoprotein\">glycoproteins</a> and also type I <a href=\"/wiki/Membrane_protein\" title=\"Membrane protein\">membrane proteins</a> (membranes containing a single <a href=\"/wiki/Transmembrane_domain\" title=\"Transmembrane domain\">transmembrane domain</a> oriented on the extracellular side).<sup class=\"reference\" id=\"cite_ref-Mechanisms_of_SARS-CoV-2_entry_into_114-1\"><a href=\"#cite_note-Mechanisms_of_SARS-CoV-2_entry_into-114\">[114]</a></sup> They are divided into two functional parts (S1 and S2).<sup class=\"reference\" id=\"cite_ref-V'kovskik:21_104-1\"><a href=\"#cite_note-V'kovskik:21-104\">[104]</a></sup> In SARS-CoV-2, the spike protein, which has been imaged at the atomic level using <a href=\"/wiki/Cryogenic_electron_microscopy\" title=\"Cryogenic electron microscopy\">cryogenic electron microscopy</a>,<sup class=\"reference\" id=\"cite_ref-SCI-20200219_148-0\"><a href=\"#cite_note-SCI-20200219-148\">[148]</a></sup><sup class=\"reference\" id=\"cite_ref-GZM-20200220_149-0\"><a href=\"#cite_note-GZM-20200220-149\">[149]</a></sup> is the protein responsible for allowing the virus to attach to and fuse with the <a href=\"/wiki/Cell_membrane\" title=\"Cell membrane\">membrane</a> of a host cell;<sup class=\"reference\" id=\"cite_ref-WuStructure_147-1\"><a href=\"#cite_note-WuStructure-147\">[147]</a></sup> specifically, its S1 <a href=\"/wiki/Protein_subunit\" title=\"Protein subunit\">subunit</a> catalyzes attachment, the S2 subunit fusion.<sup class=\"reference\" id=\"cite_ref-CEBMcoronaviruses_150-0\"><a href=\"#cite_note-CEBMcoronaviruses-150\">[150]</a></sup>\n",
       " </p>,\n",
       " <p>As of early 2022, about 7 million SARS-CoV-2 genomes had been sequenced and deposited into public databases and another 800,000 or so were added each month.<sup class=\"reference\" id=\"cite_ref-151\"><a href=\"#cite_note-151\">[151]</a></sup>\n",
       " </p>,\n",
       " <p>SARS-CoV-2 has a linear, <a href=\"/wiki/Sense_(molecular_biology)\" title=\"Sense (molecular biology)\">positive-sense</a>, single-stranded RNA genome about 30,000 bases long.<sup class=\"reference\" id=\"cite_ref-V'kovskik:21_104-2\"><a href=\"#cite_note-V'kovskik:21-104\">[104]</a></sup> Its genome has a bias against <a href=\"/wiki/GC-content\" title=\"GC-content\">cytosine (C) and guanine (G) nucleotides</a>, like other coronaviruses.<sup class=\"reference\" id=\"cite_ref-152\"><a href=\"#cite_note-152\">[152]</a></sup> The genome has the highest composition of <a href=\"/wiki/Uridine\" title=\"Uridine\">U</a> (32.2%), followed by <a href=\"/wiki/Adenosine\" title=\"Adenosine\">A</a> (29.9%), and a similar composition of <a href=\"/wiki/Guanine\" title=\"Guanine\">G</a> (19.6%) and <a href=\"/wiki/Cytosine\" title=\"Cytosine\">C</a> (18.3%).<sup class=\"reference\" id=\"cite_ref-:02_153-0\"><a href=\"#cite_note-:02-153\">[153]</a></sup> The <a href=\"/wiki/GC_skew\" title=\"GC skew\">nucleotide bias</a> arises from the mutation of guanines and cytosines to adenosines and <a href=\"/wiki/Uracil\" title=\"Uracil\">uracils</a>, respectively.<sup class=\"reference\" id=\"cite_ref-:12_154-0\"><a href=\"#cite_note-:12-154\">[154]</a></sup> The mutation of <a class=\"mw-redirect\" href=\"/wiki/CpG_dinucleotides\" title=\"CpG dinucleotides\">CG dinucleotides</a> is thought to arise to avoid the <a href=\"/wiki/Antiviral_protein\" title=\"Antiviral protein\">zinc finger antiviral protein</a> related defense mechanism of cells,<sup class=\"reference\" id=\"cite_ref-155\"><a href=\"#cite_note-155\">[155]</a></sup> and to lower the energy to unbind the genome during <a class=\"mw-redirect\" href=\"/wiki/RNA_replication\" title=\"RNA replication\">replication</a> and <a href=\"/wiki/Translation_(biology)\" title=\"Translation (biology)\">translation</a> (adenosine and uracil <a href=\"/wiki/Base_pair\" title=\"Base pair\">base pair</a> via two <a href=\"/wiki/Hydrogen_bond\" title=\"Hydrogen bond\">hydrogen bonds</a>, cytosine and guanine via three).<sup class=\"reference\" id=\"cite_ref-:12_154-1\"><a href=\"#cite_note-:12-154\">[154]</a></sup> The depletion of CG dinucleotides in its genome has led the virus to have a noticeable <a href=\"/wiki/Codon_usage_bias\" title=\"Codon usage bias\">codon usage bias</a>. For instance, arginine's six different codons have a <a href=\"/wiki/Codon_usage_bias#Methods_of_analysis\" title=\"Codon usage bias\">relative synonymous codon usage</a> of AGA (2.67), CGU (1.46), AGG (.81), CGC (.58), CGA (.29), and CGG (.19).<sup class=\"reference\" id=\"cite_ref-:02_153-1\"><a href=\"#cite_note-:02-153\">[153]</a></sup> A similar codon usage bias trend is seen in other SARS–related coronaviruses.<sup class=\"reference\" id=\"cite_ref-156\"><a href=\"#cite_note-156\">[156]</a></sup>\n",
       " </p>,\n",
       " <p>Virus infections start when viral particles bind to host surface cellular receptors.<sup class=\"reference\" id=\"cite_ref-157\"><a href=\"#cite_note-157\">[157]</a></sup> <a href=\"/wiki/Protein_structure_prediction\" title=\"Protein structure prediction\">Protein modeling</a> experiments on the spike protein of the virus soon suggested that SARS‑CoV‑2 has sufficient affinity to the receptor <a class=\"mw-redirect\" href=\"/wiki/Angiotensin_converting_enzyme_2\" title=\"Angiotensin converting enzyme 2\">angiotensin converting enzyme 2</a> (ACE2) on human cells to use them as a mechanism of <a href=\"/wiki/Viral_entry\" title=\"Viral entry\">cell entry</a>.<sup class=\"reference\" id=\"cite_ref-SCLSModeling_158-0\"><a href=\"#cite_note-SCLSModeling-158\">[158]</a></sup> By 22 January 2020, a group in China working with the full virus genome and a group in the United States using <a href=\"/wiki/Reverse_genetics\" title=\"Reverse genetics\">reverse genetics</a> methods independently and experimentally demonstrated that ACE2 could act as the receptor for SARS‑CoV‑2.<sup class=\"reference\" id=\"cite_ref-NatureZhou_80-2\"><a href=\"#cite_note-NatureZhou-80\">[80]</a></sup><sup class=\"reference\" id=\"cite_ref-Letko22Jan2020_159-0\"><a href=\"#cite_note-Letko22Jan2020-159\">[159]</a></sup><sup class=\"reference\" id=\"cite_ref-NatMicLetko_160-0\"><a href=\"#cite_note-NatMicLetko-160\">[160]</a></sup><sup class=\"reference\" id=\"cite_ref-ElSahly_161-0\"><a href=\"#cite_note-ElSahly-161\">[161]</a></sup> Studies have shown that SARS‑CoV‑2 has a higher affinity to human ACE2 than the original SARS virus.<sup class=\"reference\" id=\"cite_ref-SCI-20200219_148-1\"><a href=\"#cite_note-SCI-20200219-148\">[148]</a></sup><sup class=\"reference\" id=\"cite_ref-NIH-Structure_162-0\"><a href=\"#cite_note-NIH-Structure-162\">[162]</a></sup> SARS‑CoV‑2 may also use <a href=\"/wiki/Basigin\" title=\"Basigin\">basigin</a> to assist in cell entry.<sup class=\"reference\" id=\"cite_ref-CD147_163-0\"><a href=\"#cite_note-CD147-163\">[163]</a></sup>\n",
       " </p>,\n",
       " <p>Initial spike protein priming by <a href=\"/wiki/TMPRSS2\" title=\"TMPRSS2\">transmembrane protease, serine 2</a> (TMPRSS2) is essential for entry of SARS‑CoV‑2.<sup class=\"reference\" id=\"cite_ref-HoffmanCell_25-1\"><a href=\"#cite_note-HoffmanCell-25\">[25]</a></sup> The host protein <a href=\"/wiki/Neuropilin_1\" title=\"Neuropilin 1\">neuropilin 1</a> (NRP1) may aid the virus in host cell entry using ACE2.<sup class=\"reference\" id=\"cite_ref-CuervoNZ2020Nov_164-0\"><a href=\"#cite_note-CuervoNZ2020Nov-164\">[164]</a></sup> After a SARS‑CoV‑2 virion attaches to a target cell, the cell's TMPRSS2 cuts open the spike protein of the virus, exposing a <a class=\"mw-redirect\" href=\"/wiki/Fusion_peptide\" title=\"Fusion peptide\">fusion peptide</a> in the S2 subunit, and the host receptor ACE2.<sup class=\"reference\" id=\"cite_ref-CEBMcoronaviruses_150-1\"><a href=\"#cite_note-CEBMcoronaviruses-150\">[150]</a></sup> After fusion, an <a href=\"/wiki/Endosome\" title=\"Endosome\">endosome</a> forms around the virion, separating it from the rest of the host cell. The virion escapes when the <a href=\"/wiki/PH\" title=\"PH\">pH</a> of the endosome drops or when <a href=\"/wiki/Cathepsin\" title=\"Cathepsin\">cathepsin</a>, a host <a href=\"/wiki/Cysteine\" title=\"Cysteine\">cysteine</a> protease, cleaves it.<sup class=\"reference\" id=\"cite_ref-CEBMcoronaviruses_150-2\"><a href=\"#cite_note-CEBMcoronaviruses-150\">[150]</a></sup> The virion then releases RNA into the cell and forces the cell to produce and disseminate <a href=\"/wiki/Viral_replication\" title=\"Viral replication\">copies of the virus</a>, which infect more cells.<sup class=\"reference\" id=\"cite_ref-econ-anatomy-killer_165-0\"><a href=\"#cite_note-econ-anatomy-killer-165\">[165]</a></sup>\n",
       " </p>,\n",
       " <p>SARS‑CoV‑2 produces at least three <a href=\"/wiki/Virulence_factor\" title=\"Virulence factor\">virulence factors</a> that promote shedding of new virions from host cells and inhibit <a href=\"/wiki/Immune_response\" title=\"Immune response\">immune response</a>.<sup class=\"reference\" id=\"cite_ref-WuStructure_147-2\"><a href=\"#cite_note-WuStructure-147\">[147]</a></sup> Whether they include <a class=\"mw-redirect\" href=\"/wiki/Downregulation\" title=\"Downregulation\">downregulation</a> of ACE2, as seen in similar coronaviruses, remains under investigation (as of May 2020).<sup class=\"reference\" id=\"cite_ref-BMJ-Best-Practice_166-0\"><a href=\"#cite_note-BMJ-Best-Practice-166\">[166]</a></sup>\n",
       " </p>,\n",
       " <p>Very few drugs are known to effectively inhibit SARS‑CoV‑2. <a href=\"/wiki/Masitinib\" title=\"Masitinib\">Masitinib</a> is a clinically safe drug and was recently found to inhibit its main <a href=\"/wiki/Protease\" title=\"Protease\">protease</a>, 3CLpro and showed &gt;200-fold reduction in viral titers in the lungs and nose in mice. However, it is not approved for the treatment of COVID-19 in humans as of August 2021.<sup class=\"reference\" id=\"cite_ref-167\"><a href=\"#cite_note-167\">[167]</a></sup><sup class=\"noprint Inline-Template\" style=\"white-space:nowrap;\">[<i><a href=\"/wiki/Wikipedia:Manual_of_Style/Dates_and_numbers#Chronological_items\" title=\"Wikipedia:Manual of Style/Dates and numbers\"><span title=\"The date of the event predicted near this tag has passed. (May 2022)\">needs update</span></a></i>]</sup>  In December 2021, the <a href=\"/wiki/United_States\" title=\"United States\">United States</a> granted <a class=\"mw-redirect\" href=\"/wiki/Emergency_use_authorization\" title=\"Emergency use authorization\">emergency use authorization</a> to <a href=\"/wiki/Nirmatrelvir/ritonavir\" title=\"Nirmatrelvir/ritonavir\">Nirmatrelvir/ritonavir</a> for the treatment of the virus;<sup class=\"reference\" id=\"cite_ref-fda-factsheet_168-0\"><a href=\"#cite_note-fda-factsheet-168\">[168]</a></sup> the <a href=\"/wiki/European_Union\" title=\"European Union\">European Union</a>, <a href=\"/wiki/United_Kingdom\" title=\"United Kingdom\">United Kingdom</a>, and <a href=\"/wiki/Canada\" title=\"Canada\">Canada</a> followed suit with full authorization soon after.<sup class=\"reference\" id=\"cite_ref-Paxlovid_EPAR_169-0\"><a href=\"#cite_note-Paxlovid_EPAR-169\">[169]</a></sup><sup class=\"reference\" id=\"cite_ref-MHRA_PR_20211231_170-0\"><a href=\"#cite_note-MHRA_PR_20211231-170\">[170]</a></sup><sup class=\"reference\" id=\"cite_ref-Health_Canada_PR_20220117_171-0\"><a href=\"#cite_note-Health_Canada_PR_20220117-171\">[171]</a></sup>  One study found that Nirmatrelvir/ritonavir reduced the risk of hospitalization and death by 88%.<sup class=\"reference\" id=\"cite_ref-FDA_PR_20211222_172-0\"><a href=\"#cite_note-FDA_PR_20211222-172\">[172]</a></sup>\n",
       " </p>,\n",
       " <p><a href=\"/wiki/COVID_Moonshot\" title=\"COVID Moonshot\">COVID Moonshot</a> is an international collaborative <a class=\"mw-redirect\" href=\"/wiki/Open-science\" title=\"Open-science\">open-science</a> project started in March 2020 with the goal of developing an un-<a class=\"mw-redirect\" href=\"/wiki/Patented\" title=\"Patented\">patented</a> <a href=\"/wiki/Oral_medicine\" title=\"Oral medicine\">oral</a> <a href=\"/wiki/Antiviral_drug\" title=\"Antiviral drug\">antiviral drug</a> for treatment of SARS-CoV-2.<sup class=\"reference\" id=\"cite_ref-Whipple_173-0\"><a href=\"#cite_note-Whipple-173\">[173]</a></sup>\n",
       " </p>,\n",
       " <p>Retrospective tests collected within the Chinese surveillance system revealed no clear indication of substantial unrecognized circulation of SARS‑CoV‑2 in Wuhan during the latter part of 2019.<sup class=\"reference\" id=\"cite_ref-WHOconvened_86-4\"><a href=\"#cite_note-WHOconvened-86\">[86]</a></sup>\n",
       " </p>,\n",
       " <p>A meta-analysis from November 2020 estimated the <a href=\"/wiki/Basic_reproduction_number\" title=\"Basic reproduction number\">basic reproduction number</a> (<span class=\"mwe-math-element\"><span class=\"mwe-math-mathml-inline mwe-math-mathml-a11y\" style=\"display: none;\"><math alttext=\"{\\displaystyle R_{0}}\" xmlns=\"http://www.w3.org/1998/Math/MathML\">\n",
       " <semantics>\n",
       " <mrow class=\"MJX-TeXAtom-ORD\">\n",
       " <mstyle displaystyle=\"true\" scriptlevel=\"0\">\n",
       " <msub>\n",
       " <mi>R</mi>\n",
       " <mrow class=\"MJX-TeXAtom-ORD\">\n",
       " <mn>0</mn>\n",
       " </mrow>\n",
       " </msub>\n",
       " </mstyle>\n",
       " </mrow>\n",
       " <annotation encoding=\"application/x-tex\">{\\displaystyle R_{0}}</annotation>\n",
       " </semantics>\n",
       " </math></span><img alt=\"R_{0}\" aria-hidden=\"true\" class=\"mwe-math-fallback-image-inline\" src=\"https://wikimedia.org/api/rest_v1/media/math/render/svg/9b8916196f182fcbaaca54f931176a4a4f5769cc\" style=\"vertical-align: -0.671ex; width:2.818ex; height:2.509ex;\"/></span>) of the virus to be between 2.39 and 3.44.<sup class=\"reference\" id=\"cite_ref-BillahMiahKhan_21-1\"><a href=\"#cite_note-BillahMiahKhan-21\">[21]</a></sup> This means each infection from the virus is expected to result in 2.39 to 3.44 new infections when no members of the community are <a href=\"/wiki/Immunity_(medical)\" title=\"Immunity (medical)\">immune</a> and no <a class=\"mw-redirect\" href=\"/wiki/Infection_control\" title=\"Infection control\">preventive measures</a> are taken. The reproduction number may be higher in densely populated conditions such as those found on <a href=\"/wiki/Cruise_ship\" title=\"Cruise ship\">cruise ships</a>.<sup class=\"reference\" id=\"cite_ref-Rocklov_174-0\"><a href=\"#cite_note-Rocklov-174\">[174]</a></sup> Human behavior affects the R0 value and hence estimates of R0 differ between different countries, cultures, and social norms. For instance, one study found relatively low R0 (~3.5) in Sweden, Belgium and the Netherlands, while Spain and the US had significantly higher R0 values (5.9 to 6.4, respectively).<sup class=\"reference\" id=\"cite_ref-175\"><a href=\"#cite_note-175\">[175]</a></sup>\n",
       " </p>,\n",
       " <p>There have been about 96,000 confirmed cases of infection in mainland China.<sup class=\"reference\" id=\"cite_ref-JHU_ticker_179-0\"><a href=\"#cite_note-JHU_ticker-179\">[179]</a></sup> While the proportion of infections that result in <a href=\"/wiki/COVID-19_pandemic#Cases\" title=\"COVID-19 pandemic\">confirmed cases</a> or progress to diagnosable disease remains unclear,<sup class=\"reference\" id=\"cite_ref-STAT-Severity_180-0\"><a href=\"#cite_note-STAT-Severity-180\">[180]</a></sup> one mathematical model estimated that 75,815 people were infected on 25 January 2020 in Wuhan alone, at a time when the number of confirmed cases worldwide was only 2,015.<sup class=\"reference\" id=\"cite_ref-pmid32014114_181-0\"><a href=\"#cite_note-pmid32014114-181\">[181]</a></sup> Before 24 February 2020, over 95% of all deaths from <a href=\"/wiki/COVID-19\" title=\"COVID-19\">COVID-19</a> worldwide had occurred in <a href=\"/wiki/Hubei\" title=\"Hubei\">Hubei province</a>, where Wuhan is located.<sup class=\"reference\" id=\"cite_ref-GuardianLeap_182-0\"><a href=\"#cite_note-GuardianLeap-182\">[182]</a></sup><sup class=\"reference\" id=\"cite_ref-SunChinaAfrica_183-0\"><a href=\"#cite_note-SunChinaAfrica-183\">[183]</a></sup> As of 3 March 2023, the percentage had decreased to <span data-sort-value=\"6998467706849025799♠\" style=\"display:none\"></span>0.047%.<sup class=\"reference\" id=\"cite_ref-JHU_ticker_179-1\"><a href=\"#cite_note-JHU_ticker-179\">[179]</a></sup>\n",
       " </p>,\n",
       " <p>As of 3 March 2023, there have been 675,816,641 total confirmed cases of SARS‑CoV‑2 infection in the ongoing pandemic.<sup class=\"reference\" id=\"cite_ref-JHU_ticker_179-2\"><a href=\"#cite_note-JHU_ticker-179\">[179]</a></sup> The total number of deaths attributed to the virus is 6,876,102.<sup class=\"reference\" id=\"cite_ref-JHU_ticker_179-3\"><a href=\"#cite_note-JHU_ticker-179\">[179]</a></sup>\n",
       " </p>]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "paragraphs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "6855618e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\nSevere acute respiratory syndrome coronavirus 2 (SARS‑CoV‑2)[2] is a strain of coronavirus that causes COVID-19 (coronavirus disease 2019), the respiratory illness responsible for the ongoing COVID-19 pandemic.[3] The virus previously had a provisional name, 2019 novel coronavirus (2019-nCoV),[4][5][6][7] and has also been called the human coronavirus 2019 (HCoV-19 or hCoV-19).[8][9][10][11]  First identified in the city of Wuhan, Hubei, China, the World Health Organization declared the outbreak a public health emergency of international concern on January 30, 2020, and a pandemic on March 11, 2020.[12][13] SARS‑CoV‑2 is a positive-sense single-stranded RNA virus[14] that is contagious in humans.[15]\\nSARS‑CoV‑2 is a virus of the species severe acute respiratory syndrome–related coronavirus (SARSr-CoV), related to the SARS-CoV-1 virus that caused the 2002–2004 SARS outbreak.[2][16] Despite its close relation to SARS-CoV-1, its closest known relatives, with which it forms a sister group, are the derived SARS viruses BANAL-52 and RaTG13.[17] Available evidence indicates that it is most likely of zoonotic origin and has close genetic similarity to bat coronaviruses, suggesting it emerged from a bat-borne virus.[18] Research is ongoing as to whether SARS‑CoV‑2 came directly from bats or indirectly through any intermediate hosts.[19] The virus shows little genetic diversity, indicating that the spillover event introducing SARS‑CoV‑2 to humans is likely to have occurred in late 2019.[20]\\nEpidemiological studies estimate that, in the December 2019–September 2020 period, each infection resulted in an average of 2.4–3.4 new infections when no members of the community are immune and no preventive measures are taken.[21] However, some subsequent variants have become more infectious.[22] The virus is airborne and primarily spreads between people through close contact and via aerosols and respiratory droplets that are exhaled when talking, breathing, or otherwise exhaling, as well as those produced from coughs and sneezes.[23][24] It enters human cells by binding to angiotensin-converting enzyme 2 (ACE2), a membrane protein that regulates the renin–angiotensin system.[25][26]\\nDuring the initial outbreak in Wuhan, China, various names were used for the virus; some names used by different sources included \"the coronavirus\" or \"Wuhan coronavirus\".[27][28] In January 2020, the World Health Organization (WHO) recommended \"2019 novel coronavirus\" (2019-nCoV)[5][29] as the provisional name for the virus. This was in accordance with WHO\\'s 2015 guidance[30] against using geographical locations, animal species, or groups of people in disease and virus names.[31][32]\\nOn 11 February 2020, the International Committee on Taxonomy of Viruses adopted the official name \"severe acute respiratory syndrome coronavirus 2\" (SARS‑CoV‑2).[33] To avoid confusion with the disease SARS, the WHO sometimes refers to SARS‑CoV‑2 as \"the COVID-19 virus\" in public health communications[34][35] and the name HCoV-19 was included in some research articles.[8][9][10] Referring to COVID-19 as the \"Wuhan virus\" has been described as dangerous by WHO officials, and as xenophobic by University of California at Berkeley Asian American studies lecturer Harvey Dong.[36][37][38]\\nHuman-to-human transmission of SARS‑CoV‑2 was confirmed on 20 January 2020 during the COVID-19 pandemic.[15][39][40][41] Transmission was initially assumed to occur primarily via respiratory droplets from coughs and sneezes within a range of about 1.8 metres (6\\xa0ft).[42][43] Laser light scattering experiments suggest that speaking is an additional mode of transmission[44][45] and a far-reaching[46] one, indoors, with little air flow.[47][48] Other studies have suggested that the virus may be airborne as well, with aerosols potentially being able to transmit the virus.[49][50][51] During human-to-human transmission, between 200 and 800 infectious SARS‑CoV‑2 virions are thought to initiate a new infection.[52][53][54] If confirmed, aerosol transmission has biosafety implications because a major concern associated with the risk of working with emerging viruses in the laboratory is the generation of aerosols from various laboratory activities which are not immediately recognizable and may affect other scientific personnel.[55] Indirect contact via contaminated surfaces is another possible cause of infection.[56] Preliminary research indicates that the virus may remain viable on plastic (polypropylene) and stainless steel (AISI 304) for up to three days, but it does not survive on cardboard for more than one day or on copper for more than four hours.[10] The virus is inactivated by soap, which destabilizes its lipid bilayer.[57][58] Viral RNA has also been found in stool samples and semen from infected individuals.[59][60]\\nThe degree to which the virus is infectious during the incubation period is uncertain, but research has indicated that the pharynx reaches peak viral load approximately four days after infection[61][62] or in the first week of symptoms and declines thereafter.[63] The duration of SARS-CoV-2 RNA shedding is generally between 3 and 46 days after symptom onset.[64]\\nA study by a team of researchers from the University of North Carolina found that the nasal cavity is seemingly the dominant initial site of infection, with subsequent aspiration-mediated virus-seeding into the lungs in SARS‑CoV‑2 pathogenesis.[65] They found that there was an infection gradient from high in proximal towards low in distal pulmonary epithelial cultures, with a focal infection in ciliated cells and type 2 pneumocytes in the airway and alveolar regions respectively.[65]\\nStudies have identified a range of animals—such as cats, ferrets, hamsters, non-human primates, minks, tree shrews, raccoon dogs, fruit bats, and rabbits—that are susceptible and permissive to SARS-CoV-2 infection.[66][67][68] Some institutions have advised that those infected with SARS‑CoV‑2 restrict their contact with animals.[69][70]\\nOn 1\\xa0February 2020, the World Health Organization (WHO) indicated that \"transmission from asymptomatic cases is likely not a major driver of transmission\".[71] One meta-analysis found that 17% of infections are asymptomatic, and asymptomatic individuals were 42% less likely to transmit the virus.[72]\\nHowever, an epidemiological model of the beginning of the outbreak in China suggested that \"pre-symptomatic shedding may be typical among documented infections\" and that subclinical infections may have been the source of a majority of infections.[73] That may explain how out of 217 on board a cruise liner that docked at Montevideo, only 24 of 128 who tested positive for viral RNA showed symptoms.[74] Similarly, a study of ninety-four patients hospitalized in January and February 2020 estimated patients began shedding virus two to three days before symptoms appear and that \"a substantial proportion of transmission probably occurred before first symptoms in the index case\".[53] The authors later published a correction that showed that shedding began earlier than first estimated, four to five days before symptoms appear.[75]\\nThere is uncertainty about reinfection and long-term immunity.[76] It is not known how common reinfection is, but reports have indicated that it is occurring with variable severity.[76]\\nThe first reported case of reinfection was a 33-year-old man from Hong Kong who first tested positive on 26 March 2020, was discharged on 15 April 2020 after two negative tests, and tested positive again on 15 August 2020 (142 days later), which was confirmed by whole-genome sequencing showing that the viral genomes between the episodes belong to different clades.[77] The findings had the implications that herd immunity may not eliminate the virus if reinfection is not an uncommon occurrence and that vaccines may not be able to provide lifelong protection against the virus.[77]\\nAnother case study described a 25-year-old man from Nevada who tested positive for SARS‑CoV‑2 on 18 April 2020 and on 5 June 2020 (separated by two negative tests). Since genomic analyses showed significant genetic differences between the SARS‑CoV‑2 variant sampled on those two dates, the case study authors determined this was a reinfection.[78] The man\\'s second infection was symptomatically more severe than the first infection, but the mechanisms that could account for this are not known.[78]\\n\\nNo natural reservoir for SARS-CoV-2 has been identified.[79] Prior to the emergence of SARS-CoV-2 as a pathogen infecting humans, there had been two previous zoonosis-based coronavirus epidemics, those caused by SARS-CoV-1 and MERS-CoV.[18]\\nThe first known infections from SARS‑CoV‑2 were discovered in Wuhan, China.[80] The original source of viral transmission to humans remains unclear, as does whether the virus became pathogenic before or after the spillover event.[9][20][81] Because many of the early infectees were workers at the Huanan Seafood Market,[82][83] it has been suggested that the virus might have originated from the market.[9][84] However, other research indicates that visitors may have introduced the virus to the market, which then facilitated rapid expansion of the infections.[20][85] A March 2021 WHO-convened report stated that human spillover via an intermediate animal host was the most likely explanation, with direct spillover from bats next most likely. Introduction through the food supply chain and the Huanan Seafood Market was considered another possible, but less likely, explanation.[86] An analysis in November 2021, however, said that the earliest-known case had been misidentified and that the preponderance of early cases linked to the Huanan Market argued for it being the source.[87]\\nFor a virus recently acquired through a cross-species transmission, rapid evolution is expected.[88] The mutation rate estimated from early cases of SARS-CoV-2 was of 6.54×10−4 per site per year.[86] Coronaviruses in general have high genetic plasticity,[89] but SARS-CoV-2\\'s viral evolution is slowed by the RNA proofreading capability of its replication machinery.[90] For comparison, the viral mutation rate in vivo of SARS-CoV-2 has been found to be lower than that of influenza.[91]\\nResearch into the natural reservoir of the virus that caused the 2002–2004 SARS outbreak has resulted in the discovery of many SARS-like bat coronaviruses, most originating in horseshoe bats. The closest match by far, published on Nature (journal) in February 2022, were viruses BANAL-52 (96.8% resemblance to SARS‑CoV‑2), BANAL-103 and BANAL-236, collected in three different species of bats in Feuang, Laos.[92][93][94] An earlier source published in February 2020 identified the virus RaTG13, collected in bats in Mojiang, Yunnan, China to be the closest to SARS‑CoV‑2, with 96.1% resemblance.[80][95] None of the above are its direct ancestor.[96]\\nBats are considered the most likely natural reservoir of SARS‑CoV‑2.[86][97] Differences between the bat coronavirus and SARS‑CoV‑2 suggest that humans may have been infected via an intermediate host;[84] although the source of introduction into humans remains unknown.[98][79]\\nAlthough the role of pangolins as an intermediate host was initially posited (a study published in July 2020 suggested that pangolins are an intermediate host of SARS‑CoV‑2-like coronaviruses[99][100]), subsequent studies have not substantiated their contribution to the spillover.[86] Evidence against this hypothesis includes the fact that pangolin virus samples are too distant to SARS-CoV-2: isolates obtained from pangolins seized in Guangdong were only 92% identical in sequence to the SARS‑CoV‑2 genome (matches above 90 percent may sound high, but in genomic terms it is a wide evolutionary gap[101]). In addition, despite similarities in a few critical amino acids,[102] pangolin virus samples exhibit poor binding to the human ACE2 receptor.[103]\\nSARS‑CoV‑2 belongs to the broad family of viruses known as coronaviruses.[28] It is a positive-sense single-stranded RNA (+ssRNA) virus, with a single linear RNA segment. Coronaviruses infect humans, other mammals, including livestock and companion animals, and avian species.[104] Human coronaviruses are capable of causing illnesses ranging from the common cold to more severe diseases such as Middle East respiratory syndrome (MERS, fatality rate ~34%). SARS-CoV-2 is the seventh known coronavirus to infect people, after 229E, NL63, OC43, HKU1, MERS-CoV, and the original SARS-CoV.[105]\\nLike the SARS-related coronavirus implicated in the 2003 SARS outbreak, SARS‑CoV‑2 is a member of the subgenus Sarbecovirus (beta-CoV lineage B).[106][107] Coronaviruses undergo frequent recombination.[108] The mechanism of recombination in unsegmented RNA viruses such as SARS-CoV-2 is generally by copy-choice replication, in which gene material switches from one RNA template molecule to another during replication.[109] The SARS-CoV-2 RNA sequence is approximately 30,000 bases in length,[110] relatively long for a coronavirus—which in turn carry the largest genomes among all RNA families.[111] Its genome consists nearly entirely of protein-coding sequences, a trait shared with other coronaviruses.[108]\\nA distinguishing feature of SARS‑CoV‑2 is its incorporation of a polybasic site cleaved by furin,[102][112] which appears to be an important element enhancing its virulence.[113] It was suggested that the acquisition of the furin-cleavage site in the SARS-CoV-2 S protein was essential for zoonotic transfer to humans.[114] The furin protease recognizes the canonical peptide sequence RX[R/K] R↓X where the cleavage site is indicated by a down arrow and X is any amino acid.[115][116] In SARS-CoV-2 the recognition site is formed by the incorporated 12 codon nucleotide sequence CCT CGG CGG GCA which corresponds to the amino acid sequence P RR A.[117] This sequence is upstream of an arginine and serine which forms the S1/S2 cleavage site (P RR A R↓S) of the spike protein.[118] Although such sites are a common naturally-occurring feature of other viruses within the Subfamily Orthocoronavirinae,[117] it appears in few other viruses from the Beta-CoV genus,[119] and it is unique among members of its subgenus for such a site.[102] The furin cleavage site PRRAR↓ is highly similar to that of the feline coronavirus, an alphacoronavirus 1 strain.[120]\\nViral genetic sequence data can provide critical information about whether viruses separated by time and space are likely to be epidemiologically linked.[121] With a sufficient number of sequenced genomes, it is possible to reconstruct a phylogenetic tree of the mutation history of a family of viruses. By 12 January 2020, five genomes of SARS‑CoV‑2 had been isolated from Wuhan and reported by the Chinese Center for Disease Control and Prevention (CCDC) and other institutions;[110][122] the number of genomes increased to 42 by 30 January 2020.[123] A phylogenetic analysis of those samples showed they were \"highly related with at most seven mutations relative to a common ancestor\", implying that the first human infection occurred in November or December 2019.[123] Examination of the topology of the phylogenetic tree at the start of the pandemic also found high similarities between human isolates.[124]  As of 21\\xa0August\\xa02021,[update] 3,422 SARS‑CoV‑2 genomes, belonging to 19 strains, sampled on all continents except Antarctica were publicly available.[125]\\nOn 11 February 2020, the International Committee on Taxonomy of Viruses announced that according to existing rules that compute hierarchical relationships among coronaviruses based on five conserved sequences of nucleic acids, the differences between what was then called 2019-nCoV and the virus from the 2003 SARS outbreak were insufficient to make them separate viral species. Therefore, they identified 2019-nCoV as a virus of Severe acute respiratory syndrome–related coronavirus.[126]\\nIn July 2020, scientists reported that a more infectious SARS‑CoV‑2 variant with spike protein variant G614 has replaced D614 as the dominant form in the pandemic.[127][128]\\nCoronavirus genomes and subgenomes encode six open reading frames (ORFs).[129] In October 2020, researchers discovered a possible overlapping gene named ORF3d, in the SARS‑CoV‑2 genome. It is unknown if the protein produced by ORF3d has any function, but it provokes a strong immune response. ORF3d has been identified before, in a variant of coronavirus that infects pangolins.[130][131]\\nA phylogenetic tree based on whole-genome sequences of SARS-CoV-2 and related coronaviruses is:[132][133]\\n(Bat) Rc-o319, 81% to SARS-CoV-2, Rhinolophus cornutus, Iwate, Japan[134]\\nBat SL-ZXC21, 88% to SARS-CoV-2, Rhinolophus pusillus, Zhoushan, Zhejiang[135]\\nBat SL-ZC45, 88% to SARS-CoV-2, Rhinolophus pusillus, Zhoushan, Zhejiang[135]\\nPangolin SARSr-CoV-GX, 85.3% to SARS-CoV-2, Manis javanica, smuggled from Southeast Asia[136]\\nPangolin SARSr-CoV-GD, 90.1% to SARS-CoV-2, Manis javanica, smuggled from Southeast Asia[137]\\nBat RshSTT182, 92.6% to SARS-CoV-2, Rhinolophus shameli, Steung Treng, Cambodia[138]\\nBat RshSTT200, 92.6% to SARS-CoV-2, Rhinolophus shameli, Steung Treng, Cambodia[138]\\n(Bat) RacCS203, 91.5% to SARS-CoV-2, Rhinolophus acuminatus, Chachoengsao, Thailand[133]\\n(Bat) RmYN02, 93.3% to SARS-CoV-2, Rhinolophus malayanus, Mengla, Yunnan[139]\\n(Bat) RpYN06, 94.4% to SARS-CoV-2, Rhinolophus pusillus, Xishuangbanna, Yunnan[132]\\n(Bat) RaTG13, 96.1% to SARS-CoV-2, Rhinolophus affinis, Mojiang, Yunnan[140]\\n(Bat) BANAL-52, 96.8% to SARS-CoV-2, Rhinolophus malayanus, Vientiane, Laos[141]\\nSARS-CoV-2\\nSARS-CoV-1, 79% to SARS-CoV-2\\n\\nThere are many thousands of variants of SARS-CoV-2, which can be grouped into the much larger clades.[142] Several different clade nomenclatures have been proposed. Nextstrain divides the variants into five clades (19A, 19B, 20A, 20B, and 20C), while GISAID divides them into seven (L, O, V, S, G, GH, and GR).[143]\\nSeveral notable variants of SARS-CoV-2 emerged in late 2020. The World Health Organization has currently declared five variants of concern, which are as follows:[144]\\nOther notable variants include 6 other WHO-designated variants under investigation and Cluster 5, which emerged among mink in Denmark and resulted in a mink euthanasia campaign rendering it virtually extinct.[145]\\nEach SARS-CoV-2 virion is 60–140 nanometres (2.4×10−6–5.5×10−6\\xa0in) in diameter;[105][83] its mass within the global human populace has been estimated as being between 0.1 and 10 kilograms.[146] Like other coronaviruses, SARS-CoV-2 has four structural proteins, known as the S (spike), E (envelope), M (membrane), and N (nucleocapsid) proteins; the N protein holds the RNA genome, and the S, E, and M proteins together create the viral envelope.[147] Coronavirus S proteins are glycoproteins and also type I membrane proteins (membranes containing a single transmembrane domain oriented on the extracellular side).[114] They are divided into two functional parts (S1 and S2).[104] In SARS-CoV-2, the spike protein, which has been imaged at the atomic level using cryogenic electron microscopy,[148][149] is the protein responsible for allowing the virus to attach to and fuse with the membrane of a host cell;[147] specifically, its S1 subunit catalyzes attachment, the S2 subunit fusion.[150]\\nAs of early 2022, about 7 million SARS-CoV-2 genomes had been sequenced and deposited into public databases and another 800,000 or so were added each month.[151]\\nSARS-CoV-2 has a linear, positive-sense, single-stranded RNA genome about 30,000 bases long.[104] Its genome has a bias against cytosine (C) and guanine (G) nucleotides, like other coronaviruses.[152] The genome has the highest composition of U (32.2%), followed by A (29.9%), and a similar composition of G (19.6%) and C (18.3%).[153] The nucleotide bias arises from the mutation of guanines and cytosines to adenosines and uracils, respectively.[154] The mutation of CG dinucleotides is thought to arise to avoid the zinc finger antiviral protein related defense mechanism of cells,[155] and to lower the energy to unbind the genome during replication and translation (adenosine and uracil base pair via two hydrogen bonds, cytosine and guanine via three).[154] The depletion of CG dinucleotides in its genome has led the virus to have a noticeable codon usage bias. For instance, arginine\\'s six different codons have a relative synonymous codon usage of AGA (2.67), CGU (1.46), AGG (.81), CGC (.58), CGA (.29), and CGG (.19).[153] A similar codon usage bias trend is seen in other SARS–related coronaviruses.[156]\\nVirus infections start when viral particles bind to host surface cellular receptors.[157] Protein modeling experiments on the spike protein of the virus soon suggested that SARS‑CoV‑2 has sufficient affinity to the receptor angiotensin converting enzyme 2 (ACE2) on human cells to use them as a mechanism of cell entry.[158] By 22 January 2020, a group in China working with the full virus genome and a group in the United States using reverse genetics methods independently and experimentally demonstrated that ACE2 could act as the receptor for SARS‑CoV‑2.[80][159][160][161] Studies have shown that SARS‑CoV‑2 has a higher affinity to human ACE2 than the original SARS virus.[148][162] SARS‑CoV‑2 may also use basigin to assist in cell entry.[163]\\nInitial spike protein priming by transmembrane protease, serine 2 (TMPRSS2) is essential for entry of SARS‑CoV‑2.[25] The host protein neuropilin 1 (NRP1) may aid the virus in host cell entry using ACE2.[164] After a SARS‑CoV‑2 virion attaches to a target cell, the cell\\'s TMPRSS2 cuts open the spike protein of the virus, exposing a fusion peptide in the S2 subunit, and the host receptor ACE2.[150] After fusion, an endosome forms around the virion, separating it from the rest of the host cell. The virion escapes when the pH of the endosome drops or when cathepsin, a host cysteine protease, cleaves it.[150] The virion then releases RNA into the cell and forces the cell to produce and disseminate copies of the virus, which infect more cells.[165]\\nSARS‑CoV‑2 produces at least three virulence factors that promote shedding of new virions from host cells and inhibit immune response.[147] Whether they include downregulation of ACE2, as seen in similar coronaviruses, remains under investigation (as of May 2020).[166]\\nVery few drugs are known to effectively inhibit SARS‑CoV‑2. Masitinib is a clinically safe drug and was recently found to inhibit its main protease, 3CLpro and showed >200-fold reduction in viral titers in the lungs and nose in mice. However, it is not approved for the treatment of COVID-19 in humans as of August 2021.[167][needs update]  In December 2021, the United States granted emergency use authorization to Nirmatrelvir/ritonavir for the treatment of the virus;[168] the European Union, United Kingdom, and Canada followed suit with full authorization soon after.[169][170][171]  One study found that Nirmatrelvir/ritonavir reduced the risk of hospitalization and death by 88%.[172]\\nCOVID Moonshot is an international collaborative open-science project started in March 2020 with the goal of developing an un-patented oral antiviral drug for treatment of SARS-CoV-2.[173]\\nRetrospective tests collected within the Chinese surveillance system revealed no clear indication of substantial unrecognized circulation of SARS‑CoV‑2 in Wuhan during the latter part of 2019.[86]\\nA meta-analysis from November 2020 estimated the basic reproduction number (\\n\\n\\n\\n\\nR\\n\\n0\\n\\n\\n\\n\\n{\\\\displaystyle R_{0}}\\n\\n) of the virus to be between 2.39 and 3.44.[21] This means each infection from the virus is expected to result in 2.39 to 3.44 new infections when no members of the community are immune and no preventive measures are taken. The reproduction number may be higher in densely populated conditions such as those found on cruise ships.[174] Human behavior affects the R0 value and hence estimates of R0 differ between different countries, cultures, and social norms. For instance, one study found relatively low R0 (~3.5) in Sweden, Belgium and the Netherlands, while Spain and the US had significantly higher R0 values (5.9 to 6.4, respectively).[175]\\nThere have been about 96,000 confirmed cases of infection in mainland China.[179] While the proportion of infections that result in confirmed cases or progress to diagnosable disease remains unclear,[180] one mathematical model estimated that 75,815 people were infected on 25 January 2020 in Wuhan alone, at a time when the number of confirmed cases worldwide was only 2,015.[181] Before 24 February 2020, over 95% of all deaths from COVID-19 worldwide had occurred in Hubei province, where Wuhan is located.[182][183] As of 3 March 2023, the percentage had decreased to 0.047%.[179]\\nAs of 3 March 2023, there have been 675,816,641 total confirmed cases of SARS‑CoV‑2 infection in the ongoing pandemic.[179] The total number of deaths attributed to the virus is 6,876,102.[179]\\n'"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Clean\n",
    "for p in paragraphs:\n",
    "    article_text += p.text\n",
    "    \n",
    "article_text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "1887f1dc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "' Severe acute respiratory syndrome coronavirus 2 (SARS‑CoV‑2) is a strain of coronavirus that causes COVID-19 (coronavirus disease 2019), the respiratory illness responsible for the ongoing COVID-19 pandemic. The virus previously had a provisional name, 2019 novel coronavirus (2019-nCoV), and has also been called the human coronavirus 2019 (HCoV-19 or hCoV-19). First identified in the city of Wuhan, Hubei, China, the World Health Organization declared the outbreak a public health emergency of international concern on January 30, 2020, and a pandemic on March 11, 2020. SARS‑CoV‑2 is a positive-sense single-stranded RNA virus that is contagious in humans. SARS‑CoV‑2 is a virus of the species severe acute respiratory syndrome–related coronavirus (SARSr-CoV), related to the SARS-CoV-1 virus that caused the 2002–2004 SARS outbreak. Despite its close relation to SARS-CoV-1, its closest known relatives, with which it forms a sister group, are the derived SARS viruses BANAL-52 and RaTG13. Available evidence indicates that it is most likely of zoonotic origin and has close genetic similarity to bat coronaviruses, suggesting it emerged from a bat-borne virus. Research is ongoing as to whether SARS‑CoV‑2 came directly from bats or indirectly through any intermediate hosts. The virus shows little genetic diversity, indicating that the spillover event introducing SARS‑CoV‑2 to humans is likely to have occurred in late 2019. Epidemiological studies estimate that, in the December 2019–September 2020 period, each infection resulted in an average of 2.4–3.4 new infections when no members of the community are immune and no preventive measures are taken. However, some subsequent variants have become more infectious. The virus is airborne and primarily spreads between people through close contact and via aerosols and respiratory droplets that are exhaled when talking, breathing, or otherwise exhaling, as well as those produced from coughs and sneezes. It enters human cells by binding to angiotensin-converting enzyme 2 (ACE2), a membrane protein that regulates the renin–angiotensin system. During the initial outbreak in Wuhan, China, various names were used for the virus; some names used by different sources included \"the coronavirus\" or \"Wuhan coronavirus\". In January 2020, the World Health Organization (WHO) recommended \"2019 novel coronavirus\" (2019-nCoV) as the provisional name for the virus. This was in accordance with WHO\\'s 2015 guidance against using geographical locations, animal species, or groups of people in disease and virus names. On 11 February 2020, the International Committee on Taxonomy of Viruses adopted the official name \"severe acute respiratory syndrome coronavirus 2\" (SARS‑CoV‑2). To avoid confusion with the disease SARS, the WHO sometimes refers to SARS‑CoV‑2 as \"the COVID-19 virus\" in public health communications and the name HCoV-19 was included in some research articles. Referring to COVID-19 as the \"Wuhan virus\" has been described as dangerous by WHO officials, and as xenophobic by University of California at Berkeley Asian American studies lecturer Harvey Dong. Human-to-human transmission of SARS‑CoV‑2 was confirmed on 20 January 2020 during the COVID-19 pandemic. Transmission was initially assumed to occur primarily via respiratory droplets from coughs and sneezes within a range of about 1.8 metres (6 ft). Laser light scattering experiments suggest that speaking is an additional mode of transmission and a far-reaching one, indoors, with little air flow. Other studies have suggested that the virus may be airborne as well, with aerosols potentially being able to transmit the virus. During human-to-human transmission, between 200 and 800 infectious SARS‑CoV‑2 virions are thought to initiate a new infection. If confirmed, aerosol transmission has biosafety implications because a major concern associated with the risk of working with emerging viruses in the laboratory is the generation of aerosols from various laboratory activities which are not immediately recognizable and may affect other scientific personnel. Indirect contact via contaminated surfaces is another possible cause of infection. Preliminary research indicates that the virus may remain viable on plastic (polypropylene) and stainless steel (AISI 304) for up to three days, but it does not survive on cardboard for more than one day or on copper for more than four hours. The virus is inactivated by soap, which destabilizes its lipid bilayer. Viral RNA has also been found in stool samples and semen from infected individuals. The degree to which the virus is infectious during the incubation period is uncertain, but research has indicated that the pharynx reaches peak viral load approximately four days after infection or in the first week of symptoms and declines thereafter. The duration of SARS-CoV-2 RNA shedding is generally between 3 and 46 days after symptom onset. A study by a team of researchers from the University of North Carolina found that the nasal cavity is seemingly the dominant initial site of infection, with subsequent aspiration-mediated virus-seeding into the lungs in SARS‑CoV‑2 pathogenesis. They found that there was an infection gradient from high in proximal towards low in distal pulmonary epithelial cultures, with a focal infection in ciliated cells and type 2 pneumocytes in the airway and alveolar regions respectively. Studies have identified a range of animals—such as cats, ferrets, hamsters, non-human primates, minks, tree shrews, raccoon dogs, fruit bats, and rabbits—that are susceptible and permissive to SARS-CoV-2 infection. Some institutions have advised that those infected with SARS‑CoV‑2 restrict their contact with animals. On 1 February 2020, the World Health Organization (WHO) indicated that \"transmission from asymptomatic cases is likely not a major driver of transmission\". One meta-analysis found that 17% of infections are asymptomatic, and asymptomatic individuals were 42% less likely to transmit the virus. However, an epidemiological model of the beginning of the outbreak in China suggested that \"pre-symptomatic shedding may be typical among documented infections\" and that subclinical infections may have been the source of a majority of infections. That may explain how out of 217 on board a cruise liner that docked at Montevideo, only 24 of 128 who tested positive for viral RNA showed symptoms. Similarly, a study of ninety-four patients hospitalized in January and February 2020 estimated patients began shedding virus two to three days before symptoms appear and that \"a substantial proportion of transmission probably occurred before first symptoms in the index case\". The authors later published a correction that showed that shedding began earlier than first estimated, four to five days before symptoms appear. There is uncertainty about reinfection and long-term immunity. It is not known how common reinfection is, but reports have indicated that it is occurring with variable severity. The first reported case of reinfection was a 33-year-old man from Hong Kong who first tested positive on 26 March 2020, was discharged on 15 April 2020 after two negative tests, and tested positive again on 15 August 2020 (142 days later), which was confirmed by whole-genome sequencing showing that the viral genomes between the episodes belong to different clades. The findings had the implications that herd immunity may not eliminate the virus if reinfection is not an uncommon occurrence and that vaccines may not be able to provide lifelong protection against the virus. Another case study described a 25-year-old man from Nevada who tested positive for SARS‑CoV‑2 on 18 April 2020 and on 5 June 2020 (separated by two negative tests). Since genomic analyses showed significant genetic differences between the SARS‑CoV‑2 variant sampled on those two dates, the case study authors determined this was a reinfection. The man\\'s second infection was symptomatically more severe than the first infection, but the mechanisms that could account for this are not known. No natural reservoir for SARS-CoV-2 has been identified. Prior to the emergence of SARS-CoV-2 as a pathogen infecting humans, there had been two previous zoonosis-based coronavirus epidemics, those caused by SARS-CoV-1 and MERS-CoV. The first known infections from SARS‑CoV‑2 were discovered in Wuhan, China. The original source of viral transmission to humans remains unclear, as does whether the virus became pathogenic before or after the spillover event. Because many of the early infectees were workers at the Huanan Seafood Market, it has been suggested that the virus might have originated from the market. However, other research indicates that visitors may have introduced the virus to the market, which then facilitated rapid expansion of the infections. A March 2021 WHO-convened report stated that human spillover via an intermediate animal host was the most likely explanation, with direct spillover from bats next most likely. Introduction through the food supply chain and the Huanan Seafood Market was considered another possible, but less likely, explanation. An analysis in November 2021, however, said that the earliest-known case had been misidentified and that the preponderance of early cases linked to the Huanan Market argued for it being the source. For a virus recently acquired through a cross-species transmission, rapid evolution is expected. The mutation rate estimated from early cases of SARS-CoV-2 was of 6.54×10−4 per site per year. Coronaviruses in general have high genetic plasticity, but SARS-CoV-2\\'s viral evolution is slowed by the RNA proofreading capability of its replication machinery. For comparison, the viral mutation rate in vivo of SARS-CoV-2 has been found to be lower than that of influenza. Research into the natural reservoir of the virus that caused the 2002–2004 SARS outbreak has resulted in the discovery of many SARS-like bat coronaviruses, most originating in horseshoe bats. The closest match by far, published on Nature (journal) in February 2022, were viruses BANAL-52 (96.8% resemblance to SARS‑CoV‑2), BANAL-103 and BANAL-236, collected in three different species of bats in Feuang, Laos. An earlier source published in February 2020 identified the virus RaTG13, collected in bats in Mojiang, Yunnan, China to be the closest to SARS‑CoV‑2, with 96.1% resemblance. None of the above are its direct ancestor. Bats are considered the most likely natural reservoir of SARS‑CoV‑2. Differences between the bat coronavirus and SARS‑CoV‑2 suggest that humans may have been infected via an intermediate host; although the source of introduction into humans remains unknown. Although the role of pangolins as an intermediate host was initially posited (a study published in July 2020 suggested that pangolins are an intermediate host of SARS‑CoV‑2-like coronaviruses ), subsequent studies have not substantiated their contribution to the spillover. Evidence against this hypothesis includes the fact that pangolin virus samples are too distant to SARS-CoV-2: isolates obtained from pangolins seized in Guangdong were only 92% identical in sequence to the SARS‑CoV‑2 genome (matches above 90 percent may sound high, but in genomic terms it is a wide evolutionary gap ). In addition, despite similarities in a few critical amino acids, pangolin virus samples exhibit poor binding to the human ACE2 receptor. SARS‑CoV‑2 belongs to the broad family of viruses known as coronaviruses. It is a positive-sense single-stranded RNA (+ssRNA) virus, with a single linear RNA segment. Coronaviruses infect humans, other mammals, including livestock and companion animals, and avian species. Human coronaviruses are capable of causing illnesses ranging from the common cold to more severe diseases such as Middle East respiratory syndrome (MERS, fatality rate ~34%). SARS-CoV-2 is the seventh known coronavirus to infect people, after 229E, NL63, OC43, HKU1, MERS-CoV, and the original SARS-CoV. Like the SARS-related coronavirus implicated in the 2003 SARS outbreak, SARS‑CoV‑2 is a member of the subgenus Sarbecovirus (beta-CoV lineage B). Coronaviruses undergo frequent recombination. The mechanism of recombination in unsegmented RNA viruses such as SARS-CoV-2 is generally by copy-choice replication, in which gene material switches from one RNA template molecule to another during replication. The SARS-CoV-2 RNA sequence is approximately 30,000 bases in length, relatively long for a coronavirus—which in turn carry the largest genomes among all RNA families. Its genome consists nearly entirely of protein-coding sequences, a trait shared with other coronaviruses. A distinguishing feature of SARS‑CoV‑2 is its incorporation of a polybasic site cleaved by furin, which appears to be an important element enhancing its virulence. It was suggested that the acquisition of the furin-cleavage site in the SARS-CoV-2 S protein was essential for zoonotic transfer to humans. The furin protease recognizes the canonical peptide sequence RX[R/K] R↓X where the cleavage site is indicated by a down arrow and X is any amino acid. In SARS-CoV-2 the recognition site is formed by the incorporated 12 codon nucleotide sequence CCT CGG CGG GCA which corresponds to the amino acid sequence P RR A. This sequence is upstream of an arginine and serine which forms the S1/S2 cleavage site (P RR A R↓S) of the spike protein. Although such sites are a common naturally-occurring feature of other viruses within the Subfamily Orthocoronavirinae, it appears in few other viruses from the Beta-CoV genus, and it is unique among members of its subgenus for such a site. The furin cleavage site PRRAR↓ is highly similar to that of the feline coronavirus, an alphacoronavirus 1 strain. Viral genetic sequence data can provide critical information about whether viruses separated by time and space are likely to be epidemiologically linked. With a sufficient number of sequenced genomes, it is possible to reconstruct a phylogenetic tree of the mutation history of a family of viruses. By 12 January 2020, five genomes of SARS‑CoV‑2 had been isolated from Wuhan and reported by the Chinese Center for Disease Control and Prevention (CCDC) and other institutions; the number of genomes increased to 42 by 30 January 2020. A phylogenetic analysis of those samples showed they were \"highly related with at most seven mutations relative to a common ancestor\", implying that the first human infection occurred in November or December 2019. Examination of the topology of the phylogenetic tree at the start of the pandemic also found high similarities between human isolates. As of 21 August 2021,[update] 3,422 SARS‑CoV‑2 genomes, belonging to 19 strains, sampled on all continents except Antarctica were publicly available. On 11 February 2020, the International Committee on Taxonomy of Viruses announced that according to existing rules that compute hierarchical relationships among coronaviruses based on five conserved sequences of nucleic acids, the differences between what was then called 2019-nCoV and the virus from the 2003 SARS outbreak were insufficient to make them separate viral species. Therefore, they identified 2019-nCoV as a virus of Severe acute respiratory syndrome–related coronavirus. In July 2020, scientists reported that a more infectious SARS‑CoV‑2 variant with spike protein variant G614 has replaced D614 as the dominant form in the pandemic. Coronavirus genomes and subgenomes encode six open reading frames (ORFs). In October 2020, researchers discovered a possible overlapping gene named ORF3d, in the SARS‑CoV‑2 genome. It is unknown if the protein produced by ORF3d has any function, but it provokes a strong immune response. ORF3d has been identified before, in a variant of coronavirus that infects pangolins. A phylogenetic tree based on whole-genome sequences of SARS-CoV-2 and related coronaviruses is: (Bat) Rc-o319, 81% to SARS-CoV-2, Rhinolophus cornutus, Iwate, Japan Bat SL-ZXC21, 88% to SARS-CoV-2, Rhinolophus pusillus, Zhoushan, Zhejiang Bat SL-ZC45, 88% to SARS-CoV-2, Rhinolophus pusillus, Zhoushan, Zhejiang Pangolin SARSr-CoV-GX, 85.3% to SARS-CoV-2, Manis javanica, smuggled from Southeast Asia Pangolin SARSr-CoV-GD, 90.1% to SARS-CoV-2, Manis javanica, smuggled from Southeast Asia Bat RshSTT182, 92.6% to SARS-CoV-2, Rhinolophus shameli, Steung Treng, Cambodia Bat RshSTT200, 92.6% to SARS-CoV-2, Rhinolophus shameli, Steung Treng, Cambodia (Bat) RacCS203, 91.5% to SARS-CoV-2, Rhinolophus acuminatus, Chachoengsao, Thailand (Bat) RmYN02, 93.3% to SARS-CoV-2, Rhinolophus malayanus, Mengla, Yunnan (Bat) RpYN06, 94.4% to SARS-CoV-2, Rhinolophus pusillus, Xishuangbanna, Yunnan (Bat) RaTG13, 96.1% to SARS-CoV-2, Rhinolophus affinis, Mojiang, Yunnan (Bat) BANAL-52, 96.8% to SARS-CoV-2, Rhinolophus malayanus, Vientiane, Laos SARS-CoV-2 SARS-CoV-1, 79% to SARS-CoV-2 There are many thousands of variants of SARS-CoV-2, which can be grouped into the much larger clades. Several different clade nomenclatures have been proposed. Nextstrain divides the variants into five clades (19A, 19B, 20A, 20B, and 20C), while GISAID divides them into seven (L, O, V, S, G, GH, and GR). Several notable variants of SARS-CoV-2 emerged in late 2020. The World Health Organization has currently declared five variants of concern, which are as follows: Other notable variants include 6 other WHO-designated variants under investigation and Cluster 5, which emerged among mink in Denmark and resulted in a mink euthanasia campaign rendering it virtually extinct. Each SARS-CoV-2 virion is 60–140 nanometres (2.4×10−6–5.5×10−6 in) in diameter; its mass within the global human populace has been estimated as being between 0.1 and 10 kilograms. Like other coronaviruses, SARS-CoV-2 has four structural proteins, known as the S (spike), E (envelope), M (membrane), and N (nucleocapsid) proteins; the N protein holds the RNA genome, and the S, E, and M proteins together create the viral envelope. Coronavirus S proteins are glycoproteins and also type I membrane proteins (membranes containing a single transmembrane domain oriented on the extracellular side). They are divided into two functional parts (S1 and S2). In SARS-CoV-2, the spike protein, which has been imaged at the atomic level using cryogenic electron microscopy, is the protein responsible for allowing the virus to attach to and fuse with the membrane of a host cell; specifically, its S1 subunit catalyzes attachment, the S2 subunit fusion. As of early 2022, about 7 million SARS-CoV-2 genomes had been sequenced and deposited into public databases and another 800,000 or so were added each month. SARS-CoV-2 has a linear, positive-sense, single-stranded RNA genome about 30,000 bases long. Its genome has a bias against cytosine (C) and guanine (G) nucleotides, like other coronaviruses. The genome has the highest composition of U (32.2%), followed by A (29.9%), and a similar composition of G (19.6%) and C (18.3%). The nucleotide bias arises from the mutation of guanines and cytosines to adenosines and uracils, respectively. The mutation of CG dinucleotides is thought to arise to avoid the zinc finger antiviral protein related defense mechanism of cells, and to lower the energy to unbind the genome during replication and translation (adenosine and uracil base pair via two hydrogen bonds, cytosine and guanine via three). The depletion of CG dinucleotides in its genome has led the virus to have a noticeable codon usage bias. For instance, arginine\\'s six different codons have a relative synonymous codon usage of AGA (2.67), CGU (1.46), AGG (.81), CGC (.58), CGA (.29), and CGG (.19). A similar codon usage bias trend is seen in other SARS–related coronaviruses. Virus infections start when viral particles bind to host surface cellular receptors. Protein modeling experiments on the spike protein of the virus soon suggested that SARS‑CoV‑2 has sufficient affinity to the receptor angiotensin converting enzyme 2 (ACE2) on human cells to use them as a mechanism of cell entry. By 22 January 2020, a group in China working with the full virus genome and a group in the United States using reverse genetics methods independently and experimentally demonstrated that ACE2 could act as the receptor for SARS‑CoV‑2. Studies have shown that SARS‑CoV‑2 has a higher affinity to human ACE2 than the original SARS virus. SARS‑CoV‑2 may also use basigin to assist in cell entry. Initial spike protein priming by transmembrane protease, serine 2 (TMPRSS2) is essential for entry of SARS‑CoV‑2. The host protein neuropilin 1 (NRP1) may aid the virus in host cell entry using ACE2. After a SARS‑CoV‑2 virion attaches to a target cell, the cell\\'s TMPRSS2 cuts open the spike protein of the virus, exposing a fusion peptide in the S2 subunit, and the host receptor ACE2. After fusion, an endosome forms around the virion, separating it from the rest of the host cell. The virion escapes when the pH of the endosome drops or when cathepsin, a host cysteine protease, cleaves it. The virion then releases RNA into the cell and forces the cell to produce and disseminate copies of the virus, which infect more cells. SARS‑CoV‑2 produces at least three virulence factors that promote shedding of new virions from host cells and inhibit immune response. Whether they include downregulation of ACE2, as seen in similar coronaviruses, remains under investigation (as of May 2020). Very few drugs are known to effectively inhibit SARS‑CoV‑2. Masitinib is a clinically safe drug and was recently found to inhibit its main protease, 3CLpro and showed >200-fold reduction in viral titers in the lungs and nose in mice. However, it is not approved for the treatment of COVID-19 in humans as of August 2021. [needs update] In December 2021, the United States granted emergency use authorization to Nirmatrelvir/ritonavir for the treatment of the virus; the European Union, United Kingdom, and Canada followed suit with full authorization soon after. One study found that Nirmatrelvir/ritonavir reduced the risk of hospitalization and death by 88%. COVID Moonshot is an international collaborative open-science project started in March 2020 with the goal of developing an un-patented oral antiviral drug for treatment of SARS-CoV-2. Retrospective tests collected within the Chinese surveillance system revealed no clear indication of substantial unrecognized circulation of SARS‑CoV‑2 in Wuhan during the latter part of 2019. A meta-analysis from November 2020 estimated the basic reproduction number ( R 0 {\\\\displaystyle R_{0}} ) of the virus to be between 2.39 and 3.44. This means each infection from the virus is expected to result in 2.39 to 3.44 new infections when no members of the community are immune and no preventive measures are taken. The reproduction number may be higher in densely populated conditions such as those found on cruise ships. Human behavior affects the R0 value and hence estimates of R0 differ between different countries, cultures, and social norms. For instance, one study found relatively low R0 (~3.5) in Sweden, Belgium and the Netherlands, while Spain and the US had significantly higher R0 values (5.9 to 6.4, respectively). There have been about 96,000 confirmed cases of infection in mainland China. While the proportion of infections that result in confirmed cases or progress to diagnosable disease remains unclear, one mathematical model estimated that 75,815 people were infected on 25 January 2020 in Wuhan alone, at a time when the number of confirmed cases worldwide was only 2,015. Before 24 February 2020, over 95% of all deaths from COVID-19 worldwide had occurred in Hubei province, where Wuhan is located. As of 3 March 2023, the percentage had decreased to 0.047%. As of 3 March 2023, there have been 675,816,641 total confirmed cases of SARS‑CoV‑2 infection in the ongoing pandemic. The total number of deaths attributed to the virus is 6,876,102. '"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Removing Square Brackets and Extra Spaces\n",
    "article_text = re.sub(r'\\[[0-9]*\\]', ' ', article_text)\n",
    "article_text = re.sub(r'\\s+', ' ', article_text)\n",
    "\n",
    "article_text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "9ab5610e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "' Severe acute respiratory syndrome coronavirus SARS CoV is a strain of coronavirus that causes COVID coronavirus disease the respiratory illness responsible for the ongoing COVID pandemic The virus previously had a provisional name novel coronavirus nCoV and has also been called the human coronavirus HCoV or hCoV First identified in the city of Wuhan Hubei China the World Health Organization declared the outbreak a public health emergency of international concern on January and a pandemic on March SARS CoV is a positive sense single stranded RNA virus that is contagious in humans SARS CoV is a virus of the species severe acute respiratory syndrome related coronavirus SARSr CoV related to the SARS CoV virus that caused the SARS outbreak Despite its close relation to SARS CoV its closest known relatives with which it forms a sister group are the derived SARS viruses BANAL and RaTG Available evidence indicates that it is most likely of zoonotic origin and has close genetic similarity to bat coronaviruses suggesting it emerged from a bat borne virus Research is ongoing as to whether SARS CoV came directly from bats or indirectly through any intermediate hosts The virus shows little genetic diversity indicating that the spillover event introducing SARS CoV to humans is likely to have occurred in late Epidemiological studies estimate that in the December September period each infection resulted in an average of new infections when no members of the community are immune and no preventive measures are taken However some subsequent variants have become more infectious The virus is airborne and primarily spreads between people through close contact and via aerosols and respiratory droplets that are exhaled when talking breathing or otherwise exhaling as well as those produced from coughs and sneezes It enters human cells by binding to angiotensin converting enzyme ACE a membrane protein that regulates the renin angiotensin system During the initial outbreak in Wuhan China various names were used for the virus some names used by different sources included the coronavirus or Wuhan coronavirus In January the World Health Organization WHO recommended novel coronavirus nCoV as the provisional name for the virus This was in accordance with WHO s guidance against using geographical locations animal species or groups of people in disease and virus names On February the International Committee on Taxonomy of Viruses adopted the official name severe acute respiratory syndrome coronavirus SARS CoV To avoid confusion with the disease SARS the WHO sometimes refers to SARS CoV as the COVID virus in public health communications and the name HCoV was included in some research articles Referring to COVID as the Wuhan virus has been described as dangerous by WHO officials and as xenophobic by University of California at Berkeley Asian American studies lecturer Harvey Dong Human to human transmission of SARS CoV was confirmed on January during the COVID pandemic Transmission was initially assumed to occur primarily via respiratory droplets from coughs and sneezes within a range of about metres ft Laser light scattering experiments suggest that speaking is an additional mode of transmission and a far reaching one indoors with little air flow Other studies have suggested that the virus may be airborne as well with aerosols potentially being able to transmit the virus During human to human transmission between and infectious SARS CoV virions are thought to initiate a new infection If confirmed aerosol transmission has biosafety implications because a major concern associated with the risk of working with emerging viruses in the laboratory is the generation of aerosols from various laboratory activities which are not immediately recognizable and may affect other scientific personnel Indirect contact via contaminated surfaces is another possible cause of infection Preliminary research indicates that the virus may remain viable on plastic polypropylene and stainless steel AISI for up to three days but it does not survive on cardboard for more than one day or on copper for more than four hours The virus is inactivated by soap which destabilizes its lipid bilayer Viral RNA has also been found in stool samples and semen from infected individuals The degree to which the virus is infectious during the incubation period is uncertain but research has indicated that the pharynx reaches peak viral load approximately four days after infection or in the first week of symptoms and declines thereafter The duration of SARS CoV RNA shedding is generally between and days after symptom onset A study by a team of researchers from the University of North Carolina found that the nasal cavity is seemingly the dominant initial site of infection with subsequent aspiration mediated virus seeding into the lungs in SARS CoV pathogenesis They found that there was an infection gradient from high in proximal towards low in distal pulmonary epithelial cultures with a focal infection in ciliated cells and type pneumocytes in the airway and alveolar regions respectively Studies have identified a range of animals such as cats ferrets hamsters non human primates minks tree shrews raccoon dogs fruit bats and rabbits that are susceptible and permissive to SARS CoV infection Some institutions have advised that those infected with SARS CoV restrict their contact with animals On February the World Health Organization WHO indicated that transmission from asymptomatic cases is likely not a major driver of transmission One meta analysis found that of infections are asymptomatic and asymptomatic individuals were less likely to transmit the virus However an epidemiological model of the beginning of the outbreak in China suggested that pre symptomatic shedding may be typical among documented infections and that subclinical infections may have been the source of a majority of infections That may explain how out of on board a cruise liner that docked at Montevideo only of who tested positive for viral RNA showed symptoms Similarly a study of ninety four patients hospitalized in January and February estimated patients began shedding virus two to three days before symptoms appear and that a substantial proportion of transmission probably occurred before first symptoms in the index case The authors later published a correction that showed that shedding began earlier than first estimated four to five days before symptoms appear There is uncertainty about reinfection and long term immunity It is not known how common reinfection is but reports have indicated that it is occurring with variable severity The first reported case of reinfection was a year old man from Hong Kong who first tested positive on March was discharged on April after two negative tests and tested positive again on August days later which was confirmed by whole genome sequencing showing that the viral genomes between the episodes belong to different clades The findings had the implications that herd immunity may not eliminate the virus if reinfection is not an uncommon occurrence and that vaccines may not be able to provide lifelong protection against the virus Another case study described a year old man from Nevada who tested positive for SARS CoV on April and on June separated by two negative tests Since genomic analyses showed significant genetic differences between the SARS CoV variant sampled on those two dates the case study authors determined this was a reinfection The man s second infection was symptomatically more severe than the first infection but the mechanisms that could account for this are not known No natural reservoir for SARS CoV has been identified Prior to the emergence of SARS CoV as a pathogen infecting humans there had been two previous zoonosis based coronavirus epidemics those caused by SARS CoV and MERS CoV The first known infections from SARS CoV were discovered in Wuhan China The original source of viral transmission to humans remains unclear as does whether the virus became pathogenic before or after the spillover event Because many of the early infectees were workers at the Huanan Seafood Market it has been suggested that the virus might have originated from the market However other research indicates that visitors may have introduced the virus to the market which then facilitated rapid expansion of the infections A March WHO convened report stated that human spillover via an intermediate animal host was the most likely explanation with direct spillover from bats next most likely Introduction through the food supply chain and the Huanan Seafood Market was considered another possible but less likely explanation An analysis in November however said that the earliest known case had been misidentified and that the preponderance of early cases linked to the Huanan Market argued for it being the source For a virus recently acquired through a cross species transmission rapid evolution is expected The mutation rate estimated from early cases of SARS CoV was of per site per year Coronaviruses in general have high genetic plasticity but SARS CoV s viral evolution is slowed by the RNA proofreading capability of its replication machinery For comparison the viral mutation rate in vivo of SARS CoV has been found to be lower than that of influenza Research into the natural reservoir of the virus that caused the SARS outbreak has resulted in the discovery of many SARS like bat coronaviruses most originating in horseshoe bats The closest match by far published on Nature journal in February were viruses BANAL resemblance to SARS CoV BANAL and BANAL collected in three different species of bats in Feuang Laos An earlier source published in February identified the virus RaTG collected in bats in Mojiang Yunnan China to be the closest to SARS CoV with resemblance None of the above are its direct ancestor Bats are considered the most likely natural reservoir of SARS CoV Differences between the bat coronavirus and SARS CoV suggest that humans may have been infected via an intermediate host although the source of introduction into humans remains unknown Although the role of pangolins as an intermediate host was initially posited a study published in July suggested that pangolins are an intermediate host of SARS CoV like coronaviruses subsequent studies have not substantiated their contribution to the spillover Evidence against this hypothesis includes the fact that pangolin virus samples are too distant to SARS CoV isolates obtained from pangolins seized in Guangdong were only identical in sequence to the SARS CoV genome matches above percent may sound high but in genomic terms it is a wide evolutionary gap In addition despite similarities in a few critical amino acids pangolin virus samples exhibit poor binding to the human ACE receptor SARS CoV belongs to the broad family of viruses known as coronaviruses It is a positive sense single stranded RNA ssRNA virus with a single linear RNA segment Coronaviruses infect humans other mammals including livestock and companion animals and avian species Human coronaviruses are capable of causing illnesses ranging from the common cold to more severe diseases such as Middle East respiratory syndrome MERS fatality rate SARS CoV is the seventh known coronavirus to infect people after E NL OC HKU MERS CoV and the original SARS CoV Like the SARS related coronavirus implicated in the SARS outbreak SARS CoV is a member of the subgenus Sarbecovirus beta CoV lineage B Coronaviruses undergo frequent recombination The mechanism of recombination in unsegmented RNA viruses such as SARS CoV is generally by copy choice replication in which gene material switches from one RNA template molecule to another during replication The SARS CoV RNA sequence is approximately bases in length relatively long for a coronavirus which in turn carry the largest genomes among all RNA families Its genome consists nearly entirely of protein coding sequences a trait shared with other coronaviruses A distinguishing feature of SARS CoV is its incorporation of a polybasic site cleaved by furin which appears to be an important element enhancing its virulence It was suggested that the acquisition of the furin cleavage site in the SARS CoV S protein was essential for zoonotic transfer to humans The furin protease recognizes the canonical peptide sequence RX R K R X where the cleavage site is indicated by a down arrow and X is any amino acid In SARS CoV the recognition site is formed by the incorporated codon nucleotide sequence CCT CGG CGG GCA which corresponds to the amino acid sequence P RR A This sequence is upstream of an arginine and serine which forms the S S cleavage site P RR A R S of the spike protein Although such sites are a common naturally occurring feature of other viruses within the Subfamily Orthocoronavirinae it appears in few other viruses from the Beta CoV genus and it is unique among members of its subgenus for such a site The furin cleavage site PRRAR is highly similar to that of the feline coronavirus an alphacoronavirus strain Viral genetic sequence data can provide critical information about whether viruses separated by time and space are likely to be epidemiologically linked With a sufficient number of sequenced genomes it is possible to reconstruct a phylogenetic tree of the mutation history of a family of viruses By January five genomes of SARS CoV had been isolated from Wuhan and reported by the Chinese Center for Disease Control and Prevention CCDC and other institutions the number of genomes increased to by January A phylogenetic analysis of those samples showed they were highly related with at most seven mutations relative to a common ancestor implying that the first human infection occurred in November or December Examination of the topology of the phylogenetic tree at the start of the pandemic also found high similarities between human isolates As of August update SARS CoV genomes belonging to strains sampled on all continents except Antarctica were publicly available On February the International Committee on Taxonomy of Viruses announced that according to existing rules that compute hierarchical relationships among coronaviruses based on five conserved sequences of nucleic acids the differences between what was then called nCoV and the virus from the SARS outbreak were insufficient to make them separate viral species Therefore they identified nCoV as a virus of Severe acute respiratory syndrome related coronavirus In July scientists reported that a more infectious SARS CoV variant with spike protein variant G has replaced D as the dominant form in the pandemic Coronavirus genomes and subgenomes encode six open reading frames ORFs In October researchers discovered a possible overlapping gene named ORF d in the SARS CoV genome It is unknown if the protein produced by ORF d has any function but it provokes a strong immune response ORF d has been identified before in a variant of coronavirus that infects pangolins A phylogenetic tree based on whole genome sequences of SARS CoV and related coronaviruses is Bat Rc o to SARS CoV Rhinolophus cornutus Iwate Japan Bat SL ZXC to SARS CoV Rhinolophus pusillus Zhoushan Zhejiang Bat SL ZC to SARS CoV Rhinolophus pusillus Zhoushan Zhejiang Pangolin SARSr CoV GX to SARS CoV Manis javanica smuggled from Southeast Asia Pangolin SARSr CoV GD to SARS CoV Manis javanica smuggled from Southeast Asia Bat RshSTT to SARS CoV Rhinolophus shameli Steung Treng Cambodia Bat RshSTT to SARS CoV Rhinolophus shameli Steung Treng Cambodia Bat RacCS to SARS CoV Rhinolophus acuminatus Chachoengsao Thailand Bat RmYN to SARS CoV Rhinolophus malayanus Mengla Yunnan Bat RpYN to SARS CoV Rhinolophus pusillus Xishuangbanna Yunnan Bat RaTG to SARS CoV Rhinolophus affinis Mojiang Yunnan Bat BANAL to SARS CoV Rhinolophus malayanus Vientiane Laos SARS CoV SARS CoV to SARS CoV There are many thousands of variants of SARS CoV which can be grouped into the much larger clades Several different clade nomenclatures have been proposed Nextstrain divides the variants into five clades A B A B and C while GISAID divides them into seven L O V S G GH and GR Several notable variants of SARS CoV emerged in late The World Health Organization has currently declared five variants of concern which are as follows Other notable variants include other WHO designated variants under investigation and Cluster which emerged among mink in Denmark and resulted in a mink euthanasia campaign rendering it virtually extinct Each SARS CoV virion is nanometres in in diameter its mass within the global human populace has been estimated as being between and kilograms Like other coronaviruses SARS CoV has four structural proteins known as the S spike E envelope M membrane and N nucleocapsid proteins the N protein holds the RNA genome and the S E and M proteins together create the viral envelope Coronavirus S proteins are glycoproteins and also type I membrane proteins membranes containing a single transmembrane domain oriented on the extracellular side They are divided into two functional parts S and S In SARS CoV the spike protein which has been imaged at the atomic level using cryogenic electron microscopy is the protein responsible for allowing the virus to attach to and fuse with the membrane of a host cell specifically its S subunit catalyzes attachment the S subunit fusion As of early about million SARS CoV genomes had been sequenced and deposited into public databases and another or so were added each month SARS CoV has a linear positive sense single stranded RNA genome about bases long Its genome has a bias against cytosine C and guanine G nucleotides like other coronaviruses The genome has the highest composition of U followed by A and a similar composition of G and C The nucleotide bias arises from the mutation of guanines and cytosines to adenosines and uracils respectively The mutation of CG dinucleotides is thought to arise to avoid the zinc finger antiviral protein related defense mechanism of cells and to lower the energy to unbind the genome during replication and translation adenosine and uracil base pair via two hydrogen bonds cytosine and guanine via three The depletion of CG dinucleotides in its genome has led the virus to have a noticeable codon usage bias For instance arginine s six different codons have a relative synonymous codon usage of AGA CGU AGG CGC CGA and CGG A similar codon usage bias trend is seen in other SARS related coronaviruses Virus infections start when viral particles bind to host surface cellular receptors Protein modeling experiments on the spike protein of the virus soon suggested that SARS CoV has sufficient affinity to the receptor angiotensin converting enzyme ACE on human cells to use them as a mechanism of cell entry By January a group in China working with the full virus genome and a group in the United States using reverse genetics methods independently and experimentally demonstrated that ACE could act as the receptor for SARS CoV Studies have shown that SARS CoV has a higher affinity to human ACE than the original SARS virus SARS CoV may also use basigin to assist in cell entry Initial spike protein priming by transmembrane protease serine TMPRSS is essential for entry of SARS CoV The host protein neuropilin NRP may aid the virus in host cell entry using ACE After a SARS CoV virion attaches to a target cell the cell s TMPRSS cuts open the spike protein of the virus exposing a fusion peptide in the S subunit and the host receptor ACE After fusion an endosome forms around the virion separating it from the rest of the host cell The virion escapes when the pH of the endosome drops or when cathepsin a host cysteine protease cleaves it The virion then releases RNA into the cell and forces the cell to produce and disseminate copies of the virus which infect more cells SARS CoV produces at least three virulence factors that promote shedding of new virions from host cells and inhibit immune response Whether they include downregulation of ACE as seen in similar coronaviruses remains under investigation as of May Very few drugs are known to effectively inhibit SARS CoV Masitinib is a clinically safe drug and was recently found to inhibit its main protease CLpro and showed fold reduction in viral titers in the lungs and nose in mice However it is not approved for the treatment of COVID in humans as of August needs update In December the United States granted emergency use authorization to Nirmatrelvir ritonavir for the treatment of the virus the European Union United Kingdom and Canada followed suit with full authorization soon after One study found that Nirmatrelvir ritonavir reduced the risk of hospitalization and death by COVID Moonshot is an international collaborative open science project started in March with the goal of developing an un patented oral antiviral drug for treatment of SARS CoV Retrospective tests collected within the Chinese surveillance system revealed no clear indication of substantial unrecognized circulation of SARS CoV in Wuhan during the latter part of A meta analysis from November estimated the basic reproduction number R displaystyle R of the virus to be between and This means each infection from the virus is expected to result in to new infections when no members of the community are immune and no preventive measures are taken The reproduction number may be higher in densely populated conditions such as those found on cruise ships Human behavior affects the R value and hence estimates of R differ between different countries cultures and social norms For instance one study found relatively low R in Sweden Belgium and the Netherlands while Spain and the US had significantly higher R values to respectively There have been about confirmed cases of infection in mainland China While the proportion of infections that result in confirmed cases or progress to diagnosable disease remains unclear one mathematical model estimated that people were infected on January in Wuhan alone at a time when the number of confirmed cases worldwide was only Before February over of all deaths from COVID worldwide had occurred in Hubei province where Wuhan is located As of March the percentage had decreased to As of March there have been total confirmed cases of SARS CoV infection in the ongoing pandemic The total number of deaths attributed to the virus is '"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Removing special characters and digits\n",
    "formatted_article_text = re.sub('[^a-zA-Z]', ' ', article_text )\n",
    "formatted_article_text = re.sub(r'\\s+', ' ', formatted_article_text)\n",
    "\n",
    "formatted_article_text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "8fb69a26",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[nltk_data] Downloading package punkt to\n",
      "[nltk_data]     C:\\Users\\sxp210146\\AppData\\Roaming\\nltk_data...\n",
      "[nltk_data]   Unzipping tokenizers\\punkt.zip.\n"
     ]
    }
   ],
   "source": [
    "#splitting the article into multiple sentences\n",
    "nltk.download('punkt')\n",
    "sentence_list = nltk.sent_tokenize(article_text)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "d65fab94",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[nltk_data] Downloading package stopwords to\n",
      "[nltk_data]     C:\\Users\\sxp210146\\AppData\\Roaming\\nltk_data...\n",
      "[nltk_data]   Unzipping corpora\\stopwords.zip.\n"
     ]
    }
   ],
   "source": [
    "nltk.download('stopwords')\n",
    "stopwords = nltk.corpus.stopwords.words('english')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "a93633c1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[' Severe acute respiratory syndrome coronavirus 2 (SARS‑CoV‑2) is a strain of coronavirus that causes COVID-19 (coronavirus disease 2019), the respiratory illness responsible for the ongoing COVID-19 pandemic.',\n",
       " 'The virus previously had a provisional name, 2019 novel coronavirus (2019-nCoV), and has also been called the human coronavirus 2019 (HCoV-19 or hCoV-19).',\n",
       " 'First identified in the city of Wuhan, Hubei, China, the World Health Organization declared the outbreak a public health emergency of international concern on January 30, 2020, and a pandemic on March 11, 2020.',\n",
       " 'SARS‑CoV‑2 is a positive-sense single-stranded RNA virus that is contagious in humans.',\n",
       " 'SARS‑CoV‑2 is a virus of the species severe acute respiratory syndrome–related coronavirus (SARSr-CoV), related to the SARS-CoV-1 virus that caused the 2002–2004 SARS outbreak.',\n",
       " 'Despite its close relation to SARS-CoV-1, its closest known relatives, with which it forms a sister group, are the derived SARS viruses BANAL-52 and RaTG13.',\n",
       " 'Available evidence indicates that it is most likely of zoonotic origin and has close genetic similarity to bat coronaviruses, suggesting it emerged from a bat-borne virus.',\n",
       " 'Research is ongoing as to whether SARS‑CoV‑2 came directly from bats or indirectly through any intermediate hosts.',\n",
       " 'The virus shows little genetic diversity, indicating that the spillover event introducing SARS‑CoV‑2 to humans is likely to have occurred in late 2019.',\n",
       " 'Epidemiological studies estimate that, in the December 2019–September 2020 period, each infection resulted in an average of 2.4–3.4 new infections when no members of the community are immune and no preventive measures are taken.',\n",
       " 'However, some subsequent variants have become more infectious.',\n",
       " 'The virus is airborne and primarily spreads between people through close contact and via aerosols and respiratory droplets that are exhaled when talking, breathing, or otherwise exhaling, as well as those produced from coughs and sneezes.',\n",
       " 'It enters human cells by binding to angiotensin-converting enzyme 2 (ACE2), a membrane protein that regulates the renin–angiotensin system.',\n",
       " 'During the initial outbreak in Wuhan, China, various names were used for the virus; some names used by different sources included \"the coronavirus\" or \"Wuhan coronavirus\".',\n",
       " 'In January 2020, the World Health Organization (WHO) recommended \"2019 novel coronavirus\" (2019-nCoV) as the provisional name for the virus.',\n",
       " \"This was in accordance with WHO's 2015 guidance against using geographical locations, animal species, or groups of people in disease and virus names.\",\n",
       " 'On 11 February 2020, the International Committee on Taxonomy of Viruses adopted the official name \"severe acute respiratory syndrome coronavirus 2\" (SARS‑CoV‑2).',\n",
       " 'To avoid confusion with the disease SARS, the WHO sometimes refers to SARS‑CoV‑2 as \"the COVID-19 virus\" in public health communications and the name HCoV-19 was included in some research articles.',\n",
       " 'Referring to COVID-19 as the \"Wuhan virus\" has been described as dangerous by WHO officials, and as xenophobic by University of California at Berkeley Asian American studies lecturer Harvey Dong.',\n",
       " 'Human-to-human transmission of SARS‑CoV‑2 was confirmed on 20 January 2020 during the COVID-19 pandemic.',\n",
       " 'Transmission was initially assumed to occur primarily via respiratory droplets from coughs and sneezes within a range of about 1.8 metres (6 ft).',\n",
       " 'Laser light scattering experiments suggest that speaking is an additional mode of transmission and a far-reaching one, indoors, with little air flow.',\n",
       " 'Other studies have suggested that the virus may be airborne as well, with aerosols potentially being able to transmit the virus.',\n",
       " 'During human-to-human transmission, between 200 and 800 infectious SARS‑CoV‑2 virions are thought to initiate a new infection.',\n",
       " 'If confirmed, aerosol transmission has biosafety implications because a major concern associated with the risk of working with emerging viruses in the laboratory is the generation of aerosols from various laboratory activities which are not immediately recognizable and may affect other scientific personnel.',\n",
       " 'Indirect contact via contaminated surfaces is another possible cause of infection.',\n",
       " 'Preliminary research indicates that the virus may remain viable on plastic (polypropylene) and stainless steel (AISI 304) for up to three days, but it does not survive on cardboard for more than one day or on copper for more than four hours.',\n",
       " 'The virus is inactivated by soap, which destabilizes its lipid bilayer.',\n",
       " 'Viral RNA has also been found in stool samples and semen from infected individuals.',\n",
       " 'The degree to which the virus is infectious during the incubation period is uncertain, but research has indicated that the pharynx reaches peak viral load approximately four days after infection or in the first week of symptoms and declines thereafter.',\n",
       " 'The duration of SARS-CoV-2 RNA shedding is generally between 3 and 46 days after symptom onset.',\n",
       " 'A study by a team of researchers from the University of North Carolina found that the nasal cavity is seemingly the dominant initial site of infection, with subsequent aspiration-mediated virus-seeding into the lungs in SARS‑CoV‑2 pathogenesis.',\n",
       " 'They found that there was an infection gradient from high in proximal towards low in distal pulmonary epithelial cultures, with a focal infection in ciliated cells and type 2 pneumocytes in the airway and alveolar regions respectively.',\n",
       " 'Studies have identified a range of animals—such as cats, ferrets, hamsters, non-human primates, minks, tree shrews, raccoon dogs, fruit bats, and rabbits—that are susceptible and permissive to SARS-CoV-2 infection.',\n",
       " 'Some institutions have advised that those infected with SARS‑CoV‑2 restrict their contact with animals.',\n",
       " 'On 1 February 2020, the World Health Organization (WHO) indicated that \"transmission from asymptomatic cases is likely not a major driver of transmission\".',\n",
       " 'One meta-analysis found that 17% of infections are asymptomatic, and asymptomatic individuals were 42% less likely to transmit the virus.',\n",
       " 'However, an epidemiological model of the beginning of the outbreak in China suggested that \"pre-symptomatic shedding may be typical among documented infections\" and that subclinical infections may have been the source of a majority of infections.',\n",
       " 'That may explain how out of 217 on board a cruise liner that docked at Montevideo, only 24 of 128 who tested positive for viral RNA showed symptoms.',\n",
       " 'Similarly, a study of ninety-four patients hospitalized in January and February 2020 estimated patients began shedding virus two to three days before symptoms appear and that \"a substantial proportion of transmission probably occurred before first symptoms in the index case\".',\n",
       " 'The authors later published a correction that showed that shedding began earlier than first estimated, four to five days before symptoms appear.',\n",
       " 'There is uncertainty about reinfection and long-term immunity.',\n",
       " 'It is not known how common reinfection is, but reports have indicated that it is occurring with variable severity.',\n",
       " 'The first reported case of reinfection was a 33-year-old man from Hong Kong who first tested positive on 26 March 2020, was discharged on 15 April 2020 after two negative tests, and tested positive again on 15 August 2020 (142 days later), which was confirmed by whole-genome sequencing showing that the viral genomes between the episodes belong to different clades.',\n",
       " 'The findings had the implications that herd immunity may not eliminate the virus if reinfection is not an uncommon occurrence and that vaccines may not be able to provide lifelong protection against the virus.',\n",
       " 'Another case study described a 25-year-old man from Nevada who tested positive for SARS‑CoV‑2 on 18 April 2020 and on 5 June 2020 (separated by two negative tests).',\n",
       " 'Since genomic analyses showed significant genetic differences between the SARS‑CoV‑2 variant sampled on those two dates, the case study authors determined this was a reinfection.',\n",
       " \"The man's second infection was symptomatically more severe than the first infection, but the mechanisms that could account for this are not known.\",\n",
       " 'No natural reservoir for SARS-CoV-2 has been identified.',\n",
       " 'Prior to the emergence of SARS-CoV-2 as a pathogen infecting humans, there had been two previous zoonosis-based coronavirus epidemics, those caused by SARS-CoV-1 and MERS-CoV.',\n",
       " 'The first known infections from SARS‑CoV‑2 were discovered in Wuhan, China.',\n",
       " 'The original source of viral transmission to humans remains unclear, as does whether the virus became pathogenic before or after the spillover event.',\n",
       " 'Because many of the early infectees were workers at the Huanan Seafood Market, it has been suggested that the virus might have originated from the market.',\n",
       " 'However, other research indicates that visitors may have introduced the virus to the market, which then facilitated rapid expansion of the infections.',\n",
       " 'A March 2021 WHO-convened report stated that human spillover via an intermediate animal host was the most likely explanation, with direct spillover from bats next most likely.',\n",
       " 'Introduction through the food supply chain and the Huanan Seafood Market was considered another possible, but less likely, explanation.',\n",
       " 'An analysis in November 2021, however, said that the earliest-known case had been misidentified and that the preponderance of early cases linked to the Huanan Market argued for it being the source.',\n",
       " 'For a virus recently acquired through a cross-species transmission, rapid evolution is expected.',\n",
       " 'The mutation rate estimated from early cases of SARS-CoV-2 was of 6.54×10−4 per site per year.',\n",
       " \"Coronaviruses in general have high genetic plasticity, but SARS-CoV-2's viral evolution is slowed by the RNA proofreading capability of its replication machinery.\",\n",
       " 'For comparison, the viral mutation rate in vivo of SARS-CoV-2 has been found to be lower than that of influenza.',\n",
       " 'Research into the natural reservoir of the virus that caused the 2002–2004 SARS outbreak has resulted in the discovery of many SARS-like bat coronaviruses, most originating in horseshoe bats.',\n",
       " 'The closest match by far, published on Nature (journal) in February 2022, were viruses BANAL-52 (96.8% resemblance to SARS‑CoV‑2), BANAL-103 and BANAL-236, collected in three different species of bats in Feuang, Laos.',\n",
       " 'An earlier source published in February 2020 identified the virus RaTG13, collected in bats in Mojiang, Yunnan, China to be the closest to SARS‑CoV‑2, with 96.1% resemblance.',\n",
       " 'None of the above are its direct ancestor.',\n",
       " 'Bats are considered the most likely natural reservoir of SARS‑CoV‑2.',\n",
       " 'Differences between the bat coronavirus and SARS‑CoV‑2 suggest that humans may have been infected via an intermediate host; although the source of introduction into humans remains unknown.',\n",
       " 'Although the role of pangolins as an intermediate host was initially posited (a study published in July 2020 suggested that pangolins are an intermediate host of SARS‑CoV‑2-like coronaviruses ), subsequent studies have not substantiated their contribution to the spillover.',\n",
       " 'Evidence against this hypothesis includes the fact that pangolin virus samples are too distant to SARS-CoV-2: isolates obtained from pangolins seized in Guangdong were only 92% identical in sequence to the SARS‑CoV‑2 genome (matches above 90 percent may sound high, but in genomic terms it is a wide evolutionary gap ).',\n",
       " 'In addition, despite similarities in a few critical amino acids, pangolin virus samples exhibit poor binding to the human ACE2 receptor.',\n",
       " 'SARS‑CoV‑2 belongs to the broad family of viruses known as coronaviruses.',\n",
       " 'It is a positive-sense single-stranded RNA (+ssRNA) virus, with a single linear RNA segment.',\n",
       " 'Coronaviruses infect humans, other mammals, including livestock and companion animals, and avian species.',\n",
       " 'Human coronaviruses are capable of causing illnesses ranging from the common cold to more severe diseases such as Middle East respiratory syndrome (MERS, fatality rate ~34%).',\n",
       " 'SARS-CoV-2 is the seventh known coronavirus to infect people, after 229E, NL63, OC43, HKU1, MERS-CoV, and the original SARS-CoV.',\n",
       " 'Like the SARS-related coronavirus implicated in the 2003 SARS outbreak, SARS‑CoV‑2 is a member of the subgenus Sarbecovirus (beta-CoV lineage B).',\n",
       " 'Coronaviruses undergo frequent recombination.',\n",
       " 'The mechanism of recombination in unsegmented RNA viruses such as SARS-CoV-2 is generally by copy-choice replication, in which gene material switches from one RNA template molecule to another during replication.',\n",
       " 'The SARS-CoV-2 RNA sequence is approximately 30,000 bases in length, relatively long for a coronavirus—which in turn carry the largest genomes among all RNA families.',\n",
       " 'Its genome consists nearly entirely of protein-coding sequences, a trait shared with other coronaviruses.',\n",
       " 'A distinguishing feature of SARS‑CoV‑2 is its incorporation of a polybasic site cleaved by furin, which appears to be an important element enhancing its virulence.',\n",
       " 'It was suggested that the acquisition of the furin-cleavage site in the SARS-CoV-2 S protein was essential for zoonotic transfer to humans.',\n",
       " 'The furin protease recognizes the canonical peptide sequence RX[R/K] R↓X where the cleavage site is indicated by a down arrow and X is any amino acid.',\n",
       " 'In SARS-CoV-2 the recognition site is formed by the incorporated 12 codon nucleotide sequence CCT CGG CGG GCA which corresponds to the amino acid sequence P RR A.',\n",
       " 'This sequence is upstream of an arginine and serine which forms the S1/S2 cleavage site (P RR A R↓S) of the spike protein.',\n",
       " 'Although such sites are a common naturally-occurring feature of other viruses within the Subfamily Orthocoronavirinae, it appears in few other viruses from the Beta-CoV genus, and it is unique among members of its subgenus for such a site.',\n",
       " 'The furin cleavage site PRRAR↓ is highly similar to that of the feline coronavirus, an alphacoronavirus 1 strain.',\n",
       " 'Viral genetic sequence data can provide critical information about whether viruses separated by time and space are likely to be epidemiologically linked.',\n",
       " 'With a sufficient number of sequenced genomes, it is possible to reconstruct a phylogenetic tree of the mutation history of a family of viruses.',\n",
       " 'By 12 January 2020, five genomes of SARS‑CoV‑2 had been isolated from Wuhan and reported by the Chinese Center for Disease Control and Prevention (CCDC) and other institutions; the number of genomes increased to 42 by 30 January 2020.',\n",
       " 'A phylogenetic analysis of those samples showed they were \"highly related with at most seven mutations relative to a common ancestor\", implying that the first human infection occurred in November or December 2019.',\n",
       " 'Examination of the topology of the phylogenetic tree at the start of the pandemic also found high similarities between human isolates.',\n",
       " 'As of 21 August 2021,[update] 3,422 SARS‑CoV‑2 genomes, belonging to 19 strains, sampled on all continents except Antarctica were publicly available.',\n",
       " 'On 11 February 2020, the International Committee on Taxonomy of Viruses announced that according to existing rules that compute hierarchical relationships among coronaviruses based on five conserved sequences of nucleic acids, the differences between what was then called 2019-nCoV and the virus from the 2003 SARS outbreak were insufficient to make them separate viral species.',\n",
       " 'Therefore, they identified 2019-nCoV as a virus of Severe acute respiratory syndrome–related coronavirus.',\n",
       " 'In July 2020, scientists reported that a more infectious SARS‑CoV‑2 variant with spike protein variant G614 has replaced D614 as the dominant form in the pandemic.',\n",
       " 'Coronavirus genomes and subgenomes encode six open reading frames (ORFs).',\n",
       " 'In October 2020, researchers discovered a possible overlapping gene named ORF3d, in the SARS‑CoV‑2 genome.',\n",
       " 'It is unknown if the protein produced by ORF3d has any function, but it provokes a strong immune response.',\n",
       " 'ORF3d has been identified before, in a variant of coronavirus that infects pangolins.',\n",
       " 'A phylogenetic tree based on whole-genome sequences of SARS-CoV-2 and related coronaviruses is: (Bat) Rc-o319, 81% to SARS-CoV-2, Rhinolophus cornutus, Iwate, Japan Bat SL-ZXC21, 88% to SARS-CoV-2, Rhinolophus pusillus, Zhoushan, Zhejiang Bat SL-ZC45, 88% to SARS-CoV-2, Rhinolophus pusillus, Zhoushan, Zhejiang Pangolin SARSr-CoV-GX, 85.3% to SARS-CoV-2, Manis javanica, smuggled from Southeast Asia Pangolin SARSr-CoV-GD, 90.1% to SARS-CoV-2, Manis javanica, smuggled from Southeast Asia Bat RshSTT182, 92.6% to SARS-CoV-2, Rhinolophus shameli, Steung Treng, Cambodia Bat RshSTT200, 92.6% to SARS-CoV-2, Rhinolophus shameli, Steung Treng, Cambodia (Bat) RacCS203, 91.5% to SARS-CoV-2, Rhinolophus acuminatus, Chachoengsao, Thailand (Bat) RmYN02, 93.3% to SARS-CoV-2, Rhinolophus malayanus, Mengla, Yunnan (Bat) RpYN06, 94.4% to SARS-CoV-2, Rhinolophus pusillus, Xishuangbanna, Yunnan (Bat) RaTG13, 96.1% to SARS-CoV-2, Rhinolophus affinis, Mojiang, Yunnan (Bat) BANAL-52, 96.8% to SARS-CoV-2, Rhinolophus malayanus, Vientiane, Laos SARS-CoV-2 SARS-CoV-1, 79% to SARS-CoV-2 There are many thousands of variants of SARS-CoV-2, which can be grouped into the much larger clades.',\n",
       " 'Several different clade nomenclatures have been proposed.',\n",
       " 'Nextstrain divides the variants into five clades (19A, 19B, 20A, 20B, and 20C), while GISAID divides them into seven (L, O, V, S, G, GH, and GR).',\n",
       " 'Several notable variants of SARS-CoV-2 emerged in late 2020.',\n",
       " 'The World Health Organization has currently declared five variants of concern, which are as follows: Other notable variants include 6 other WHO-designated variants under investigation and Cluster 5, which emerged among mink in Denmark and resulted in a mink euthanasia campaign rendering it virtually extinct.',\n",
       " 'Each SARS-CoV-2 virion is 60–140 nanometres (2.4×10−6–5.5×10−6 in) in diameter; its mass within the global human populace has been estimated as being between 0.1 and 10 kilograms.',\n",
       " 'Like other coronaviruses, SARS-CoV-2 has four structural proteins, known as the S (spike), E (envelope), M (membrane), and N (nucleocapsid) proteins; the N protein holds the RNA genome, and the S, E, and M proteins together create the viral envelope.',\n",
       " 'Coronavirus S proteins are glycoproteins and also type I membrane proteins (membranes containing a single transmembrane domain oriented on the extracellular side).',\n",
       " 'They are divided into two functional parts (S1 and S2).',\n",
       " 'In SARS-CoV-2, the spike protein, which has been imaged at the atomic level using cryogenic electron microscopy, is the protein responsible for allowing the virus to attach to and fuse with the membrane of a host cell; specifically, its S1 subunit catalyzes attachment, the S2 subunit fusion.',\n",
       " 'As of early 2022, about 7 million SARS-CoV-2 genomes had been sequenced and deposited into public databases and another 800,000 or so were added each month.',\n",
       " 'SARS-CoV-2 has a linear, positive-sense, single-stranded RNA genome about 30,000 bases long.',\n",
       " 'Its genome has a bias against cytosine (C) and guanine (G) nucleotides, like other coronaviruses.',\n",
       " 'The genome has the highest composition of U (32.2%), followed by A (29.9%), and a similar composition of G (19.6%) and C (18.3%).',\n",
       " 'The nucleotide bias arises from the mutation of guanines and cytosines to adenosines and uracils, respectively.',\n",
       " 'The mutation of CG dinucleotides is thought to arise to avoid the zinc finger antiviral protein related defense mechanism of cells, and to lower the energy to unbind the genome during replication and translation (adenosine and uracil base pair via two hydrogen bonds, cytosine and guanine via three).',\n",
       " 'The depletion of CG dinucleotides in its genome has led the virus to have a noticeable codon usage bias.',\n",
       " \"For instance, arginine's six different codons have a relative synonymous codon usage of AGA (2.67), CGU (1.46), AGG (.81), CGC (.58), CGA (.29), and CGG (.19).\",\n",
       " 'A similar codon usage bias trend is seen in other SARS–related coronaviruses.',\n",
       " 'Virus infections start when viral particles bind to host surface cellular receptors.',\n",
       " 'Protein modeling experiments on the spike protein of the virus soon suggested that SARS‑CoV‑2 has sufficient affinity to the receptor angiotensin converting enzyme 2 (ACE2) on human cells to use them as a mechanism of cell entry.',\n",
       " 'By 22 January 2020, a group in China working with the full virus genome and a group in the United States using reverse genetics methods independently and experimentally demonstrated that ACE2 could act as the receptor for SARS‑CoV‑2.',\n",
       " 'Studies have shown that SARS‑CoV‑2 has a higher affinity to human ACE2 than the original SARS virus.',\n",
       " 'SARS‑CoV‑2 may also use basigin to assist in cell entry.',\n",
       " 'Initial spike protein priming by transmembrane protease, serine 2 (TMPRSS2) is essential for entry of SARS‑CoV‑2.',\n",
       " 'The host protein neuropilin 1 (NRP1) may aid the virus in host cell entry using ACE2.',\n",
       " \"After a SARS‑CoV‑2 virion attaches to a target cell, the cell's TMPRSS2 cuts open the spike protein of the virus, exposing a fusion peptide in the S2 subunit, and the host receptor ACE2.\",\n",
       " 'After fusion, an endosome forms around the virion, separating it from the rest of the host cell.',\n",
       " 'The virion escapes when the pH of the endosome drops or when cathepsin, a host cysteine protease, cleaves it.',\n",
       " 'The virion then releases RNA into the cell and forces the cell to produce and disseminate copies of the virus, which infect more cells.',\n",
       " 'SARS‑CoV‑2 produces at least three virulence factors that promote shedding of new virions from host cells and inhibit immune response.',\n",
       " 'Whether they include downregulation of ACE2, as seen in similar coronaviruses, remains under investigation (as of May 2020).',\n",
       " 'Very few drugs are known to effectively inhibit SARS‑CoV‑2.',\n",
       " 'Masitinib is a clinically safe drug and was recently found to inhibit its main protease, 3CLpro and showed >200-fold reduction in viral titers in the lungs and nose in mice.',\n",
       " 'However, it is not approved for the treatment of COVID-19 in humans as of August 2021.',\n",
       " '[needs update] In December 2021, the United States granted emergency use authorization to Nirmatrelvir/ritonavir for the treatment of the virus; the European Union, United Kingdom, and Canada followed suit with full authorization soon after.',\n",
       " 'One study found that Nirmatrelvir/ritonavir reduced the risk of hospitalization and death by 88%.',\n",
       " 'COVID Moonshot is an international collaborative open-science project started in March 2020 with the goal of developing an un-patented oral antiviral drug for treatment of SARS-CoV-2.',\n",
       " 'Retrospective tests collected within the Chinese surveillance system revealed no clear indication of substantial unrecognized circulation of SARS‑CoV‑2 in Wuhan during the latter part of 2019.',\n",
       " 'A meta-analysis from November 2020 estimated the basic reproduction number ( R 0 {\\\\displaystyle R_{0}} ) of the virus to be between 2.39 and 3.44.',\n",
       " 'This means each infection from the virus is expected to result in 2.39 to 3.44 new infections when no members of the community are immune and no preventive measures are taken.',\n",
       " 'The reproduction number may be higher in densely populated conditions such as those found on cruise ships.',\n",
       " 'Human behavior affects the R0 value and hence estimates of R0 differ between different countries, cultures, and social norms.',\n",
       " 'For instance, one study found relatively low R0 (~3.5) in Sweden, Belgium and the Netherlands, while Spain and the US had significantly higher R0 values (5.9 to 6.4, respectively).',\n",
       " 'There have been about 96,000 confirmed cases of infection in mainland China.',\n",
       " 'While the proportion of infections that result in confirmed cases or progress to diagnosable disease remains unclear, one mathematical model estimated that 75,815 people were infected on 25 January 2020 in Wuhan alone, at a time when the number of confirmed cases worldwide was only 2,015.',\n",
       " 'Before 24 February 2020, over 95% of all deaths from COVID-19 worldwide had occurred in Hubei province, where Wuhan is located.',\n",
       " 'As of 3 March 2023, the percentage had decreased to 0.047%.',\n",
       " 'As of 3 March 2023, there have been 675,816,641 total confirmed cases of SARS‑CoV‑2 infection in the ongoing pandemic.',\n",
       " 'The total number of deaths attributed to the virus is 6,876,102.']"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sentence_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "b74b77ef",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "88"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Finding word frequencies for every word\n",
    "\n",
    "word_frequencies = {}\n",
    "for word in nltk.word_tokenize(formatted_article_text):\n",
    "    if word not in stopwords:\n",
    "        if word not in word_frequencies.keys():\n",
    "            word_frequencies[word] = 1\n",
    "        else:\n",
    "            word_frequencies[word] += 1\n",
    "\n",
    "maximum_frequncy = max(word_frequencies.values())\n",
    "\n",
    "maximum_frequncy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "5513a34f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Severe': 0.022727272727272728,\n",
       " 'acute': 0.045454545454545456,\n",
       " 'respiratory': 0.09090909090909091,\n",
       " 'syndrome': 0.056818181818181816,\n",
       " 'coronavirus': 0.20454545454545456,\n",
       " 'SARS': 1.0,\n",
       " 'CoV': 0.9659090909090909,\n",
       " 'strain': 0.022727272727272728,\n",
       " 'causes': 0.011363636363636364,\n",
       " 'COVID': 0.09090909090909091,\n",
       " 'disease': 0.045454545454545456,\n",
       " 'illness': 0.011363636363636364,\n",
       " 'responsible': 0.022727272727272728,\n",
       " 'ongoing': 0.03409090909090909,\n",
       " 'pandemic': 0.06818181818181818,\n",
       " 'The': 0.3181818181818182,\n",
       " 'virus': 0.5113636363636364,\n",
       " 'previously': 0.011363636363636364,\n",
       " 'provisional': 0.022727272727272728,\n",
       " 'name': 0.045454545454545456,\n",
       " 'novel': 0.022727272727272728,\n",
       " 'nCoV': 0.045454545454545456,\n",
       " 'also': 0.056818181818181816,\n",
       " 'called': 0.022727272727272728,\n",
       " 'human': 0.14772727272727273,\n",
       " 'HCoV': 0.022727272727272728,\n",
       " 'hCoV': 0.011363636363636364,\n",
       " 'First': 0.011363636363636364,\n",
       " 'identified': 0.06818181818181818,\n",
       " 'city': 0.011363636363636364,\n",
       " 'Wuhan': 0.10227272727272728,\n",
       " 'Hubei': 0.022727272727272728,\n",
       " 'China': 0.07954545454545454,\n",
       " 'World': 0.045454545454545456,\n",
       " 'Health': 0.045454545454545456,\n",
       " 'Organization': 0.045454545454545456,\n",
       " 'declared': 0.022727272727272728,\n",
       " 'outbreak': 0.07954545454545454,\n",
       " 'public': 0.03409090909090909,\n",
       " 'health': 0.022727272727272728,\n",
       " 'emergency': 0.022727272727272728,\n",
       " 'international': 0.022727272727272728,\n",
       " 'concern': 0.03409090909090909,\n",
       " 'January': 0.09090909090909091,\n",
       " 'March': 0.06818181818181818,\n",
       " 'positive': 0.07954545454545454,\n",
       " 'sense': 0.03409090909090909,\n",
       " 'single': 0.056818181818181816,\n",
       " 'stranded': 0.03409090909090909,\n",
       " 'RNA': 0.1590909090909091,\n",
       " 'contagious': 0.011363636363636364,\n",
       " 'humans': 0.10227272727272728,\n",
       " 'species': 0.06818181818181818,\n",
       " 'severe': 0.045454545454545456,\n",
       " 'related': 0.09090909090909091,\n",
       " 'SARSr': 0.03409090909090909,\n",
       " 'caused': 0.03409090909090909,\n",
       " 'Despite': 0.011363636363636364,\n",
       " 'close': 0.03409090909090909,\n",
       " 'relation': 0.011363636363636364,\n",
       " 'closest': 0.03409090909090909,\n",
       " 'known': 0.10227272727272728,\n",
       " 'relatives': 0.011363636363636364,\n",
       " 'forms': 0.03409090909090909,\n",
       " 'sister': 0.011363636363636364,\n",
       " 'group': 0.03409090909090909,\n",
       " 'derived': 0.011363636363636364,\n",
       " 'viruses': 0.10227272727272728,\n",
       " 'BANAL': 0.056818181818181816,\n",
       " 'RaTG': 0.03409090909090909,\n",
       " 'Available': 0.011363636363636364,\n",
       " 'evidence': 0.011363636363636364,\n",
       " 'indicates': 0.03409090909090909,\n",
       " 'likely': 0.10227272727272728,\n",
       " 'zoonotic': 0.022727272727272728,\n",
       " 'origin': 0.011363636363636364,\n",
       " 'genetic': 0.056818181818181816,\n",
       " 'similarity': 0.011363636363636364,\n",
       " 'bat': 0.045454545454545456,\n",
       " 'coronaviruses': 0.13636363636363635,\n",
       " 'suggesting': 0.011363636363636364,\n",
       " 'emerged': 0.03409090909090909,\n",
       " 'borne': 0.011363636363636364,\n",
       " 'Research': 0.022727272727272728,\n",
       " 'whether': 0.03409090909090909,\n",
       " 'came': 0.011363636363636364,\n",
       " 'directly': 0.011363636363636364,\n",
       " 'bats': 0.06818181818181818,\n",
       " 'indirectly': 0.011363636363636364,\n",
       " 'intermediate': 0.056818181818181816,\n",
       " 'hosts': 0.011363636363636364,\n",
       " 'shows': 0.011363636363636364,\n",
       " 'little': 0.022727272727272728,\n",
       " 'diversity': 0.011363636363636364,\n",
       " 'indicating': 0.011363636363636364,\n",
       " 'spillover': 0.056818181818181816,\n",
       " 'event': 0.022727272727272728,\n",
       " 'introducing': 0.011363636363636364,\n",
       " 'occurred': 0.045454545454545456,\n",
       " 'late': 0.022727272727272728,\n",
       " 'Epidemiological': 0.011363636363636364,\n",
       " 'studies': 0.045454545454545456,\n",
       " 'estimate': 0.011363636363636364,\n",
       " 'December': 0.03409090909090909,\n",
       " 'September': 0.011363636363636364,\n",
       " 'period': 0.022727272727272728,\n",
       " 'infection': 0.1590909090909091,\n",
       " 'resulted': 0.03409090909090909,\n",
       " 'average': 0.011363636363636364,\n",
       " 'new': 0.045454545454545456,\n",
       " 'infections': 0.11363636363636363,\n",
       " 'members': 0.03409090909090909,\n",
       " 'community': 0.022727272727272728,\n",
       " 'immune': 0.045454545454545456,\n",
       " 'preventive': 0.022727272727272728,\n",
       " 'measures': 0.022727272727272728,\n",
       " 'taken': 0.022727272727272728,\n",
       " 'However': 0.045454545454545456,\n",
       " 'subsequent': 0.03409090909090909,\n",
       " 'variants': 0.07954545454545454,\n",
       " 'become': 0.011363636363636364,\n",
       " 'infectious': 0.045454545454545456,\n",
       " 'airborne': 0.022727272727272728,\n",
       " 'primarily': 0.022727272727272728,\n",
       " 'spreads': 0.011363636363636364,\n",
       " 'people': 0.045454545454545456,\n",
       " 'contact': 0.03409090909090909,\n",
       " 'via': 0.07954545454545454,\n",
       " 'aerosols': 0.03409090909090909,\n",
       " 'droplets': 0.022727272727272728,\n",
       " 'exhaled': 0.011363636363636364,\n",
       " 'talking': 0.011363636363636364,\n",
       " 'breathing': 0.011363636363636364,\n",
       " 'otherwise': 0.011363636363636364,\n",
       " 'exhaling': 0.011363636363636364,\n",
       " 'well': 0.022727272727272728,\n",
       " 'produced': 0.022727272727272728,\n",
       " 'coughs': 0.022727272727272728,\n",
       " 'sneezes': 0.022727272727272728,\n",
       " 'It': 0.056818181818181816,\n",
       " 'enters': 0.011363636363636364,\n",
       " 'cells': 0.06818181818181818,\n",
       " 'binding': 0.022727272727272728,\n",
       " 'angiotensin': 0.03409090909090909,\n",
       " 'converting': 0.022727272727272728,\n",
       " 'enzyme': 0.022727272727272728,\n",
       " 'ACE': 0.09090909090909091,\n",
       " 'membrane': 0.045454545454545456,\n",
       " 'protein': 0.1590909090909091,\n",
       " 'regulates': 0.011363636363636364,\n",
       " 'renin': 0.011363636363636364,\n",
       " 'system': 0.022727272727272728,\n",
       " 'During': 0.022727272727272728,\n",
       " 'initial': 0.022727272727272728,\n",
       " 'various': 0.022727272727272728,\n",
       " 'names': 0.03409090909090909,\n",
       " 'used': 0.022727272727272728,\n",
       " 'different': 0.06818181818181818,\n",
       " 'sources': 0.011363636363636364,\n",
       " 'included': 0.022727272727272728,\n",
       " 'In': 0.07954545454545454,\n",
       " 'WHO': 0.07954545454545454,\n",
       " 'recommended': 0.011363636363636364,\n",
       " 'This': 0.03409090909090909,\n",
       " 'accordance': 0.011363636363636364,\n",
       " 'guidance': 0.011363636363636364,\n",
       " 'using': 0.045454545454545456,\n",
       " 'geographical': 0.011363636363636364,\n",
       " 'locations': 0.011363636363636364,\n",
       " 'animal': 0.022727272727272728,\n",
       " 'groups': 0.011363636363636364,\n",
       " 'On': 0.03409090909090909,\n",
       " 'February': 0.07954545454545454,\n",
       " 'International': 0.022727272727272728,\n",
       " 'Committee': 0.022727272727272728,\n",
       " 'Taxonomy': 0.022727272727272728,\n",
       " 'Viruses': 0.022727272727272728,\n",
       " 'adopted': 0.011363636363636364,\n",
       " 'official': 0.011363636363636364,\n",
       " 'To': 0.011363636363636364,\n",
       " 'avoid': 0.022727272727272728,\n",
       " 'confusion': 0.011363636363636364,\n",
       " 'sometimes': 0.011363636363636364,\n",
       " 'refers': 0.011363636363636364,\n",
       " 'communications': 0.011363636363636364,\n",
       " 'research': 0.045454545454545456,\n",
       " 'articles': 0.011363636363636364,\n",
       " 'Referring': 0.011363636363636364,\n",
       " 'described': 0.022727272727272728,\n",
       " 'dangerous': 0.011363636363636364,\n",
       " 'officials': 0.011363636363636364,\n",
       " 'xenophobic': 0.011363636363636364,\n",
       " 'University': 0.022727272727272728,\n",
       " 'California': 0.011363636363636364,\n",
       " 'Berkeley': 0.011363636363636364,\n",
       " 'Asian': 0.011363636363636364,\n",
       " 'American': 0.011363636363636364,\n",
       " 'lecturer': 0.011363636363636364,\n",
       " 'Harvey': 0.011363636363636364,\n",
       " 'Dong': 0.011363636363636364,\n",
       " 'Human': 0.03409090909090909,\n",
       " 'transmission': 0.10227272727272728,\n",
       " 'confirmed': 0.07954545454545454,\n",
       " 'Transmission': 0.011363636363636364,\n",
       " 'initially': 0.022727272727272728,\n",
       " 'assumed': 0.011363636363636364,\n",
       " 'occur': 0.011363636363636364,\n",
       " 'within': 0.045454545454545456,\n",
       " 'range': 0.022727272727272728,\n",
       " 'metres': 0.011363636363636364,\n",
       " 'ft': 0.011363636363636364,\n",
       " 'Laser': 0.011363636363636364,\n",
       " 'light': 0.011363636363636364,\n",
       " 'scattering': 0.011363636363636364,\n",
       " 'experiments': 0.022727272727272728,\n",
       " 'suggest': 0.022727272727272728,\n",
       " 'speaking': 0.011363636363636364,\n",
       " 'additional': 0.011363636363636364,\n",
       " 'mode': 0.011363636363636364,\n",
       " 'far': 0.022727272727272728,\n",
       " 'reaching': 0.011363636363636364,\n",
       " 'one': 0.056818181818181816,\n",
       " 'indoors': 0.011363636363636364,\n",
       " 'air': 0.011363636363636364,\n",
       " 'flow': 0.011363636363636364,\n",
       " 'Other': 0.022727272727272728,\n",
       " 'suggested': 0.06818181818181818,\n",
       " 'may': 0.1590909090909091,\n",
       " 'potentially': 0.011363636363636364,\n",
       " 'able': 0.022727272727272728,\n",
       " 'transmit': 0.022727272727272728,\n",
       " 'virions': 0.022727272727272728,\n",
       " 'thought': 0.022727272727272728,\n",
       " 'initiate': 0.011363636363636364,\n",
       " 'If': 0.011363636363636364,\n",
       " 'aerosol': 0.011363636363636364,\n",
       " 'biosafety': 0.011363636363636364,\n",
       " 'implications': 0.022727272727272728,\n",
       " 'major': 0.022727272727272728,\n",
       " 'associated': 0.011363636363636364,\n",
       " 'risk': 0.022727272727272728,\n",
       " 'working': 0.022727272727272728,\n",
       " 'emerging': 0.011363636363636364,\n",
       " 'laboratory': 0.022727272727272728,\n",
       " 'generation': 0.011363636363636364,\n",
       " 'activities': 0.011363636363636364,\n",
       " 'immediately': 0.011363636363636364,\n",
       " 'recognizable': 0.011363636363636364,\n",
       " 'affect': 0.011363636363636364,\n",
       " 'scientific': 0.011363636363636364,\n",
       " 'personnel': 0.011363636363636364,\n",
       " 'Indirect': 0.011363636363636364,\n",
       " 'contaminated': 0.011363636363636364,\n",
       " 'surfaces': 0.011363636363636364,\n",
       " 'another': 0.045454545454545456,\n",
       " 'possible': 0.045454545454545456,\n",
       " 'cause': 0.011363636363636364,\n",
       " 'Preliminary': 0.011363636363636364,\n",
       " 'remain': 0.011363636363636364,\n",
       " 'viable': 0.011363636363636364,\n",
       " 'plastic': 0.011363636363636364,\n",
       " 'polypropylene': 0.011363636363636364,\n",
       " 'stainless': 0.011363636363636364,\n",
       " 'steel': 0.011363636363636364,\n",
       " 'AISI': 0.011363636363636364,\n",
       " 'three': 0.056818181818181816,\n",
       " 'days': 0.06818181818181818,\n",
       " 'survive': 0.011363636363636364,\n",
       " 'cardboard': 0.011363636363636364,\n",
       " 'day': 0.011363636363636364,\n",
       " 'copper': 0.011363636363636364,\n",
       " 'four': 0.056818181818181816,\n",
       " 'hours': 0.011363636363636364,\n",
       " 'inactivated': 0.011363636363636364,\n",
       " 'soap': 0.011363636363636364,\n",
       " 'destabilizes': 0.011363636363636364,\n",
       " 'lipid': 0.011363636363636364,\n",
       " 'bilayer': 0.011363636363636364,\n",
       " 'Viral': 0.022727272727272728,\n",
       " 'found': 0.11363636363636363,\n",
       " 'stool': 0.011363636363636364,\n",
       " 'samples': 0.045454545454545456,\n",
       " 'semen': 0.011363636363636364,\n",
       " 'infected': 0.045454545454545456,\n",
       " 'individuals': 0.022727272727272728,\n",
       " 'degree': 0.011363636363636364,\n",
       " 'incubation': 0.011363636363636364,\n",
       " 'uncertain': 0.011363636363636364,\n",
       " 'indicated': 0.045454545454545456,\n",
       " 'pharynx': 0.011363636363636364,\n",
       " 'reaches': 0.011363636363636364,\n",
       " 'peak': 0.011363636363636364,\n",
       " 'viral': 0.11363636363636363,\n",
       " 'load': 0.011363636363636364,\n",
       " 'approximately': 0.022727272727272728,\n",
       " 'first': 0.09090909090909091,\n",
       " 'week': 0.011363636363636364,\n",
       " 'symptoms': 0.056818181818181816,\n",
       " 'declines': 0.011363636363636364,\n",
       " 'thereafter': 0.011363636363636364,\n",
       " 'duration': 0.011363636363636364,\n",
       " 'shedding': 0.056818181818181816,\n",
       " 'generally': 0.022727272727272728,\n",
       " 'symptom': 0.011363636363636364,\n",
       " 'onset': 0.011363636363636364,\n",
       " 'A': 0.13636363636363635,\n",
       " 'study': 0.07954545454545454,\n",
       " 'team': 0.011363636363636364,\n",
       " 'researchers': 0.022727272727272728,\n",
       " 'North': 0.011363636363636364,\n",
       " 'Carolina': 0.011363636363636364,\n",
       " 'nasal': 0.011363636363636364,\n",
       " 'cavity': 0.011363636363636364,\n",
       " 'seemingly': 0.011363636363636364,\n",
       " 'dominant': 0.022727272727272728,\n",
       " 'site': 0.10227272727272728,\n",
       " 'aspiration': 0.011363636363636364,\n",
       " 'mediated': 0.011363636363636364,\n",
       " 'seeding': 0.011363636363636364,\n",
       " 'lungs': 0.022727272727272728,\n",
       " 'pathogenesis': 0.011363636363636364,\n",
       " 'They': 0.022727272727272728,\n",
       " 'gradient': 0.011363636363636364,\n",
       " 'high': 0.045454545454545456,\n",
       " 'proximal': 0.011363636363636364,\n",
       " 'towards': 0.011363636363636364,\n",
       " 'low': 0.022727272727272728,\n",
       " 'distal': 0.011363636363636364,\n",
       " 'pulmonary': 0.011363636363636364,\n",
       " 'epithelial': 0.011363636363636364,\n",
       " 'cultures': 0.022727272727272728,\n",
       " 'focal': 0.011363636363636364,\n",
       " 'ciliated': 0.011363636363636364,\n",
       " 'type': 0.022727272727272728,\n",
       " 'pneumocytes': 0.011363636363636364,\n",
       " 'airway': 0.011363636363636364,\n",
       " 'alveolar': 0.011363636363636364,\n",
       " 'regions': 0.011363636363636364,\n",
       " 'respectively': 0.03409090909090909,\n",
       " 'Studies': 0.022727272727272728,\n",
       " 'animals': 0.03409090909090909,\n",
       " 'cats': 0.011363636363636364,\n",
       " 'ferrets': 0.011363636363636364,\n",
       " 'hamsters': 0.011363636363636364,\n",
       " 'non': 0.011363636363636364,\n",
       " 'primates': 0.011363636363636364,\n",
       " 'minks': 0.011363636363636364,\n",
       " 'tree': 0.045454545454545456,\n",
       " 'shrews': 0.011363636363636364,\n",
       " 'raccoon': 0.011363636363636364,\n",
       " 'dogs': 0.011363636363636364,\n",
       " 'fruit': 0.011363636363636364,\n",
       " 'rabbits': 0.011363636363636364,\n",
       " 'susceptible': 0.011363636363636364,\n",
       " 'permissive': 0.011363636363636364,\n",
       " 'Some': 0.011363636363636364,\n",
       " 'institutions': 0.022727272727272728,\n",
       " 'advised': 0.011363636363636364,\n",
       " 'restrict': 0.011363636363636364,\n",
       " 'asymptomatic': 0.03409090909090909,\n",
       " 'cases': 0.07954545454545454,\n",
       " 'driver': 0.011363636363636364,\n",
       " 'One': 0.022727272727272728,\n",
       " 'meta': 0.022727272727272728,\n",
       " 'analysis': 0.045454545454545456,\n",
       " 'less': 0.022727272727272728,\n",
       " 'epidemiological': 0.011363636363636364,\n",
       " 'model': 0.022727272727272728,\n",
       " 'beginning': 0.011363636363636364,\n",
       " 'pre': 0.011363636363636364,\n",
       " 'symptomatic': 0.011363636363636364,\n",
       " 'typical': 0.011363636363636364,\n",
       " 'among': 0.056818181818181816,\n",
       " 'documented': 0.011363636363636364,\n",
       " 'subclinical': 0.011363636363636364,\n",
       " 'source': 0.056818181818181816,\n",
       " 'majority': 0.011363636363636364,\n",
       " 'That': 0.011363636363636364,\n",
       " 'explain': 0.011363636363636364,\n",
       " 'board': 0.011363636363636364,\n",
       " 'cruise': 0.022727272727272728,\n",
       " 'liner': 0.011363636363636364,\n",
       " 'docked': 0.011363636363636364,\n",
       " 'Montevideo': 0.011363636363636364,\n",
       " 'tested': 0.045454545454545456,\n",
       " 'showed': 0.056818181818181816,\n",
       " 'Similarly': 0.011363636363636364,\n",
       " 'ninety': 0.011363636363636364,\n",
       " 'patients': 0.022727272727272728,\n",
       " 'hospitalized': 0.011363636363636364,\n",
       " 'estimated': 0.06818181818181818,\n",
       " 'began': 0.022727272727272728,\n",
       " 'two': 0.07954545454545454,\n",
       " 'appear': 0.022727272727272728,\n",
       " 'substantial': 0.022727272727272728,\n",
       " 'proportion': 0.022727272727272728,\n",
       " 'probably': 0.011363636363636364,\n",
       " 'index': 0.011363636363636364,\n",
       " 'case': 0.056818181818181816,\n",
       " 'authors': 0.022727272727272728,\n",
       " 'later': 0.022727272727272728,\n",
       " 'published': 0.045454545454545456,\n",
       " 'correction': 0.011363636363636364,\n",
       " 'earlier': 0.022727272727272728,\n",
       " 'five': 0.056818181818181816,\n",
       " 'There': 0.03409090909090909,\n",
       " 'uncertainty': 0.011363636363636364,\n",
       " 'reinfection': 0.056818181818181816,\n",
       " 'long': 0.03409090909090909,\n",
       " 'term': 0.011363636363636364,\n",
       " 'immunity': 0.022727272727272728,\n",
       " 'common': 0.045454545454545456,\n",
       " 'reports': 0.011363636363636364,\n",
       " 'occurring': 0.022727272727272728,\n",
       " 'variable': 0.011363636363636364,\n",
       " 'severity': 0.011363636363636364,\n",
       " 'reported': 0.03409090909090909,\n",
       " 'year': 0.03409090909090909,\n",
       " 'old': 0.022727272727272728,\n",
       " 'man': 0.03409090909090909,\n",
       " 'Hong': 0.011363636363636364,\n",
       " 'Kong': 0.011363636363636364,\n",
       " 'discharged': 0.011363636363636364,\n",
       " 'April': 0.022727272727272728,\n",
       " 'negative': 0.022727272727272728,\n",
       " 'tests': 0.03409090909090909,\n",
       " 'August': 0.03409090909090909,\n",
       " 'whole': 0.022727272727272728,\n",
       " 'genome': 0.13636363636363635,\n",
       " 'sequencing': 0.011363636363636364,\n",
       " 'showing': 0.011363636363636364,\n",
       " 'genomes': 0.09090909090909091,\n",
       " 'episodes': 0.011363636363636364,\n",
       " 'belong': 0.011363636363636364,\n",
       " 'clades': 0.03409090909090909,\n",
       " 'findings': 0.011363636363636364,\n",
       " 'herd': 0.011363636363636364,\n",
       " 'eliminate': 0.011363636363636364,\n",
       " 'uncommon': 0.011363636363636364,\n",
       " 'occurrence': 0.011363636363636364,\n",
       " 'vaccines': 0.011363636363636364,\n",
       " 'provide': 0.022727272727272728,\n",
       " 'lifelong': 0.011363636363636364,\n",
       " 'protection': 0.011363636363636364,\n",
       " 'Another': 0.011363636363636364,\n",
       " 'Nevada': 0.011363636363636364,\n",
       " 'June': 0.011363636363636364,\n",
       " 'separated': 0.022727272727272728,\n",
       " 'Since': 0.011363636363636364,\n",
       " 'genomic': 0.022727272727272728,\n",
       " 'analyses': 0.011363636363636364,\n",
       " 'significant': 0.011363636363636364,\n",
       " 'differences': 0.022727272727272728,\n",
       " 'variant': 0.045454545454545456,\n",
       " 'sampled': 0.022727272727272728,\n",
       " 'dates': 0.011363636363636364,\n",
       " 'determined': 0.011363636363636364,\n",
       " 'second': 0.011363636363636364,\n",
       " 'symptomatically': 0.011363636363636364,\n",
       " 'mechanisms': 0.011363636363636364,\n",
       " 'could': 0.022727272727272728,\n",
       " 'account': 0.011363636363636364,\n",
       " 'No': 0.011363636363636364,\n",
       " 'natural': 0.03409090909090909,\n",
       " 'reservoir': 0.03409090909090909,\n",
       " 'Prior': 0.011363636363636364,\n",
       " 'emergence': 0.011363636363636364,\n",
       " 'pathogen': 0.011363636363636364,\n",
       " 'infecting': 0.011363636363636364,\n",
       " 'previous': 0.011363636363636364,\n",
       " 'zoonosis': 0.011363636363636364,\n",
       " 'based': 0.03409090909090909,\n",
       " 'epidemics': 0.011363636363636364,\n",
       " 'MERS': 0.03409090909090909,\n",
       " 'discovered': 0.022727272727272728,\n",
       " 'original': 0.03409090909090909,\n",
       " 'remains': 0.045454545454545456,\n",
       " 'unclear': 0.022727272727272728,\n",
       " 'became': 0.011363636363636364,\n",
       " 'pathogenic': 0.011363636363636364,\n",
       " 'Because': 0.011363636363636364,\n",
       " 'many': 0.03409090909090909,\n",
       " 'early': 0.045454545454545456,\n",
       " 'infectees': 0.011363636363636364,\n",
       " 'workers': 0.011363636363636364,\n",
       " 'Huanan': 0.03409090909090909,\n",
       " 'Seafood': 0.022727272727272728,\n",
       " 'Market': 0.03409090909090909,\n",
       " 'might': 0.011363636363636364,\n",
       " 'originated': 0.011363636363636364,\n",
       " 'market': 0.022727272727272728,\n",
       " 'visitors': 0.011363636363636364,\n",
       " 'introduced': 0.011363636363636364,\n",
       " 'facilitated': 0.011363636363636364,\n",
       " 'rapid': 0.022727272727272728,\n",
       " 'expansion': 0.011363636363636364,\n",
       " 'convened': 0.011363636363636364,\n",
       " 'report': 0.011363636363636364,\n",
       " 'stated': 0.011363636363636364,\n",
       " 'host': 0.13636363636363635,\n",
       " 'explanation': 0.022727272727272728,\n",
       " 'direct': 0.022727272727272728,\n",
       " 'next': 0.011363636363636364,\n",
       " 'Introduction': 0.011363636363636364,\n",
       " 'food': 0.011363636363636364,\n",
       " 'supply': 0.011363636363636364,\n",
       " 'chain': 0.011363636363636364,\n",
       " 'considered': 0.022727272727272728,\n",
       " 'An': 0.022727272727272728,\n",
       " 'November': 0.03409090909090909,\n",
       " 'however': 0.011363636363636364,\n",
       " 'said': 0.011363636363636364,\n",
       " 'earliest': 0.011363636363636364,\n",
       " 'misidentified': 0.011363636363636364,\n",
       " 'preponderance': 0.011363636363636364,\n",
       " 'linked': 0.022727272727272728,\n",
       " 'argued': 0.011363636363636364,\n",
       " 'For': 0.045454545454545456,\n",
       " 'recently': 0.022727272727272728,\n",
       " 'acquired': 0.011363636363636364,\n",
       " 'cross': 0.011363636363636364,\n",
       " 'evolution': 0.022727272727272728,\n",
       " 'expected': 0.022727272727272728,\n",
       " 'mutation': 0.056818181818181816,\n",
       " 'rate': 0.03409090909090909,\n",
       " 'per': 0.022727272727272728,\n",
       " 'Coronaviruses': 0.03409090909090909,\n",
       " 'general': 0.011363636363636364,\n",
       " 'plasticity': 0.011363636363636364,\n",
       " 'slowed': 0.011363636363636364,\n",
       " 'proofreading': 0.011363636363636364,\n",
       " 'capability': 0.011363636363636364,\n",
       " 'replication': 0.045454545454545456,\n",
       " 'machinery': 0.011363636363636364,\n",
       " 'comparison': 0.011363636363636364,\n",
       " 'vivo': 0.011363636363636364,\n",
       " 'lower': 0.022727272727272728,\n",
       " 'influenza': 0.011363636363636364,\n",
       " 'discovery': 0.011363636363636364,\n",
       " 'like': 0.03409090909090909,\n",
       " 'originating': 0.011363636363636364,\n",
       " 'horseshoe': 0.011363636363636364,\n",
       " 'match': 0.011363636363636364,\n",
       " 'Nature': 0.011363636363636364,\n",
       " 'journal': 0.011363636363636364,\n",
       " 'resemblance': 0.022727272727272728,\n",
       " 'collected': 0.03409090909090909,\n",
       " 'Feuang': 0.011363636363636364,\n",
       " 'Laos': 0.022727272727272728,\n",
       " 'Mojiang': 0.022727272727272728,\n",
       " 'Yunnan': 0.045454545454545456,\n",
       " 'None': 0.011363636363636364,\n",
       " 'ancestor': 0.022727272727272728,\n",
       " 'Bats': 0.011363636363636364,\n",
       " 'Differences': 0.011363636363636364,\n",
       " 'although': 0.011363636363636364,\n",
       " 'introduction': 0.011363636363636364,\n",
       " 'unknown': 0.022727272727272728,\n",
       " 'Although': 0.022727272727272728,\n",
       " 'role': 0.011363636363636364,\n",
       " 'pangolins': 0.045454545454545456,\n",
       " 'posited': 0.011363636363636364,\n",
       " 'July': 0.022727272727272728,\n",
       " 'substantiated': 0.011363636363636364,\n",
       " 'contribution': 0.011363636363636364,\n",
       " 'Evidence': 0.011363636363636364,\n",
       " 'hypothesis': 0.011363636363636364,\n",
       " 'includes': 0.011363636363636364,\n",
       " 'fact': 0.011363636363636364,\n",
       " 'pangolin': 0.022727272727272728,\n",
       " 'distant': 0.011363636363636364,\n",
       " 'isolates': 0.022727272727272728,\n",
       " 'obtained': 0.011363636363636364,\n",
       " 'seized': 0.011363636363636364,\n",
       " 'Guangdong': 0.011363636363636364,\n",
       " 'identical': 0.011363636363636364,\n",
       " 'sequence': 0.07954545454545454,\n",
       " 'matches': 0.011363636363636364,\n",
       " 'percent': 0.011363636363636364,\n",
       " 'sound': 0.011363636363636364,\n",
       " 'terms': 0.011363636363636364,\n",
       " 'wide': 0.011363636363636364,\n",
       " 'evolutionary': 0.011363636363636364,\n",
       " 'gap': 0.011363636363636364,\n",
       " 'addition': 0.011363636363636364,\n",
       " 'despite': 0.011363636363636364,\n",
       " 'similarities': 0.022727272727272728,\n",
       " 'critical': 0.022727272727272728,\n",
       " 'amino': 0.03409090909090909,\n",
       " 'acids': 0.022727272727272728,\n",
       " 'exhibit': 0.011363636363636364,\n",
       " 'poor': 0.011363636363636364,\n",
       " 'receptor': 0.045454545454545456,\n",
       " 'belongs': 0.011363636363636364,\n",
       " 'broad': 0.011363636363636364,\n",
       " 'family': 0.022727272727272728,\n",
       " 'ssRNA': 0.011363636363636364,\n",
       " 'linear': 0.022727272727272728,\n",
       " 'segment': 0.011363636363636364,\n",
       " 'infect': 0.03409090909090909,\n",
       " 'mammals': 0.011363636363636364,\n",
       " 'including': 0.011363636363636364,\n",
       " 'livestock': 0.011363636363636364,\n",
       " 'companion': 0.011363636363636364,\n",
       " 'avian': 0.011363636363636364,\n",
       " 'capable': 0.011363636363636364,\n",
       " 'causing': 0.011363636363636364,\n",
       " 'illnesses': 0.011363636363636364,\n",
       " 'ranging': 0.011363636363636364,\n",
       " 'cold': 0.011363636363636364,\n",
       " 'diseases': 0.011363636363636364,\n",
       " 'Middle': 0.011363636363636364,\n",
       " 'East': 0.011363636363636364,\n",
       " 'fatality': 0.011363636363636364,\n",
       " 'seventh': 0.011363636363636364,\n",
       " 'E': 0.03409090909090909,\n",
       " 'NL': 0.011363636363636364,\n",
       " 'OC': 0.011363636363636364,\n",
       " 'HKU': 0.011363636363636364,\n",
       " 'Like': 0.022727272727272728,\n",
       " 'implicated': 0.011363636363636364,\n",
       " 'member': 0.011363636363636364,\n",
       " 'subgenus': 0.022727272727272728,\n",
       " 'Sarbecovirus': 0.011363636363636364,\n",
       " 'beta': 0.011363636363636364,\n",
       " 'lineage': 0.011363636363636364,\n",
       " 'B': 0.03409090909090909,\n",
       " 'undergo': 0.011363636363636364,\n",
       " 'frequent': 0.011363636363636364,\n",
       " 'recombination': 0.022727272727272728,\n",
       " 'mechanism': 0.03409090909090909,\n",
       " 'unsegmented': 0.011363636363636364,\n",
       " 'copy': 0.011363636363636364,\n",
       " 'choice': 0.011363636363636364,\n",
       " 'gene': 0.022727272727272728,\n",
       " 'material': 0.011363636363636364,\n",
       " 'switches': 0.011363636363636364,\n",
       " 'template': 0.011363636363636364,\n",
       " 'molecule': 0.011363636363636364,\n",
       " 'bases': 0.022727272727272728,\n",
       " 'length': 0.011363636363636364,\n",
       " 'relatively': 0.022727272727272728,\n",
       " 'turn': 0.011363636363636364,\n",
       " 'carry': 0.011363636363636364,\n",
       " 'largest': 0.011363636363636364,\n",
       " 'families': 0.011363636363636364,\n",
       " 'Its': 0.022727272727272728,\n",
       " 'consists': 0.011363636363636364,\n",
       " 'nearly': 0.011363636363636364,\n",
       " 'entirely': 0.011363636363636364,\n",
       " 'coding': 0.011363636363636364,\n",
       " 'sequences': 0.03409090909090909,\n",
       " 'trait': 0.011363636363636364,\n",
       " 'shared': 0.011363636363636364,\n",
       " 'distinguishing': 0.011363636363636364,\n",
       " 'feature': 0.022727272727272728,\n",
       " 'incorporation': 0.011363636363636364,\n",
       " 'polybasic': 0.011363636363636364,\n",
       " 'cleaved': 0.011363636363636364,\n",
       " 'furin': 0.045454545454545456,\n",
       " 'appears': 0.022727272727272728,\n",
       " 'important': 0.011363636363636364,\n",
       " 'element': 0.011363636363636364,\n",
       " 'enhancing': 0.011363636363636364,\n",
       " 'virulence': 0.022727272727272728,\n",
       " 'acquisition': 0.011363636363636364,\n",
       " 'cleavage': 0.045454545454545456,\n",
       " 'S': 0.14772727272727273,\n",
       " 'essential': 0.022727272727272728,\n",
       " 'transfer': 0.011363636363636364,\n",
       " 'protease': 0.045454545454545456,\n",
       " 'recognizes': 0.011363636363636364,\n",
       " 'canonical': 0.011363636363636364,\n",
       " 'peptide': 0.022727272727272728,\n",
       " 'RX': 0.011363636363636364,\n",
       " 'R': 0.10227272727272728,\n",
       " 'K': 0.011363636363636364,\n",
       " 'X': 0.022727272727272728,\n",
       " 'arrow': 0.011363636363636364,\n",
       " 'acid': 0.022727272727272728,\n",
       " 'recognition': 0.011363636363636364,\n",
       " 'formed': 0.011363636363636364,\n",
       " 'incorporated': 0.011363636363636364,\n",
       " 'codon': 0.045454545454545456,\n",
       " 'nucleotide': 0.022727272727272728,\n",
       " 'CCT': 0.011363636363636364,\n",
       " 'CGG': 0.03409090909090909,\n",
       " 'GCA': 0.011363636363636364,\n",
       " 'corresponds': 0.011363636363636364,\n",
       " 'P': 0.022727272727272728,\n",
       " 'RR': 0.022727272727272728,\n",
       " 'upstream': 0.011363636363636364,\n",
       " 'arginine': 0.022727272727272728,\n",
       " 'serine': 0.022727272727272728,\n",
       " 'spike': 0.07954545454545454,\n",
       " 'sites': 0.011363636363636364,\n",
       " 'naturally': 0.011363636363636364,\n",
       " 'Subfamily': 0.011363636363636364,\n",
       " 'Orthocoronavirinae': 0.011363636363636364,\n",
       " 'Beta': 0.011363636363636364,\n",
       " 'genus': 0.011363636363636364,\n",
       " 'unique': 0.011363636363636364,\n",
       " 'PRRAR': 0.011363636363636364,\n",
       " 'highly': 0.022727272727272728,\n",
       " 'similar': 0.045454545454545456,\n",
       " 'feline': 0.011363636363636364,\n",
       " 'alphacoronavirus': 0.011363636363636364,\n",
       " 'data': 0.011363636363636364,\n",
       " 'information': 0.011363636363636364,\n",
       " 'time': 0.022727272727272728,\n",
       " 'space': 0.011363636363636364,\n",
       " 'epidemiologically': 0.011363636363636364,\n",
       " 'With': 0.011363636363636364,\n",
       " 'sufficient': 0.022727272727272728,\n",
       " 'number': 0.06818181818181818,\n",
       " 'sequenced': 0.022727272727272728,\n",
       " 'reconstruct': 0.011363636363636364,\n",
       " 'phylogenetic': 0.045454545454545456,\n",
       " 'history': 0.011363636363636364,\n",
       " 'By': 0.022727272727272728,\n",
       " 'isolated': 0.011363636363636364,\n",
       " 'Chinese': 0.022727272727272728,\n",
       " 'Center': 0.011363636363636364,\n",
       " 'Disease': 0.011363636363636364,\n",
       " 'Control': 0.011363636363636364,\n",
       " 'Prevention': 0.011363636363636364,\n",
       " 'CCDC': 0.011363636363636364,\n",
       " 'increased': 0.011363636363636364,\n",
       " 'seven': 0.022727272727272728,\n",
       " 'mutations': 0.011363636363636364,\n",
       " 'relative': 0.022727272727272728,\n",
       " 'implying': 0.011363636363636364,\n",
       " 'Examination': 0.011363636363636364,\n",
       " 'topology': 0.011363636363636364,\n",
       " 'start': 0.022727272727272728,\n",
       " 'As': 0.045454545454545456,\n",
       " 'update': 0.022727272727272728,\n",
       " 'belonging': 0.011363636363636364,\n",
       " 'strains': 0.011363636363636364,\n",
       " 'continents': 0.011363636363636364,\n",
       " 'except': 0.011363636363636364,\n",
       " 'Antarctica': 0.011363636363636364,\n",
       " 'publicly': 0.011363636363636364,\n",
       " 'available': 0.011363636363636364,\n",
       " 'announced': 0.011363636363636364,\n",
       " 'according': 0.011363636363636364,\n",
       " 'existing': 0.011363636363636364,\n",
       " 'rules': 0.011363636363636364,\n",
       " 'compute': 0.011363636363636364,\n",
       " 'hierarchical': 0.011363636363636364,\n",
       " 'relationships': 0.011363636363636364,\n",
       " 'conserved': 0.011363636363636364,\n",
       " 'nucleic': 0.011363636363636364,\n",
       " 'insufficient': 0.011363636363636364,\n",
       " 'make': 0.011363636363636364,\n",
       " 'separate': 0.011363636363636364,\n",
       " 'Therefore': 0.011363636363636364,\n",
       " 'scientists': 0.011363636363636364,\n",
       " 'G': 0.045454545454545456,\n",
       " 'replaced': 0.011363636363636364,\n",
       " 'D': 0.011363636363636364,\n",
       " 'form': 0.011363636363636364,\n",
       " 'Coronavirus': 0.022727272727272728,\n",
       " 'subgenomes': 0.011363636363636364,\n",
       " 'encode': 0.011363636363636364,\n",
       " 'six': 0.022727272727272728,\n",
       " 'open': 0.03409090909090909,\n",
       " 'reading': 0.011363636363636364,\n",
       " 'frames': 0.011363636363636364,\n",
       " 'ORFs': 0.011363636363636364,\n",
       " 'October': 0.011363636363636364,\n",
       " 'overlapping': 0.011363636363636364,\n",
       " 'named': 0.011363636363636364,\n",
       " 'ORF': 0.03409090909090909,\n",
       " 'function': 0.011363636363636364,\n",
       " 'provokes': 0.011363636363636364,\n",
       " 'strong': 0.011363636363636364,\n",
       " 'response': 0.022727272727272728,\n",
       " 'infects': 0.011363636363636364,\n",
       " 'Bat': 0.11363636363636363,\n",
       " 'Rc': 0.011363636363636364,\n",
       " 'Rhinolophus': 0.11363636363636363,\n",
       " 'cornutus': 0.011363636363636364,\n",
       " 'Iwate': 0.011363636363636364,\n",
       " 'Japan': 0.011363636363636364,\n",
       " 'SL': 0.022727272727272728,\n",
       " 'ZXC': 0.011363636363636364,\n",
       " 'pusillus': 0.03409090909090909,\n",
       " 'Zhoushan': 0.022727272727272728,\n",
       " 'Zhejiang': 0.022727272727272728,\n",
       " 'ZC': 0.011363636363636364,\n",
       " 'Pangolin': 0.022727272727272728,\n",
       " 'GX': 0.011363636363636364,\n",
       " 'Manis': 0.022727272727272728,\n",
       " 'javanica': 0.022727272727272728,\n",
       " 'smuggled': 0.022727272727272728,\n",
       " 'Southeast': 0.022727272727272728,\n",
       " 'Asia': 0.022727272727272728,\n",
       " 'GD': 0.011363636363636364,\n",
       " 'RshSTT': 0.022727272727272728,\n",
       " 'shameli': 0.022727272727272728,\n",
       " 'Steung': 0.022727272727272728,\n",
       " 'Treng': 0.022727272727272728,\n",
       " 'Cambodia': 0.022727272727272728,\n",
       " 'RacCS': 0.011363636363636364,\n",
       " 'acuminatus': 0.011363636363636364,\n",
       " 'Chachoengsao': 0.011363636363636364,\n",
       " 'Thailand': 0.011363636363636364,\n",
       " 'RmYN': 0.011363636363636364,\n",
       " 'malayanus': 0.022727272727272728,\n",
       " 'Mengla': 0.011363636363636364,\n",
       " 'RpYN': 0.011363636363636364,\n",
       " 'Xishuangbanna': 0.011363636363636364,\n",
       " 'affinis': 0.011363636363636364,\n",
       " 'Vientiane': 0.011363636363636364,\n",
       " 'thousands': 0.011363636363636364,\n",
       " 'grouped': 0.011363636363636364,\n",
       " 'much': 0.011363636363636364,\n",
       " 'larger': 0.011363636363636364,\n",
       " 'Several': 0.022727272727272728,\n",
       " 'clade': 0.011363636363636364,\n",
       " 'nomenclatures': 0.011363636363636364,\n",
       " 'proposed': 0.011363636363636364,\n",
       " 'Nextstrain': 0.011363636363636364,\n",
       " 'divides': 0.022727272727272728,\n",
       " 'C': 0.03409090909090909,\n",
       " 'GISAID': 0.011363636363636364,\n",
       " 'L': 0.011363636363636364,\n",
       " 'O': 0.011363636363636364,\n",
       " 'V': 0.011363636363636364,\n",
       " 'GH': 0.011363636363636364,\n",
       " 'GR': 0.011363636363636364,\n",
       " 'notable': 0.022727272727272728,\n",
       " 'currently': 0.011363636363636364,\n",
       " 'follows': 0.011363636363636364,\n",
       " 'include': 0.022727272727272728,\n",
       " 'designated': 0.011363636363636364,\n",
       " 'investigation': 0.022727272727272728,\n",
       " 'Cluster': 0.011363636363636364,\n",
       " 'mink': 0.022727272727272728,\n",
       " 'Denmark': 0.011363636363636364,\n",
       " 'euthanasia': 0.011363636363636364,\n",
       " 'campaign': 0.011363636363636364,\n",
       " 'rendering': 0.011363636363636364,\n",
       " 'virtually': 0.011363636363636364,\n",
       " 'extinct': 0.011363636363636364,\n",
       " 'Each': 0.011363636363636364,\n",
       " 'virion': 0.056818181818181816,\n",
       " 'nanometres': 0.011363636363636364,\n",
       " 'diameter': 0.011363636363636364,\n",
       " 'mass': 0.011363636363636364,\n",
       " 'global': 0.011363636363636364,\n",
       " 'populace': 0.011363636363636364,\n",
       " 'kilograms': 0.011363636363636364,\n",
       " 'structural': 0.011363636363636364,\n",
       " 'proteins': 0.056818181818181816,\n",
       " 'envelope': 0.022727272727272728,\n",
       " 'M': 0.022727272727272728,\n",
       " 'N': 0.022727272727272728,\n",
       " 'nucleocapsid': 0.011363636363636364,\n",
       " 'holds': 0.011363636363636364,\n",
       " 'together': 0.011363636363636364,\n",
       " 'create': 0.011363636363636364,\n",
       " 'glycoproteins': 0.011363636363636364,\n",
       " 'I': 0.011363636363636364,\n",
       " 'membranes': 0.011363636363636364,\n",
       " 'containing': 0.011363636363636364,\n",
       " 'transmembrane': 0.022727272727272728,\n",
       " 'domain': 0.011363636363636364,\n",
       " 'oriented': 0.011363636363636364,\n",
       " 'extracellular': 0.011363636363636364,\n",
       " 'side': 0.011363636363636364,\n",
       " 'divided': 0.011363636363636364,\n",
       " 'functional': 0.011363636363636364,\n",
       " 'parts': 0.011363636363636364,\n",
       " 'imaged': 0.011363636363636364,\n",
       " 'atomic': 0.011363636363636364,\n",
       " 'level': 0.011363636363636364,\n",
       " 'cryogenic': 0.011363636363636364,\n",
       " 'electron': 0.011363636363636364,\n",
       " 'microscopy': 0.011363636363636364,\n",
       " 'allowing': 0.011363636363636364,\n",
       " 'attach': 0.011363636363636364,\n",
       " 'fuse': 0.011363636363636364,\n",
       " 'cell': 0.10227272727272728,\n",
       " 'specifically': 0.011363636363636364,\n",
       " 'subunit': 0.03409090909090909,\n",
       " 'catalyzes': 0.011363636363636364,\n",
       " 'attachment': 0.011363636363636364,\n",
       " 'fusion': 0.03409090909090909,\n",
       " 'million': 0.011363636363636364,\n",
       " 'deposited': 0.011363636363636364,\n",
       " 'databases': 0.011363636363636364,\n",
       " 'added': 0.011363636363636364,\n",
       " 'month': 0.011363636363636364,\n",
       " 'bias': 0.045454545454545456,\n",
       " 'cytosine': 0.022727272727272728,\n",
       " 'guanine': 0.022727272727272728,\n",
       " 'nucleotides': 0.011363636363636364,\n",
       " 'highest': 0.011363636363636364,\n",
       " 'composition': 0.022727272727272728,\n",
       " 'U': 0.011363636363636364,\n",
       " 'followed': 0.022727272727272728,\n",
       " 'arises': 0.011363636363636364,\n",
       " 'guanines': 0.011363636363636364,\n",
       " 'cytosines': 0.011363636363636364,\n",
       " 'adenosines': 0.011363636363636364,\n",
       " 'uracils': 0.011363636363636364,\n",
       " 'CG': 0.022727272727272728,\n",
       " 'dinucleotides': 0.022727272727272728,\n",
       " 'arise': 0.011363636363636364,\n",
       " 'zinc': 0.011363636363636364,\n",
       " 'finger': 0.011363636363636364,\n",
       " 'antiviral': 0.022727272727272728,\n",
       " 'defense': 0.011363636363636364,\n",
       " 'energy': 0.011363636363636364,\n",
       " 'unbind': 0.011363636363636364,\n",
       " 'translation': 0.011363636363636364,\n",
       " 'adenosine': 0.011363636363636364,\n",
       " 'uracil': 0.011363636363636364,\n",
       " 'base': 0.011363636363636364,\n",
       " 'pair': 0.011363636363636364,\n",
       " 'hydrogen': 0.011363636363636364,\n",
       " 'bonds': 0.011363636363636364,\n",
       " 'depletion': 0.011363636363636364,\n",
       " 'led': 0.011363636363636364,\n",
       " 'noticeable': 0.011363636363636364,\n",
       " 'usage': 0.03409090909090909,\n",
       " 'instance': 0.022727272727272728,\n",
       " 'codons': 0.011363636363636364,\n",
       " 'synonymous': 0.011363636363636364,\n",
       " 'AGA': 0.011363636363636364,\n",
       " 'CGU': 0.011363636363636364,\n",
       " 'AGG': 0.011363636363636364,\n",
       " 'CGC': 0.011363636363636364,\n",
       " 'CGA': 0.011363636363636364,\n",
       " 'trend': 0.011363636363636364,\n",
       " 'seen': 0.022727272727272728,\n",
       " 'Virus': 0.011363636363636364,\n",
       " 'particles': 0.011363636363636364,\n",
       " 'bind': 0.011363636363636364,\n",
       " 'surface': 0.011363636363636364,\n",
       " 'cellular': 0.011363636363636364,\n",
       " 'receptors': 0.011363636363636364,\n",
       " 'Protein': 0.011363636363636364,\n",
       " 'modeling': 0.011363636363636364,\n",
       " 'soon': 0.022727272727272728,\n",
       " 'affinity': 0.022727272727272728,\n",
       " 'use': 0.03409090909090909,\n",
       " 'entry': 0.045454545454545456,\n",
       " 'full': 0.022727272727272728,\n",
       " 'United': 0.03409090909090909,\n",
       " 'States': 0.022727272727272728,\n",
       " 'reverse': 0.011363636363636364,\n",
       " 'genetics': 0.011363636363636364,\n",
       " 'methods': 0.011363636363636364,\n",
       " 'independently': 0.011363636363636364,\n",
       " 'experimentally': 0.011363636363636364,\n",
       " 'demonstrated': 0.011363636363636364,\n",
       " 'act': 0.011363636363636364,\n",
       " 'shown': 0.011363636363636364,\n",
       " 'higher': 0.03409090909090909,\n",
       " 'basigin': 0.011363636363636364,\n",
       " 'assist': 0.011363636363636364,\n",
       " 'Initial': 0.011363636363636364,\n",
       " 'priming': 0.011363636363636364,\n",
       " 'TMPRSS': 0.022727272727272728,\n",
       " 'neuropilin': 0.011363636363636364,\n",
       " 'NRP': 0.011363636363636364,\n",
       " 'aid': 0.011363636363636364,\n",
       " 'After': 0.022727272727272728,\n",
       " 'attaches': 0.011363636363636364,\n",
       " 'target': 0.011363636363636364,\n",
       " 'cuts': 0.011363636363636364,\n",
       " 'exposing': 0.011363636363636364,\n",
       " 'endosome': 0.022727272727272728,\n",
       " 'around': 0.011363636363636364,\n",
       " 'separating': 0.011363636363636364,\n",
       " 'rest': 0.011363636363636364,\n",
       " 'escapes': 0.011363636363636364,\n",
       " 'pH': 0.011363636363636364,\n",
       " 'drops': 0.011363636363636364,\n",
       " 'cathepsin': 0.011363636363636364,\n",
       " 'cysteine': 0.011363636363636364,\n",
       " 'cleaves': 0.011363636363636364,\n",
       " 'releases': 0.011363636363636364,\n",
       " 'forces': 0.011363636363636364,\n",
       " 'produce': 0.011363636363636364,\n",
       " 'disseminate': 0.011363636363636364,\n",
       " 'copies': 0.011363636363636364,\n",
       " 'produces': 0.011363636363636364,\n",
       " 'least': 0.011363636363636364,\n",
       " 'factors': 0.011363636363636364,\n",
       " 'promote': 0.011363636363636364,\n",
       " 'inhibit': 0.03409090909090909,\n",
       " 'Whether': 0.011363636363636364,\n",
       " 'downregulation': 0.011363636363636364,\n",
       " 'May': 0.011363636363636364,\n",
       " 'Very': 0.011363636363636364,\n",
       " 'drugs': 0.011363636363636364,\n",
       " ...}"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Weighted Frequency of words\n",
    "\n",
    "for word in word_frequencies.keys():\n",
    "    word_frequencies[word] = (word_frequencies[word]/maximum_frequncy)\n",
    "\n",
    "word_frequencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "72fe2f76",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{' Severe acute respiratory syndrome coronavirus 2 (SARS‑CoV‑2) is a strain of coronavirus that causes COVID-19 (coronavirus disease 2019), the respiratory illness responsible for the ongoing COVID-19 pandemic.': 1.1590909090909092,\n",
       " 'The virus previously had a provisional name, 2019 novel coronavirus (2019-nCoV), and has also been called the human coronavirus 2019 (HCoV-19 or hCoV-19).': 1.25,\n",
       " 'SARS‑CoV‑2 is a positive-sense single-stranded RNA virus that is contagious in humans.': 0.625,\n",
       " 'SARS‑CoV‑2 is a virus of the species severe acute respiratory syndrome–related coronavirus (SARSr-CoV), related to the SARS-CoV-1 virus that caused the 2002–2004 SARS outbreak.': 1.681818181818182,\n",
       " 'Despite its close relation to SARS-CoV-1, its closest known relatives, with which it forms a sister group, are the derived SARS viruses BANAL-52 and RaTG13.': 0.3977272727272727,\n",
       " 'Available evidence indicates that it is most likely of zoonotic origin and has close genetic similarity to bat coronaviruses, suggesting it emerged from a bat-borne virus.': 1.0340909090909092,\n",
       " 'Research is ongoing as to whether SARS‑CoV‑2 came directly from bats or indirectly through any intermediate hosts.': 0.28409090909090906,\n",
       " 'The virus shows little genetic diversity, indicating that the spillover event introducing SARS‑CoV‑2 to humans is likely to have occurred in late 2019.': 0.9886363636363634,\n",
       " 'However, some subsequent variants have become more infectious.': 0.18181818181818182,\n",
       " 'It enters human cells by binding to angiotensin-converting enzyme 2 (ACE2), a membrane protein that regulates the renin–angiotensin system.': 0.5113636363636364,\n",
       " 'During the initial outbreak in Wuhan, China, various names were used for the virus; some names used by different sources included \"the coronavirus\" or \"Wuhan coronavirus\".': 1.2613636363636362,\n",
       " 'In January 2020, the World Health Organization (WHO) recommended \"2019 novel coronavirus\" (2019-nCoV) as the provisional name for the virus.': 0.8409090909090908,\n",
       " \"This was in accordance with WHO's 2015 guidance against using geographical locations, animal species, or groups of people in disease and virus names.\": 0.8295454545454545,\n",
       " 'On 11 February 2020, the International Committee on Taxonomy of Viruses adopted the official name \"severe acute respiratory syndrome coronavirus 2\" (SARS‑CoV‑2).': 0.6363636363636364,\n",
       " 'Human-to-human transmission of SARS‑CoV‑2 was confirmed on 20 January 2020 during the COVID-19 pandemic.': 0.25,\n",
       " 'Transmission was initially assumed to occur primarily via respiratory droplets from coughs and sneezes within a range of about 1.8 metres (6 ft).': 0.49999999999999994,\n",
       " 'Laser light scattering experiments suggest that speaking is an additional mode of transmission and a far-reaching one, indoors, with little air flow.': 0.3181818181818181,\n",
       " 'Other studies have suggested that the virus may be airborne as well, with aerosols potentially being able to transmit the virus.': 1.4318181818181817,\n",
       " 'During human-to-human transmission, between 200 and 800 infectious SARS‑CoV‑2 virions are thought to initiate a new infection.': 0.40909090909090906,\n",
       " 'Indirect contact via contaminated surfaces is another possible cause of infection.': 0.3977272727272727,\n",
       " 'The virus is inactivated by soap, which destabilizes its lipid bilayer.': 0.5681818181818181,\n",
       " 'Viral RNA has also been found in stool samples and semen from infected individuals.': 0.4204545454545454,\n",
       " 'The duration of SARS-CoV-2 RNA shedding is generally between 3 and 46 days after symptom onset.': 0.1818181818181818,\n",
       " 'Studies have identified a range of animals—such as cats, ferrets, hamsters, non-human primates, minks, tree shrews, raccoon dogs, fruit bats, and rabbits—that are susceptible and permissive to SARS-CoV-2 infection.': 0.534090909090909,\n",
       " 'Some institutions have advised that those infected with SARS‑CoV‑2 restrict their contact with animals.': 0.1590909090909091,\n",
       " 'On 1 February 2020, the World Health Organization (WHO) indicated that \"transmission from asymptomatic cases is likely not a major driver of transmission\".': 0.5227272727272727,\n",
       " 'One meta-analysis found that 17% of infections are asymptomatic, and asymptomatic individuals were 42% less likely to transmit the virus.': 1.034090909090909,\n",
       " 'That may explain how out of 217 on board a cruise liner that docked at Montevideo, only 24 of 128 who tested positive for viral RNA showed symptoms.': 0.5795454545454545,\n",
       " 'The authors later published a correction that showed that shedding began earlier than first estimated, four to five days before symptoms appear.': 0.6818181818181818,\n",
       " 'There is uncertainty about reinfection and long-term immunity.': 0.09090909090909091,\n",
       " 'It is not known how common reinfection is, but reports have indicated that it is occurring with variable severity.': 0.30681818181818177,\n",
       " 'Another case study described a 25-year-old man from Nevada who tested positive for SARS‑CoV‑2 on 18 April 2020 and on 5 June 2020 (separated by two negative tests).': 0.5227272727272727,\n",
       " 'Since genomic analyses showed significant genetic differences between the SARS‑CoV‑2 variant sampled on those two dates, the case study authors determined this was a reinfection.': 0.5681818181818181,\n",
       " \"The man's second infection was symptomatically more severe than the first infection, but the mechanisms that could account for this are not known.\": 0.6590909090909091,\n",
       " 'No natural reservoir for SARS-CoV-2 has been identified.': 0.13636363636363635,\n",
       " 'Prior to the emergence of SARS-CoV-2 as a pathogen infecting humans, there had been two previous zoonosis-based coronavirus epidemics, those caused by SARS-CoV-1 and MERS-CoV.': 0.4772727272727272,\n",
       " 'The first known infections from SARS‑CoV‑2 were discovered in Wuhan, China.': 0.32954545454545453,\n",
       " 'The original source of viral transmission to humans remains unclear, as does whether the virus became pathogenic before or after the spillover event.': 1.1250000000000002,\n",
       " 'Because many of the early infectees were workers at the Huanan Seafood Market, it has been suggested that the virus might have originated from the market.': 0.75,\n",
       " 'However, other research indicates that visitors may have introduced the virus to the market, which then facilitated rapid expansion of the infections.': 0.9659090909090908,\n",
       " 'A March 2021 WHO-convened report stated that human spillover via an intermediate animal host was the most likely explanation, with direct spillover from bats next most likely.': 0.9090909090909091,\n",
       " 'Introduction through the food supply chain and the Huanan Seafood Market was considered another possible, but less likely, explanation.': 0.32954545454545453,\n",
       " 'For a virus recently acquired through a cross-species transmission, rapid evolution is expected.': 0.7159090909090908,\n",
       " 'The mutation rate estimated from early cases of SARS-CoV-2 was of 6.54×10−4 per site per year.': 0.46590909090909083,\n",
       " \"Coronaviruses in general have high genetic plasticity, but SARS-CoV-2's viral evolution is slowed by the RNA proofreading capability of its replication machinery.\": 0.4886363636363636,\n",
       " 'For comparison, the viral mutation rate in vivo of SARS-CoV-2 has been found to be lower than that of influenza.': 0.37499999999999994,\n",
       " 'Research into the natural reservoir of the virus that caused the 2002–2004 SARS outbreak has resulted in the discovery of many SARS-like bat coronaviruses, most originating in horseshoe bats.': 1.0909090909090908,\n",
       " 'An earlier source published in February 2020 identified the virus RaTG13, collected in bats in Mojiang, Yunnan, China to be the closest to SARS‑CoV‑2, with 96.1% resemblance.': 0.8636363636363636,\n",
       " 'None of the above are its direct ancestor.': 0.045454545454545456,\n",
       " 'Bats are considered the most likely natural reservoir of SARS‑CoV‑2.': 0.26136363636363635,\n",
       " 'Differences between the bat coronavirus and SARS‑CoV‑2 suggest that humans may have been infected via an intermediate host; although the source of introduction into humans remains unknown.': 1.1249999999999998,\n",
       " 'In addition, despite similarities in a few critical amino acids, pangolin virus samples exhibit poor binding to the human ACE2 receptor.': 0.943181818181818,\n",
       " 'SARS‑CoV‑2 belongs to the broad family of viruses known as coronaviruses.': 0.38636363636363635,\n",
       " 'It is a positive-sense single-stranded RNA (+ssRNA) virus, with a single linear RNA segment.': 0.6022727272727272,\n",
       " 'Coronaviruses infect humans, other mammals, including livestock and companion animals, and avian species.': 0.4318181818181817,\n",
       " 'Human coronaviruses are capable of causing illnesses ranging from the common cold to more severe diseases such as Middle East respiratory syndrome (MERS, fatality rate ~34%).': 0.6363636363636362,\n",
       " 'SARS-CoV-2 is the seventh known coronavirus to infect people, after 229E, NL63, OC43, HKU1, MERS-CoV, and the original SARS-CoV.': 0.4318181818181819,\n",
       " 'Like the SARS-related coronavirus implicated in the 2003 SARS outbreak, SARS‑CoV‑2 is a member of the subgenus Sarbecovirus (beta-CoV lineage B).': 0.37499999999999994,\n",
       " 'Coronaviruses undergo frequent recombination.': 0.1818181818181818,\n",
       " 'The SARS-CoV-2 RNA sequence is approximately 30,000 bases in length, relatively long for a coronavirus—which in turn carry the largest genomes among all RNA families.': 0.3863636363636363,\n",
       " 'Its genome consists nearly entirely of protein-coding sequences, a trait shared with other coronaviruses.': 0.36363636363636354,\n",
       " 'A distinguishing feature of SARS‑CoV‑2 is its incorporation of a polybasic site cleaved by furin, which appears to be an important element enhancing its virulence.': 0.29545454545454547,\n",
       " 'It was suggested that the acquisition of the furin-cleavage site in the SARS-CoV-2 S protein was essential for zoonotic transfer to humans.': 0.5,\n",
       " 'The furin protease recognizes the canonical peptide sequence RX[R/K] R↓X where the cleavage site is indicated by a down arrow and X is any amino acid.': 0.4772727272727273,\n",
       " 'In SARS-CoV-2 the recognition site is formed by the incorporated 12 codon nucleotide sequence CCT CGG CGG GCA which corresponds to the amino acid sequence P RR A.': 0.4318181818181818,\n",
       " 'This sequence is upstream of an arginine and serine which forms the S1/S2 cleavage site (P RR A R↓S) of the spike protein.': 0.5568181818181819,\n",
       " 'The furin cleavage site PRRAR↓ is highly similar to that of the feline coronavirus, an alphacoronavirus 1 strain.': 0.5113636363636364,\n",
       " 'Viral genetic sequence data can provide critical information about whether viruses separated by time and space are likely to be epidemiologically linked.': 0.6477272727272726,\n",
       " 'With a sufficient number of sequenced genomes, it is possible to reconstruct a phylogenetic tree of the mutation history of a family of viruses.': 0.5454545454545454,\n",
       " 'Examination of the topology of the phylogenetic tree at the start of the pandemic also found high similarities between human isolates.': 0.6022727272727273,\n",
       " 'As of 21 August 2021,[update] 3,422 SARS‑CoV‑2 genomes, belonging to 19 strains, sampled on all continents except Antarctica were publicly available.': 0.2045454545454545,\n",
       " 'Therefore, they identified 2019-nCoV as a virus of Severe acute respiratory syndrome–related coronavirus.': 0.9659090909090909,\n",
       " 'In July 2020, scientists reported that a more infectious SARS‑CoV‑2 variant with spike protein variant G614 has replaced D614 as the dominant form in the pandemic.': 0.5340909090909091,\n",
       " 'Coronavirus genomes and subgenomes encode six open reading frames (ORFs).': 0.3977272727272727,\n",
       " 'In October 2020, researchers discovered a possible overlapping gene named ORF3d, in the SARS‑CoV‑2 genome.': 0.2727272727272727,\n",
       " 'It is unknown if the protein produced by ORF3d has any function, but it provokes a strong immune response.': 0.30681818181818177,\n",
       " 'ORF3d has been identified before, in a variant of coronavirus that infects pangolins.': 0.375,\n",
       " 'Several different clade nomenclatures have been proposed.': 0.10227272727272728,\n",
       " 'Nextstrain divides the variants into five clades (19A, 19B, 20A, 20B, and 20C), while GISAID divides them into seven (L, O, V, S, G, GH, and GR).': 0.23863636363636365,\n",
       " 'Several notable variants of SARS-CoV-2 emerged in late 2020.': 0.1590909090909091,\n",
       " 'Each SARS-CoV-2 virion is 60–140 nanometres (2.4×10−6–5.5×10−6 in) in diameter; its mass within the global human populace has been estimated as being between 0.1 and 10 kilograms.': 0.3863636363636363,\n",
       " 'Coronavirus S proteins are glycoproteins and also type I membrane proteins (membranes containing a single transmembrane domain oriented on the extracellular side).': 0.6022727272727272,\n",
       " 'They are divided into two functional parts (S1 and S2).': 0.11363636363636365,\n",
       " 'As of early 2022, about 7 million SARS-CoV-2 genomes had been sequenced and deposited into public databases and another 800,000 or so were added each month.': 0.2954545454545454,\n",
       " 'SARS-CoV-2 has a linear, positive-sense, single-stranded RNA genome about 30,000 bases long.': 0.2159090909090909,\n",
       " 'Its genome has a bias against cytosine (C) and guanine (G) nucleotides, like other coronaviruses.': 0.40909090909090906,\n",
       " 'The genome has the highest composition of U (32.2%), followed by A (29.9%), and a similar composition of G (19.6%) and C (18.3%).': 0.26136363636363635,\n",
       " 'The nucleotide bias arises from the mutation of guanines and cytosines to adenosines and uracils, respectively.': 0.21590909090909086,\n",
       " 'The depletion of CG dinucleotides in its genome has led the virus to have a noticeable codon usage bias.': 0.8295454545454544,\n",
       " \"For instance, arginine's six different codons have a relative synonymous codon usage of AGA (2.67), CGU (1.46), AGG (.81), CGC (.58), CGA (.29), and CGG (.19).\": 0.26136363636363635,\n",
       " 'A similar codon usage bias trend is seen in other SARS–related coronaviruses.': 0.34090909090909094,\n",
       " 'Virus infections start when viral particles bind to host surface cellular receptors.': 0.9545454545454545,\n",
       " 'Studies have shown that SARS‑CoV‑2 has a higher affinity to human ACE2 than the original SARS virus.': 0.8068181818181818,\n",
       " 'SARS‑CoV‑2 may also use basigin to assist in cell entry.': 0.42045454545454547,\n",
       " 'Initial spike protein priming by transmembrane protease, serine 2 (TMPRSS2) is essential for entry of SARS‑CoV‑2.': 0.43181818181818177,\n",
       " 'The host protein neuropilin 1 (NRP1) may aid the virus in host cell entry using ACE2.': 1.3181818181818181,\n",
       " 'After fusion, an endosome forms around the virion, separating it from the rest of the host cell.': 0.4204545454545454,\n",
       " 'The virion escapes when the pH of the endosome drops or when cathepsin, a host cysteine protease, cleaves it.': 0.3181818181818182,\n",
       " 'The virion then releases RNA into the cell and forces the cell to produce and disseminate copies of the virus, which infect more cells.': 0.9318181818181819,\n",
       " 'SARS‑CoV‑2 produces at least three virulence factors that promote shedding of new virions from host cells and inhibit immune response.': 0.5568181818181818,\n",
       " 'Whether they include downregulation of ACE2, as seen in similar coronaviruses, remains under investigation (as of May 2020).': 0.5,\n",
       " 'Very few drugs are known to effectively inhibit SARS‑CoV‑2.': 0.1590909090909091,\n",
       " 'However, it is not approved for the treatment of COVID-19 in humans as of August 2021.': 0.1590909090909091,\n",
       " 'One study found that Nirmatrelvir/ritonavir reduced the risk of hospitalization and death by 88%.': 0.30681818181818177,\n",
       " 'COVID Moonshot is an international collaborative open-science project started in March 2020 with the goal of developing an un-patented oral antiviral drug for treatment of SARS-CoV-2.': 0.17045454545454547,\n",
       " 'Retrospective tests collected within the Chinese surveillance system revealed no clear indication of substantial unrecognized circulation of SARS‑CoV‑2 in Wuhan during the latter part of 2019.': 0.24999999999999994,\n",
       " 'A meta-analysis from November 2020 estimated the basic reproduction number ( R 0 {\\\\displaystyle R_{0}} ) of the virus to be between 2.39 and 3.44.': 0.6818181818181818,\n",
       " 'The reproduction number may be higher in densely populated conditions such as those found on cruise ships.': 0.46590909090909083,\n",
       " 'Human behavior affects the R0 value and hence estimates of R0 differ between different countries, cultures, and social norms.': 0.34090909090909083,\n",
       " 'For instance, one study found relatively low R0 (~3.5) in Sweden, Belgium and the Netherlands, while Spain and the US had significantly higher R0 values (5.9 to 6.4, respectively).': 0.40909090909090895,\n",
       " 'There have been about 96,000 confirmed cases of infection in mainland China.': 0.32954545454545453,\n",
       " 'Before 24 February 2020, over 95% of all deaths from COVID-19 worldwide had occurred in Hubei province, where Wuhan is located.': 0.11363636363636365,\n",
       " 'As of 3 March 2023, the percentage had decreased to 0.047%.': 0.022727272727272728,\n",
       " 'As of 3 March 2023, there have been 675,816,641 total confirmed cases of SARS‑CoV‑2 infection in the ongoing pandemic.': 0.4431818181818182,\n",
       " 'The total number of deaths attributed to the virus is 6,876,102.': 0.6363636363636364}"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Finding sentence score by adding the scores of all words present in that sentence\n",
    "\n",
    "sentence_scores = {}\n",
    "for sent in sentence_list:\n",
    "    for word in nltk.word_tokenize(sent.lower()):\n",
    "        if word in word_frequencies.keys():\n",
    "            if len(sent.split(' ')) < 30:\n",
    "                if sent not in sentence_scores.keys():\n",
    "                    sentence_scores[sent] = word_frequencies[word]\n",
    "                else:\n",
    "                    sentence_scores[sent] += word_frequencies[word]\n",
    "sentence_scores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "a91e0b29",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SARS‑CoV‑2 is a virus of the species severe acute respiratory syndrome–related coronavirus (SARSr-CoV), related to the SARS-CoV-1 virus that caused the 2002–2004 SARS outbreak. Other studies have suggested that the virus may be airborne as well, with aerosols potentially being able to transmit the virus. The host protein neuropilin 1 (NRP1) may aid the virus in host cell entry using ACE2. During the initial outbreak in Wuhan, China, various names were used for the virus; some names used by different sources included \"the coronavirus\" or \"Wuhan coronavirus\". The virus previously had a provisional name, 2019 novel coronavirus (2019-nCoV), and has also been called the human coronavirus 2019 (HCoV-19 or hCoV-19).  Severe acute respiratory syndrome coronavirus 2 (SARS‑CoV‑2) is a strain of coronavirus that causes COVID-19 (coronavirus disease 2019), the respiratory illness responsible for the ongoing COVID-19 pandemic. The original source of viral transmission to humans remains unclear, as does whether the virus became pathogenic before or after the spillover event. Differences between the bat coronavirus and SARS‑CoV‑2 suggest that humans may have been infected via an intermediate host; although the source of introduction into humans remains unknown. Research into the natural reservoir of the virus that caused the 2002–2004 SARS outbreak has resulted in the discovery of many SARS-like bat coronaviruses, most originating in horseshoe bats. Available evidence indicates that it is most likely of zoonotic origin and has close genetic similarity to bat coronaviruses, suggesting it emerged from a bat-borne virus.\n"
     ]
    }
   ],
   "source": [
    "import heapq\n",
    "summary_sentences = heapq.nlargest(10, sentence_scores, key=sentence_scores.get)\n",
    "\n",
    "summary = ' '.join(summary_sentences)\n",
    "print(summary)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "31e344b5",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
